---
title: "Metagenomic analysis of Limnohabitans populations in Lake Michigan"
author: "Ruben Props"
date: "Today"
output:
  html_document:
    code_folding: show
    css: report_styles.css
    highlight: haddock
    keep_md: yes
    theme: united
    toc: yes
    toc_float:
      collapsed: no
      smooth_scroll: yes
      toc_depth: 2
  pdf_document:
    toc: yes
editor_options: 
  chunk_output_type: console
---

# Data/Library loading

```{r setup, include = FALSE, warning = FALSE}
library("knitr")
knitr::opts_chunk$set(eval = TRUE, 
                      echo = TRUE, 
                      cache = TRUE,
                      include = TRUE,
                      
                      collapse = FALSE,
                      dependson = NULL,
                      engine = "R", # Chunks will always have R code, unless noted
                      error = TRUE,
                      fig.path="Figures/cached/",  # Set the figure options
                      fig.align = "center",
                      fig.width = 7, 
                      fig.height = 6,
                      dev.args=list(bg='transparent'))
# Load libraries
library('tidyr')
library('dplyr')
library('ggplot2')
library("scales")
library("grid")
library("scales")
library("gridExtra")
library("mclust")
library("reshape2")
library("Phenoflow")
library("phyloseq")
library("data.table")
library("ggrepel")
library("RColorBrewer")
library("igraph")
library("stringr")
library("UpSetR")
library("corrplot")
library("DESeq2")
library("SpiecEasi")
library("clusterProfiler")
source("functions.R")


fill_palette <- c(rev(brewer.pal(11, "BrBG")[1:4]),
                  brewer.pal(11, "PiYG")[9:10],
                  brewer.pal(9, "PuBu")[4:7]
                  )

set.seed(777)
```


```{r read-format data, warning = FALSE}
# Read data
mean_coverage <- read.table("./anvio_output/rebin/mean_coverage_selected_final.tsv", header = TRUE)
std_coverage <- read.table("./anvio_output/rebin/std_coverage_selected_final.tsv", header = TRUE)
bin_size <- read.table("./anvio_output/rebin/general_bins_summary_selected_final.tsv", header = TRUE)[, c(2,5)]
total_reads <- read.table("./anvio_output/sample_reads.tsv", header = TRUE)
read_length <- 150

# From wide to long format
mean_coverage_long <- tidyr::gather(mean_coverage, Sample_ID, coverage, 
                             Fa13_BD_MLB_DN:Su13_BD_MM15_SN_C, factor_key=TRUE)
mean_coverage_long[,2] <- gsub(mean_coverage_long[,2], pattern = "_C", 
                               replacement = "")

std_coverage_long <- tidyr::gather(std_coverage, Sample_ID, std_coverage, 
                            Fa13_BD_MLB_DN:Su13_BD_MM15_SN_C, 
                            factor_key=TRUE)
std_coverage_long[,2] <- gsub(std_coverage_long[,2], pattern = "_C", 
                            replacement = "")
 
coverage_data <- data.frame(mean_coverage_long, 
                            std_coverage = std_coverage_long[,3])

# Read and add metadata
meta <- read.csv2("metadata.csv")
meta$Sample_ID <- gsub(meta$Sample_ID, pattern = ".", replacement = "_", fixed = TRUE)
data_total <- left_join(coverage_data, total_reads, by = c("Sample_ID" = "sample"))
data_total <- left_join(data_total, bin_size, by = "bins")
data_total <- left_join(data_total, meta, by =  "Sample_ID")

# Calculate relative abundance of the bins
data_total$mean_rel_abundance <- 100*(data_total$coverage*data_total$bin_size)/(read_length*data_total$Total_reads)
data_total$upper_rel_abundance <- 100*((data_total$coverage+data_total$std_coverage)*data_total$bin_size)/(read_length*data_total$Total_reads)
data_total$lower_rel_abundance <- 100*((data_total$coverage-data_total$std_coverage)*data_total$bin_size)/(read_length*data_total$Total_reads)
```

# 1a. Environmental parameters

```{r, warning = FALSE, fig.width = 9, fig.height = 9, dpi = 500}
# import
meta_env <- readxl::read_excel(path = "./Figures/Table1.xlsx") %>% 
  gather(., measurement_type, measurement_value,`Temperature (°C)`:PAR) %>% 
  mutate(measurement_value = as.numeric(as.character(measurement_value)))

meta_env %>% 
  dplyr::filter(measurement_type %in% c("Chl (ug/L)", "TP ug/L", "SpCond (µS/cm)", "Temperature (°C)")) %>% 
  group_by(measurement_type, Site, Lake) %>% 
  summarise(median = median(measurement_value,  na.rm = TRUE)) %>% 
  droplevels() %>% 
  ggplot(., aes(x = Site, size = median, y = measurement_type, fill = median))+
  ggplot2::geom_count(color = "black", shape = 21, stat = "identity")+
  scale_size("", range = c(3.5,8), breaks=seq(1,10,by=3)) +
  facet_grid(.~Lake, scales = "free")+
  ggplot2::scale_fill_distiller("",palette="RdBu", na.value="white") + 
  theme_bw()+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
        axis.text.x = element_text(size = 14, angle = 45, hjust = 1),
      title=element_text(size=20), legend.text=element_text(size=12),
      legend.background = element_rect(fill="transparent"),
      strip.text=element_text(size=14), legend.position = "right",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))

meta_env %>% 
  dplyr::filter(measurement_type %in% c("Chl (ug/L)", "TP ug/L", "SpCond (µS/cm)", "Temperature (°C)")) %>% 
  group_by(measurement_type, Lake) %>%
  summarise(median = median(measurement_value,  na.rm = TRUE),
    sd = sd(measurement_value, na.rm = TRUE)) %>% 
  write.csv(., file = "./Figures/summary_env.csv", quote = FALSE, row.names = FALSE)


```


# 1b. Phylogeny

## Phylogenomic tree  


## 16S rRNA gene phylogenetic tree
![Annotated 16S rRNA gene phylogenetic tree](./Tree/16S/fasttree.png)

### Microdiversity in LimB

**At each iteration, SSU sequences were merged into one sequence if the identity of non-gapped positions in a global alignment was greater than 97%. A single SSU sequence (and its prior probability) was divided into two sequences if the second most probable base in more than 4% of all positions had a probability greater than 10%. In this way, sequences that evolved over iterations to be the same were merged, and sequences with evidence from the reads for multiple OTUs were duplicated and allowed to evolve as separate OTUs in future iterations.** [Reference](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2011-12-5-r44)  

The workflow for postprocessing the `EMIRGE` reconstructed 16S rRNA gene sequences:  

```{r EMIRGE-analysis-1, warning = FALSE, fig.width = 9, fig.height = 9, dpi = 500}
# Import seq-table that was generated after EMIRGE post-processing
EMIRGE_OTU <- read.table("./Tree/EMIRGE/read.info.txt", 
                         stringsAsFactors = FALSE)
colnames(EMIRGE_OTU) <- c("Sample",  "Sequence", "Prior", "Length", "Prior_norm")
EMIRGE_names <- read.table("./Tree/EMIRGE/EMIRGE_Limno.names")

EMIRGE_OTU_97 <- read.table("./Tree/EMIRGE/final.otu.emirge.pick.an.0.03.rep.names",
                            stringsAsFactors = FALSE)
EMIRGE_OTU_99 <- read.table("./Tree/EMIRGE/final.otu.emirge.pick.an.0.01.rep.names",
                            stringsAsFactors = FALSE)

# Filter out putative Limnohabitans OTUs
EMIRGE_OTU <- EMIRGE_OTU[EMIRGE_OTU$Sequence %in% EMIRGE_names$V1, ]

# Formate sample names
EMIRGE_OTU$Sample <- gsub(".A","", EMIRGE_OTU$Sample, fixed = TRUE)
EMIRGE_OTU$Sample <- gsub(".C","", EMIRGE_OTU$Sample, fixed = TRUE)

# Add OTU cluster results as new columns (0.97 level and 0.99 level)
EMIRGE_OTU$OTU_97 <- EMIRGE_OTU$Sequence

for(i in 1:nrow(EMIRGE_OTU_97)){
   EMIRGE_OTU$OTU_97[EMIRGE_OTU$Sequence %in% 
                            do.call(rbind, strsplit(EMIRGE_OTU_97$V2[i], ","))] <-
     EMIRGE_OTU_97$V3[i]
}

EMIRGE_OTU$OTU_99 <- EMIRGE_OTU$Sequence

for(i in 1:nrow(EMIRGE_OTU_99)){
  EMIRGE_OTU$OTU_99[EMIRGE_OTU$Sequence %in% 
                            do.call(rbind, strsplit(EMIRGE_OTU_99$V2[i], ","))] <-
     EMIRGE_OTU_99$V3[i]
}


# Calculate new abundance for the relative abundance of each OTU (0.97 and 0.99 level)
EMIRGE_OTU <- EMIRGE_OTU %>% dplyr::group_by(OTU_97, Sample) %>% 
  dplyr::mutate(Prior_norm_OTU97 = sum(Prior_norm))

EMIRGE_OTU <- EMIRGE_OTU %>% dplyr::group_by(OTU_99, Sample) %>% 
  dplyr::mutate(Prior_norm_OTU99 = sum(Prior_norm))

# Add metadata variables as new column
meta_em <- meta[-1]; meta_em$Sample_ID <- gsub("_",".", meta_em$Sample_ID)
meta_em <- meta_em %>% dplyr::distinct()
EMIRGE_OTU <- dplyr::left_join(EMIRGE_OTU, meta_em, by = c("Sample" = "Sample_ID"))

# to change
```

```{r EMIRGE-analysis-2, warning = FALSE, fig.width = 9, fig.height = 15, dpi = 500}
# Plot
p_emirge_abund_otu97 <- ggplot(EMIRGE_OTU, aes(x = Sample, y = 100*Prior_norm, fill = OTU_97))+
  theme_bw()+
  geom_bar(stat = "identity", size = 1, color = "black", alpha = 0.7)+
  scale_fill_brewer(palette = "Accent")+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
        axis.text.x = element_text(size = 14, angle = 45, hjust = 1),
      title=element_text(size=20), legend.text=element_text(size=12),
      legend.background = element_rect(fill="transparent"),
      strip.text=element_text(size=14), legend.position = "right",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  guides(fill=guide_legend(nrow = 3))+
  facet_grid(.~Site, scales = "free_x")+
  ylab("Length-normalized prior (%)")+
  xlab("")+
  ggtitle("97% clustered OTUs")
  # coord_trans(x = "atanh", y = "atanh")

# Plot
p_emirge_abund_otu99 <- ggplot(EMIRGE_OTU, aes(x = Sample, y = 100*Prior_norm, fill = OTU_99))+
  theme_bw()+
  geom_bar(stat = "identity", size = 1, color = "black", alpha = 0.7)+
  # geom_point(size = 4, shape = 21, color = "black", alpha = 0.7)+
  scale_fill_brewer(palette = "Accent")+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
        axis.text.x = element_text(size = 14, angle = 45, hjust = 1),
      title=element_text(size=20), legend.text=element_text(size=12),
      legend.background = element_rect(fill="transparent"),
      strip.text=element_text(size=14), legend.position = "right",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  guides(fill=guide_legend(nrow = 3))+
  facet_grid(.~Site, scales = "free_x")+
  ylab("Length-normalized prior (%)")+
  xlab("")+
  ggtitle("99% clustered OTUs")
  # coord_trans(x = "atanh", y = "atanh")

# Plot
p_emirge_abund_seq <- ggplot(EMIRGE_OTU, aes(x = Sample, y = 100*Prior_norm, fill = OTU_97))+
  theme_bw()+
  # geom_point(size = 4, shape = 21, color = "black", alpha = 0.7)+
  geom_bar(stat = "identity", size = 1, color = "black", alpha = 0.7)+
  scale_fill_brewer(palette = "Accent")+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
        axis.text.x = element_text(size = 14, angle = 45, hjust = 1),
      title=element_text(size=20), legend.text=element_text(size=12),
      legend.background = element_rect(fill="transparent"),
      strip.text=element_text(size=14), legend.position = "right",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  guides(fill=guide_legend(nrow = 3))+
  facet_grid(.~Site, scales = "free_x")+
  ylab("Length-normalized prior (%)")+
  xlab("")+
  ggtitle("Unique sequences")
  # coord_trans(x = "atanh", y = "atanh")

grid.arrange(p_emirge_abund_otu97, p_emirge_abund_otu99, p_emirge_abund_seq ,nrow = 3)

```

```{r EMIRGE-analysis-3, warning = FALSE, fig.width = 5, fig.height = 4, dpi = 500, dev = c("png","pdf")}
# Import blast results
blast_emirge <- read.csv("./Tree/EMIRGE/blast_table.csv", header = FALSE)[, 1:3]
colnames(blast_emirge) <- c("seq1", "seq2", "pid")
# Annotate based on 97% clusters
EMIRGE_OTU_annot <- EMIRGE_OTU %>% dplyr::select(Sequence, OTU_97) %>%
  distinct() %>% 
  left_join(., EMIRGE_OTU_97[, c("V3", "V4")], by = c("OTU_97" = "V3"))

# label sequence 1
blast_emirge_annot <- left_join(blast_emirge, EMIRGE_OTU_annot,
                              by = c("seq1" = "Sequence")) %>% 
  dplyr::filter(V4 != "not_limno")
colnames(blast_emirge_annot)[7] <- "seq1_label"
# label sequence 2
blast_emirge_annot <- left_join(blast_emirge_annot, EMIRGE_OTU_annot[, c(3,5)],
                              by = c("seq2" = "Sequence")) %>% 
  dplyr::filter(V4 != "not_limno")
colnames(blast_emirge_annot)[8] <- "seq2_label"

# Create interaction effect
blast_emirge_annot$label_interaction <- interaction(blast_emirge_annot$seq1_label,
                                                    blast_emirge_annot$seq2_label,
                                                    sep= "-")
# Only visualize within-group blast identities
subset_seq_labels <- c("LimB-LimB", "Lhab-A4-Lhab-A4")
svg("./Figures/16S_id.svg", height = 4, width = 5)
blast_emirge_annot %>% 
  dplyr::filter(label_interaction %in% subset_seq_labels &
                                       seq1 != seq2) %>% 
ggplot(aes(x = label_interaction, y = pid, fill = label_interaction))+
  geom_jitter(size = 2.5,
             color = "black", shape = 21, width = 0.1, alpha = 0.5)+
  geom_violin(alpha = 0.4, adjust = 1, draw_quantiles = TRUE, width = 0.3)+
  stat_summary(fun.data=mean_sdl, fun.args = list(mult = 1), 
                 geom="pointrange", color="black")+
  ylim(95,100)+
  scale_fill_manual("", values = rev(c("#e31a1c","#fdbf6f")))+
  theme_bw()+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
        axis.text.x=element_text(size=14),
        title=element_text(size=16), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        strip.text.x=element_text(size=18))+
  guides(fill = FALSE)+
  labs(x="")
dev.off()

blast_emirge_annot %>% dplyr::filter(label_interaction %in% subset_seq_labels &
                                       seq1 != seq2) %>% 
  group_by(label_interaction) %>% 
  summarize(mean(pid), sd(pid))

# Only look at the outlying sequence of LimB
# 10|JN626652.1.1352
blast_emirge_annot$label_interaction <- interaction(blast_emirge_annot$seq1_label,
                                                    blast_emirge_annot$seq2_label,
                                                    sep= "-")
# Only visualize within-group blast identities
subset_seq_labels <- c("LimB-LimB", "Lhab-A4-Lhab-A4")
blast_emirge_annot %>% dplyr::filter(label_interaction %in% subset_seq_labels &
                                      seq1 == "10|JN626652.1.1352") %>% 
  summarize(mean(pid), sd(pid))

```

# ANI analysis

```{r ANI-analysis, warning = FALSE, fig.width = 9, fig.height = 9, dpi = 500}
# read file
data.ANI <- read.table("./ANI/ANIb_percentage_identity.tab")

# Replace genome names by better annotated names
map_ani <- read.delim("./ANI/pyani_ref_names.tsv", stringsAsFactors = FALSE)
for(i in 1:nrow(map_ani)){
 colnames(data.ANI)[colnames(data.ANI) %in% map_ani$ani_file[i]] <- map_ani$ref_name[i]
 rownames(data.ANI)[rownames(data.ANI) %in% map_ani$ani_file[i]] <- map_ani$ref_name[i]
}

# Remove pnec and root genomes from dataframe
data.ANI <- data.ANI[grep("MAG|Lim", rownames(data.ANI)), grep("MAG|Lim", rownames(data.ANI))]

# Order y-axis according to phylogenetic tree order
ord_list_bin <- c(
  "MAG5.SP-M110-DD", "MAG2.FA-MLB-SN",
  "MAG3.FA-MLB-SN", "MAG4.FA-M110-DN",
  "MAG1.FA-MLB-DN", "MAG10.SU-M15-SN",
  "MAG6.SP-M15-SD","MAG8.SU-M110-DCMD",
  "Lim. sp. Rim11", "Lim. sp. 103DPR2",
  "MAG7.SU-MLB-SD", "Lim. sp. Rim47",
  "MAG9.SU-M15-SN", "Lim. sp. Rim28",
  "Lim. sp. 63ED37-2","Lim. sp. 2KL-27",
  "Lim. sp. 2KL-3", "Lim. sp. DM1",
  "Lim. sp. II-D5"
  )

# order rows and columns
data.ANI <- data.ANI[, order(match(colnames(data.ANI), ord_list_bin))]
data.ANI <- data.ANI[order(match(rownames(data.ANI), ord_list_bin)), ]

data.ANI <- get_upper_tri(data.ANI)

# melt to dataframe
df_pyani <- melt(as.matrix(data.ANI), na.rm = TRUE) # reshape into dataframe
names(df_pyani)[c(1:3)] <- c("bin1", "bin2", "ANI")
df_pyani[, 1] <- as.character(df_pyani[, 1]); df_pyani[, 2] <- as.character(df_pyani[, 2])
df_pyani$bin2 <- factor(df_pyani$bin2, levels = ord_list_bin)
df_pyani$bin1 <- factor(df_pyani$bin1, levels = rev(ord_list_bin))

# make plot
p_ani <- ggplot(data = df_pyani,
       aes(x = bin2, y = bin1, fill = 100*ANI))+ 
  geom_raster()+
  scale_fill_distiller(palette = "RdBu", name = "Average\nNucleotide\nIdentity (%)\n",
                       limits = 100*c(0.75,1.0), oob=squish) +
  geom_text(aes(label = round(100*ANI, 0)), size = 4.5)+
  xlab("")+
  ylab("")+
  scale_x_discrete(position = "top") +
  theme(axis.title=element_text(size=14), strip.text.x=element_text(size=14),
        legend.title=element_text(size=14), legend.text=element_text(size=14),
        # axis.text.y = element_blank(),
        axis.text.y = element_text(size = 14),
        axis.text.x = element_text(size=14, angle = 55, hjust = 0),
        title=element_text(size=20),
        plot.margin = unit(c(1.1,1.1,1.1,1.1), "cm"),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(fill = "transparent",colour = NA),
        plot.background = element_rect(fill = "transparent",colour = NA)
  )

png("ANI_heatmap.png", height= 9, width = 12, res = 300, units = "in", bg = "transparent")
print(p_ani)
dev.off()
```

# 1b. Investigate predicted generation time (MGT)
```{r MGT-1, warning = FALSE, fig.width = 9.5, fig.height = 4.5, dpi = 500, dev = c("png","pdf")}
MGT_df <- read.table("./Growthpred/GP_results.tsv", header = TRUE,
                     stringsAsFactors = FALSE, sep = "\t")

# Order genome_ids according to the phylogenetic clustering
MGT_df$Genome_ID <- factor(MGT_df$Genome_ID, levels = ord_list_bin)
MGT_df$Lineage <- factor(MGT_df$Lineage, levels = c("LimDEA_1", "LimDEA_2",
                                                    "LimB", "LimC"))

# Make barplot with st.dev to visualize MGT and optimal temperature
selected_points <- data.frame(Genome_ID = MGT_df$Genome_ID, 
                              ypos = c(rep(0.25, 10), rep(NA,9)))

p_MGT <- ggplot(MGT_df, aes(x = Genome_ID, y = log(2)/MGT, fill = Lineage, group = Genome_ID))+
  theme_bw()+
  geom_bar(alpha = 1, stat = "identity", color = "black",
           position = position_dodge(width = 1), width = 0.7)+
  scale_fill_manual(expression("Growth rate - "~"h"^-1),
                    values = c(rgb(red=t(col2rgb("#deebf7ff")), maxColorValue  = 255), 
                               rgb(red=t(col2rgb("#c6dbefff")), maxColorValue  = 255),
                               rgb(red=t(col2rgb("#9ecae1ff")), maxColorValue  = 255),
                               rgb(red=t(col2rgb("#6baed6ff")), maxColorValue  = 255)
                              ))+
  theme(axis.text=element_text(size=15, face = "bold"), axis.title=element_text(size=20),
        title=element_text(size=20), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        # axis.text.x = element_text(angle = 65, hjust = 1),
        strip.text.x=element_text(size = 18),
        legend.position="bottom",
        axis.text.x=element_text(size = 13, angle =45, hjust= 1),
        axis.title.x=element_blank(),
        plot.title = element_text(hjust = 0, size=18))+
  guides(fill=FALSE)+
  geom_errorbar(aes(ymin = log(2)/(MGT - sd.MGT), ymax = log(2)/(MGT + sd.MGT)), width = 0.15,
                position = position_dodge(width = 1))+
  ylab("")+
  xlab("")+
  ggtitle(expression("Growth rate - h"^-1))+
  geom_point(data = selected_points, aes(x = Genome_ID, y = ypos),
             shape = 25, fill = "black", col = "black", size = 3)

# svg(filename = "MGT_figure.svg", width = 9.5, height = 4)
print(p_MGT)
# dev.off()

# Make the same plot for optimal growth temperature
selected_points <- data.frame(Genome_ID = MGT_df$Genome_ID, 
                              ypos = c(rep(30, 10), rep(NA,9)))

p_Topt <- ggplot(MGT_df, aes(x = Genome_ID, y = Topt, fill = Lineage, group = Genome_ID))+
  theme_bw()+
  geom_bar(alpha = 0.4, stat = "identity", color = "black",
           position = position_dodge(width = 1), width = 0.7)+
  scale_fill_manual("Optimal growth temperature (°C)",
                    values = c("#deebf7ff", "#c6dbefff","#9ecae1ff",
                               "#6baed6ff"))+
  theme(axis.text=element_text(size=15, face = "bold"), axis.title=element_text(size=20),
        title=element_text(size=20), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        # axis.text.x = element_text(angle = 65, hjust = 1),
        strip.text.x=element_text(size = 18),
        legend.position="bottom",
        axis.text.x=element_text(size = 13, angle =45, hjust= 1),
        axis.title.x=element_blank(),
        plot.title = element_text(hjust = 0, size=18))+
  guides(fill=FALSE)+
  ylab("")+
  xlab("")+
  ggtitle("Optimal growth temperature (°C)")+
  geom_point(data = selected_points, aes(x = Genome_ID, y = ypos),
             shape = 25, fill = "black", col = "black", size = 3)

print(p_Topt)
```


# 1c. Network analysis based on 16S data

Get data from here: https://doi.org/10.6084/m9.figshare.7608854

```{r network_16S, warning = FALSE, fig.width = 9, fig.height = 8}
# import data
df_phy <- import_mothur(mothur_shared_file = "./Amplicon_16S/stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.an.unique_list.shared",
                        mothur_constaxonomy_file = "./Amplicon_16S/stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.an.unique_list.0.03.cons.taxonomy")

# Filter out 2013 samples
df_phy <- prune_samples(grep(pattern = ".", sample_names(df_phy), fixed = TRUE,
     value = TRUE), df_phy)
df_phy <- prune_samples(grep(pattern = "cD", sample_names(df_phy), fixed = TRUE,
     value = TRUE, invert = TRUE), df_phy)
df_phy <- prune_samples(grep(pattern = "ED", sample_names(df_phy), fixed = TRUE,
     value = TRUE, invert = TRUE), df_phy)

# Calculate diversity
diversity_df_phy <- Diversity_16S(df_phy, brea = FALSE, R = 100)

# Perform prevalence filtering
df_phy <- filter_taxa(df_phy, function(x) sum(x > 30) > (0.25*length(x)), TRUE)

# Run spiec-easi
sp_easi <- spiec.easi(df_phy, method='mb', lambda.min.ratio=1e-2,
                           nlambda=20, icov.select.params=list(rep.num=50))

ig.mb <- adj2igraph(sp_easi$refit,  vertex.attr = list(name=taxa_names(df_phy)))
vsize <- Biobase::rowMax(clr(t(otu_table(df_phy)), 1))+10
Lineage_rel <- tax_table(df_phy)[,"Rank2"]
Lineage_rel <- factor(Lineage_rel, levels = unique(Lineage_rel))

# OTUs that are Limnohabitans
limno_otus <- taxa_names(df_phy)[tax_table(df_phy)[,"Rank6"] == "betI_A"]
limno_otus <- limno_otus[!is.na(limno_otus)]

# Make Limno label
limno_labs <- c()
limno_labs[vertex.attributes(ig.mb)$name %in% limno_otus] <- "Limnohabitans sp."
limno_labs[is.na(limno_labs)] <- ""

# Plot network
p_16S_network <- plot_network_custom(ig.mb, df_phy, type='taxa',
             line_weight = 2, hjust = 0.5,
             point_size = 0.1, alpha = 0.01, label=NULL, label_size = 3.95)+
  scale_fill_brewer(palette = "Paired")+
  scale_color_brewer(palette = "Paired")+
  geom_point(aes(size = vsize, fill = Lineage_rel), alpha = 0.5,
             colour="black", shape=21)+
  guides(size = FALSE,
    fill  = guide_legend(title = "Phylum", override.aes = list(size = 5),
                         nrow = 4),
    color = FALSE)+
  theme(legend.position="bottom", legend.text=element_text(size=12),
        text = element_text(size = 12),
        plot.margin = unit(c(1,1,1,1), "cm"))+
  scale_size(range = c(5, 15))+
  geom_label_repel(aes(label = limno_labs), 
                   fontface = 'bold', color = 'black',
                   box.padding = 0.35, point.padding = 0.5,
                   segment.color = 'black',
                   size = 4,
                       # Width of the line segments.
                   segment.size = 1.5,
                   # Draw an arrow from the label to the data point.
                   arrow = arrow(length = unit(0.015, 'npc')),
                   nudge_x = -0.1,
                   nudge_y = 0.6
  )

print(p_16S_network)
```

# 2. Investigate MAG- and 16S-based abundances  

## 2.1. From coverage info (anvi'o)  

Formula used to calculate relative abundances:
$$Relative\ abundance =100*(\frac{mean\ coverage * bin\ size}{read\ length*total\ sample\ reads })$$

```{r plot-abundances-1, warning = FALSE, fig.width = 10, fig.height = 6,dev = c("png","pdf"), dpi = 500}
# Normalize for bin sizes
data_total_abund <- data_total %>% group_by(bins) %>% 
  mutate(norm_mean_rel_abundance = mean_rel_abundance/(bin_size/1e6),
         norm_upper_rel_abundance = upper_rel_abundance/(bin_size/1e6),
         norm_lower_rel_abundance = lower_rel_abundance/(bin_size/1e6)
         )

# Import bin name file used in manuscript
new_bin_names <- read.table("./anvio_output/rebin/general_bins_summary_selected_final.tsv", header = TRUE)[, c(2,3)]
data_total_abund <- left_join(data_total_abund,
                              new_bin_names, by = c("bins" = "bins"))

# Remove non-Limno bin
data_total_abund <- data_total_abund %>% 
  dplyr::filter(new_bin_name != "MAG.noLIM")

# Add putative limnohabitans lineage ID
data_total_abund <- dplyr::left_join(data_total_abund, MGT_df,
                                     by = c("new_bin_name" = "Genome_ID"))

# Order names for phylogenomic tree
data_total_abund$new_bin_name <- as.character(data_total_abund$new_bin_name)
data_total_abund$new_bin_name <- factor(data_total_abund$new_bin_name,
                                        levels = c(
                                          "MAG5.SP-M110-DD",
                                          "MAG2.FA-MLB-SN",
                                          "MAG3.FA-MLB-SN",
                                          "MAG4.FA-M110-DN",
                                          "MAG1.FA-MLB-DN",
                                          "MAG10.SU-M15-SN",
                                          "MAG6.SP-M15-SD",
                                          "MAG8.SU-M110-DCMD",
                                          "MAG7.SU-MLB-SD", 
                                          "MAG9.SU-M15-SN"
                                          )
)

# Plot abundance distributions of all bins
p_season1 <- ggplot(data = data_total_abund, 
                    aes(x = new_bin_name, y = norm_mean_rel_abundance/(bin_size/1e6),
                        fill = Lineage))+
  # geom_point(size = 4, shape = 21, alpha = 0.7)+
  geom_boxplot(alpha = 1, outlier.shape = 21, outlier.size = 4, size = 0.5)+
  # geom_violin(scale = "width")+
  # stat_summary(fun.data=mean_sdl, fun.args = list(mult = 1), 
                 # geom="pointrange", color="#333333", size = 0.5)+
  theme_bw()+
  scale_fill_manual(values = c(rgb(red=t(col2rgb("#deebf7ff")),
                                   maxColorValue  = 255), 
                               rgb(red=t(col2rgb("#c6dbefff")), 
                                   maxColorValue  = 255),
                               rgb(red=t(col2rgb("#9ecae1ff")), 
                                   maxColorValue  = 255),
                               rgb(red=t(col2rgb("#6baed6ff")), 
                                   maxColorValue  = 255)
                              ))+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
        title=element_text(size=20), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        axis.text.x = element_text(angle = 90, hjust = 1),
        strip.text=element_text(size=18),
        axis.line = element_line(size = 1, colour = "grey80"),
        panel.border = element_blank())+
  ylab("Normalized relative abundance (%)")+
  scale_y_sqrt(limits = c(0,3))+
  xlab("")

p_season1

```

## 2.1. From mapped reads info (BBmap - `pileup.sh`)  

From: https://doi.org/10.1186/s13059-015-0834-7  

*"The relative abundance of each MAG was estimated using the fraction of reads in each sample mapping to the respective MAG. Normalized on the size of that bin, this yielded the measure fraction of reads per nucleotide in bin. This measure was chosen since it is comparable across samples with varying sequencing output and different bin sizes. Using the CONCOCT input table, multiplying the average coverage per nucleotide with the length of the contig in question and summing over all contigs within a bin and within a sample gave the number of reads per bin within a sample. The fraction of reads in each sample mapping to each bin was then calculated by dividing this value with the total number of reads from each sample, after having removed duplicated reads."*

```{r plot-abundances-2, dpi = 500, warning = FALSE, fig.width = 15, fig.height = 15}
# Generated from SAM file with pileup.sh from BBmap toolset
df_map_reads <- read.delim("./anvio_output/collection_vizbin_pileup_cov.tsv",
                           stringsAsFactors = FALSE)
df_map_reads$Sample <- gsub(df_map_reads$Sample, pattern = "_cov.tsv",
                            replacement = "")

# This file was generated in anvio with anvi-export-collections
lab_map_reads <- read.delim("./anvio_output/collection_vizbin.tsv",
                            header = FALSE)
colnames(lab_map_reads) <- c("contig_split", "Bin")

# Select Limnohabitans bins
lab_map_reads <- lab_map_reads[lab_map_reads$Bin %in% data_total$bins, ]
lab_map_reads$contig <- paste(
  do.call(rbind, strsplit(as.character(lab_map_reads$contig_split), "_"))[, 1],
  do.call(rbind, strsplit(as.character(lab_map_reads$contig_split), "_"))[, 2],
  sep = "_"
  )

lab_map_reads <- lab_map_reads[ ,2:3] %>% distinct()

# Merge dataframes
df_map_merged <- left_join(lab_map_reads, df_map_reads, by = c("contig" = "X.ID")
)
df_map_merged[, 4:13] <- apply(df_map_merged[, 4:13], 2, FUN = function(x) as.numeric(x))

df_map_merged <- df_map_merged %>% group_by(Bin, Sample) %>% summarise(sum_map_read = sum(Plus_reads))

# Merge with metadata/bin sizes
meta$Sample <- gsub(".C", "", meta$Sample, fixed = TRUE)
meta$Sample <- gsub(".A", "", meta$Sample, fixed = TRUE)
df_map_merged$Sample <- gsub(".C", "", df_map_merged$Sample, fixed = TRUE)
df_map_merged <- left_join(df_map_merged, bin_size, by = c("Bin" = "bins"))
df_map_merged <- left_join(df_map_merged, meta, "Sample")
total_reads$sample <- gsub("_", ".", fixed = TRUE,total_reads$sample)
df_map_merged <- left_join(df_map_merged, total_reads, c("Sample" = "sample"))

df_map_merged <- df_map_merged %>% group_by(Bin, Sample) %>%
  mutate(rel_abundance = 100*sum_map_read/Total_reads)

df_map_merged <- df_map_merged %>% group_by(Bin, Sample) %>%
  mutate(rel_norm_abundance = 100*sum_map_read/Total_reads/(bin_size/1e6))

# Rename bin names to more sensible names
# Add extra column with new bin names
new_bin_names <- read.table("./anvio_output/rebin/general_bins_summary_selected_final.tsv", header = TRUE)[, c(2,3)]
df_map_merged <- left_join(df_map_merged, new_bin_names, by = c("Bin" = "bins"))
df_map_merged$new_bin_name <- as.character(df_map_merged$new_bin_name)
df_map_merged$new_bin_name <- factor(df_map_merged$new_bin_name, levels =
                                      c("MAG1.FA-MLB-DN","MAG2.FA-MLB-SN",
                                        "MAG3.FA-MLB-SN", "MAG4.FA-M110-DN",
                                        "MAG5.SP-M110-DD","MAG6.SP-M15-SD",
                                        "MAG7.SU-MLB-SD","MAG8.SU-M110-DCMD",
                                        "MAG9.SU-M15-SN","MAG10.SU-M15-SN"))
df_map_merged$Site <- as.character(df_map_merged$Site)
df_map_merged$Site <- gsub("110", "Lake Michigan\nsite M110", df_map_merged$Site)
df_map_merged$Site <- gsub("15", "Lake Michigan\nsite M15", df_map_merged$Site)
df_map_merged$Site <- gsub("Buoy", "Muskegon Lake", df_map_merged$Site)
df_map_merged$Site <- factor(df_map_merged$Site, levels = c("Muskegon Lake",
                                                            "Lake Michigan\nsite M15",
                                                            "Lake Michigan\nsite M110"))
df_map_merged$Season <- as.character(df_map_merged$Season)
df_map_merged$Season <- factor(df_map_merged$Season, levels = c("Spring", "Summer","Fall"))

# Remove non-limno bin
df_map_merged <- df_map_merged %>% dplyr::filter(new_bin_name != "MAG.noLIM")

# Make plots
p_abs2 <- ggplot(data = df_map_merged, aes(x = new_bin_name, y = rel_norm_abundance, fill = new_bin_name))+
  theme_bw()+
  scale_fill_manual("", values = fill_palette)+
  geom_jitter(size = 4, shape = 21, color = "black", alpha = 0.7, width = 0.15)+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
      title=element_text(size=20), legend.text=element_text(size=12),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_blank(),
      strip.text=element_text(size=14), legend.position = "bottom",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylab(paste0("Norm. relative abundance (%)"))+
  xlab("")+
  guides(fill=guide_legend(nrow = 3))+
  facet_grid(Season~Site, scales ="free")+
  scale_y_continuous(labels=scaleFUN, limits = c(0,1.25))+
  coord_trans(y = "sqrt")

p_abs2
```


# 3. Expression analysis  

Workflow:
  1. Mapedp reads with `bwa mem`
  2. Extracted mapping statistics with `pileup.sh` from `BBtools`  
  3. Get gene length through the DESMAN command: `python $DESMAN/scripts/Lengths.py -i CDS.fa > CDS.len`

The file needed to map `genome_ids` to `gene_ids` was generated with this code individually for each IMG `genes.fna` file and the concatenated with `cat`:  
```
grep ">" 2757320404.genes.fna | sed "s/>//g" | awk '{ print $1, $2 }' > 2757320404.genes.ids
awk -F ' ' -v samname="2757320404" '{$(1)=samname FS $1;}1' OFS=" " 2757320404.genes.ids > 2757320404.genes.names
```

## Import and process data  


```{r import and process data, dpi = 300, warning = FALSE, fig.width = 7, fig.height = 7}
expr_cov <- read.delim("./metaT/metaT_pileup.tsv", header = TRUE,
                       stringsAsFactors = FALSE)

# Merge gene annotations from all genomes in one file
file_list <- list.files(path = "./IMG_annotation", recursive = FALSE, 
                        pattern = "IMG", full.names = TRUE)

# Import annotations of all MAG genes
merged_file <- merge_annotations(file_list[1:10], genoid_seqid = FALSE)
merged_file$ko_id <- gsub(merged_file$ko_id, pattern = "KO:", 
                          replacement = "")

# import KO annotation hierarchy 
ko_df <- format_ko(path = "./Mapping_files/ko00000.keg")

# Annotated merged_file
merged_file_annot <- dplyr::left_join(merged_file, ko_df, by = "ko_id")

# Genes that were not annotated can be assigned a genome ID using another file
# made from the genes.fna IMG generated files
genome_gene_map <- read.table("IMG_annotation/genome_geneids.txt",
                              header = FALSE)[, 1:2]
colnames(genome_gene_map) <- c("Genome_ID", "gene_oid")
genome_gene_map <- data.frame(apply(genome_gene_map, 2, as.character))

# Add genome_ID to all genes
expr_cov <- dplyr::left_join(expr_cov, genome_gene_map, by = "gene_oid")

# Annotate this expression table with Kegg Orthology
expr_cov <- dplyr::left_join(expr_cov, merged_file[, c("gene_oid","ko_id")], by = c("gene_oid"))
expr_cov$Plus_reads <- as.integer(expr_cov$Plus_reads)
expr_cov$Length <- as.integer(expr_cov$Length)
expr_cov <- expr_cov[!is.na(expr_cov$Plus_reads),]

expr_cov_long <- dplyr::mutate(expr_cov, 
                               reads_per_kb = Plus_reads/Length/1000
                               )
expr_cov_long <- expr_cov_long %>% group_by(Sample) %>% 
  mutate(RPK_scaling = sum(reads_per_kb)/1e6
         )
expr_cov_long <- expr_cov_long %>% 
  mutate(TPM = reads_per_kb/RPK_scaling
         )

# Now add the metadata to this long dataframe
# Metadata file
meta_metaT <- distinct(meta[, 2:nrow(meta)])
meta_metaT$Sample_ID <- gsub(meta_metaT$Sample_ID, pattern="_", replacement = ".")
rownames(meta_metaT) <- meta_metaT$Sample_ID
expr_cov_long <- left_join(expr_cov_long, meta_metaT[, 1:11], 
                           by = c("Sample" = "Sample_ID"))
expr_cov_long$Genome_ID <- as.factor(expr_cov_long$Genome_ID)

# Remove duplicate rows
expr_cov_long <- expr_cov_long %>% distinct()

# rename column
colnames(expr_cov_long)[colnames(expr_cov_long) == "Plus_reads"] <- "mapped_reads"
```

## PCA analysis

```{r metaT-disper, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 5, dev=c("png","pdf")}
# For each MAG calculate the PCA in expression profiles across samples
for(i in 1:nlevels(expr_cov_long$Genome_ID)){
# Perform PCA on gene-length normalize data (TPM scaling)
pca_exp_table <- expr_cov_long %>% 
  dplyr::filter(Genome_ID == levels(expr_cov_long$Genome_ID)[i]) %>% 
  dplyr::select(Sample, gene_oid, mapped_reads) %>% 
  distinct() %>% 
  tidyr::spread(gene_oid, mapped_reads)
tmp_row <- pca_exp_table$Sample
pca_exp_table <- as.matrix(pca_exp_table[, -1])
rownames(pca_exp_table) <- tmp_row
pca_exp_table <- pca_exp_table[, colSums(pca_exp_table) != 0]

# Perform TMMwsp transformation
# tmp_gene_length <- expr_cov %>% 
#   dplyr::filter(Genome_ID == levels(expr_cov_long$Genome_ID)[i] & gene_oid %in% colnames(pca_exp_table)) %>% 
#   dplyr::select(gene_oid, Length) %>% 
#   distinct()
pca_dgelist <- DGEList(counts = pca_exp_table)
pca_norm <- edgeR::calcNormFactors(pca_dgelist, method = "TMMwsp")
pca_norm <- cpm(pca_norm, log = TRUE)

# Perform PCA
pca_exp_res <- prcomp(t(pca_exp_table), center = TRUE)

# Store results
if (i == 1)
  results_PCA <- data.frame(
  Sample = rownames(pca_exp_res$rotation),
  pca_exp_res$rotation,
  Genome = levels(expr_cov_long$Genome_ID)[i]
  )
  else {
  results_PCA <- rbind(
  results_PCA,
  data.frame(
  Sample = rownames(pca_exp_res$rotation),
  pca_exp_res$rotation,
  Genome = levels(expr_cov_long$Genome_ID)[i]
  )
  )
  }

  remove(pca_exp_table)
}

# Merge with correct sample names
new_bin_names2 <- read.table("./anvio_output/rebin/general_bins_summary_selected_final.tsv", header = TRUE)[, c(3,8:10)]; 
new_bin_names2$IMG_taxID <- as.character(new_bin_names2$IMG_taxID)
results_PCA <- left_join(results_PCA, new_bin_names2, 
                          by = c("Genome" = "IMG_taxID"))
results_PCA$new_bin_name <- as.character(results_PCA$new_bin_name)
results_PCA$new_bin_name <- factor(results_PCA$new_bin_name, levels =
                                      c("MAG5.SP-M110-DD","MAG2.FA-MLB-SN",
                                        "MAG3.FA-MLB-SN", "MAG4.FA-M110-DN",
                                        "MAG1.FA-MLB-DN", "MAG10.SU-M15-SN",
                                        "MAG6.SP-M15-SD", "MAG8.SU-M110-DCMD",
                                        "MAG7.SU-MLB-SD","MAG9.SU-M15-SN"))

# Add metadata
results_PCA <- left_join(results_PCA, meta_metaT, by = c("Sample" = "Sample_ID"))
results_PCA$Season <- factor(results_PCA$Season, levels = c("Spring", "Summer", "Fall"))
results_PCA$Site <- gsub("Buoy","Muskegon Lake", results_PCA$Site)
results_PCA$Site <- gsub("110","Lake Michigan\nsite M110", results_PCA$Site)
results_PCA$Site <- gsub("15","Lake Michigan\nsite M15", results_PCA$Site)

# Plot
p_exp_PCA <- results_PCA %>%   
  dplyr::filter(Depth != "Mid") %>%
  ggplot(aes(x = PC1, y = PC2, fill = Season, shape = Site))+
  scale_shape_manual("", values = c(21, 22, 25))+
  geom_point(size = 3.5, alpha = 0.7)+
  facet_wrap(.~new_bin_name, ncol = 5)+
  scale_fill_brewer("", palette = "Accent")+
  theme_bw()+
  theme(axis.text.y=element_text(size=12), axis.title=element_text(size=13),
        axis.text.x=element_text(size=12),
        title=element_text(size=12), legend.text=element_text(size=12),
        legend.background = element_rect(fill="transparent"),
        strip.text.x=element_text(size=14))+
  labs(x="PC1", y = "PC2")+
  guides(fill = guide_legend(override.aes = list(shape = 22)))
 
png("./Figures/PCA_expression.png", width = 14, height = 6, units = "in", res = 500)
print(p_exp_PCA)
dev.off()

p_exp_PCA2 <- results_PCA %>%   
  dplyr::filter(Site == "Lake Michigan\nsite M110" & Depth != "Mid") %>%
  ggplot(aes(x = PC1, y = PC2, fill = Season, shape = Depth))+
  scale_shape_manual("", values = c(21, 22, 25))+
  geom_point(size = 3.5, alpha = 0.7)+
  facet_wrap(.~new_bin_name, ncol = 5)+
  scale_fill_brewer("", palette = "Accent")+
  theme_bw()+
  theme(axis.text.y=element_text(size=12), axis.title=element_text(size=13),
        axis.text.x=element_text(size=12),
        title=element_text(size=12), legend.text=element_text(size=12),
        legend.background = element_rect(fill="transparent"),
        strip.text.x=element_text(size=14))+
  labs(x="PC1", y = "PC2")+
  guides(fill = guide_legend(override.aes = list(shape = 22)))
 
png("./Figures/PCA_expression2.png", width = 14, height = 6, units = "in", res = 500)
print(p_exp_PCA2)
dev.off()
```

## Run DESeq  

__Important DESeq2 reminder:__   
__In order to benefit from the default settings of the package, you should put__ 
__the variable of interest at the end of the formula and make sure the control__ 
__level is the first level.__ 

```{r expression analysis, dpi = 500, warning = FALSE, fig.width = 14, fig.height = 10}
# Select surface samples to run DESeq2 on
sel_env <- meta_metaT[, "Depth"] == "Surface" 
sum(sel_env) # 15 samples
meta_metaT_subs <- meta_metaT[sel_env, ] # subset metadata
expr_cov_long_sb <- expr_cov_long %>% 
  dplyr::filter(Sample %in% meta_metaT_subs[,"Sample_ID"])

# Format data for DESeq2
## Put count matrices in list including a count matrix for each bin 
expr_cov_bins <- list()
for(i in 1:nlevels(expr_cov_long_sb$Genome_ID)){
  expr_cov_bins[[i]] <-  expr_cov_long_sb %>% 
  dplyr::filter(Genome_ID == levels(expr_cov_long_sb$Genome_ID)[i]) %>% 
  dplyr::select(gene_oid, Sample, mapped_reads) %>% 
  distinct() %>% 
  tidyr::spread(Sample, mapped_reads)
  r.bin <- expr_cov_bins[[i]]$gene_oid
  expr_cov_bins[[i]] <- as.matrix(expr_cov_bins[[i]][, -1])
  rownames(expr_cov_bins[[i]]) <- r.bin
}

# Check order of colnames in count and rownames in metadata matrix 
# and make sure these are in the same order!
meta_metaT_subs <- as.matrix(meta_metaT_subs)
meta_metaT_subs <- meta_metaT_subs[match(colnames(expr_cov_bins[[1]]), rownames(meta_metaT_subs)), ]
all(rownames(meta_metaT_subs) %in% colnames(expr_cov_bins[[1]]))

# Perform DESeq2 for differential abundance testing for each genome separately

## Season effect
General_deseq_results_season <- list()
deseq_comparisons_season <- list()
for(i in 1:nlevels(expr_cov_long_sb$Genome_ID)){
  cat(" --- Running DESeq2 on Genome_ID:",levels(expr_cov_long_sb$Genome_ID)[i], "\n",sep = " ")

  # Test for season but controlling for site
  dds <- DESeq2::DESeqDataSetFromMatrix(countData = expr_cov_bins[[i]],
                              colData = meta_metaT_subs,
                              design= ~ Site + Season) 
  # Run DESeq
  dds <- DESeq2::DESeq(dds, quiet = TRUE)
  
  # Calculate contrasts for all comparisons
  comp1 <- DESeq2::results(dds, contrast=c("Season","Fall","Summer"))[order(results(dds)$padj), ]
  comp2 <- DESeq2::results(dds, contrast=c("Season","Fall","Spring"))[order(results(dds)$padj), ]
  comp3 <- DESeq2::results(dds, contrast=c("Season","Spring","Summer"))[order(results(dds)$padj), ]
  
  # Store data in single list with dataframes
  General_deseq_results_season[[i]] <- rbind(comp1, comp2, comp3)
  deseq_comparisons_season[[i]] <- data.frame(comparison = 
                                           c(rep("Fall-Summer", nrow(comp1)),
                                           rep("Fall-Spring", nrow(comp2)),
                                           rep("Spring-Summer", nrow(comp3))),
                                         design = "~ Site + Season"
                                         )
}

## Site effect
General_deseq_results_site <- list()
deseq_comparisons_site <- list()
for(i in 1:nlevels(expr_cov_long$Genome_ID)){
  cat(" --- Running DESeq2 on Genome_ID:",levels(expr_cov_long$Genome_ID)[i], "\n",sep = " ")
  dds <- DESeq2::DESeqDataSetFromMatrix(countData = expr_cov_bins[[i]],
                              colData = meta_metaT_subs,
                              design= ~ Season + Site) # Test for site but controlling for season
  dds <- DESeq2::DESeq(dds, quiet = TRUE)
  
    # Calculate contrasts for all comparisons
  comp1 <- DESeq2::results(dds, 
                   contrast=c("Site","Buoy","110"))[order(results(dds)$padj), ]
  comp2 <- DESeq2::results(dds, 
                   contrast=c("Site","Buoy","15"))[order(results(dds)$padj), ]
  comp3 <- DESeq2::results(dds, 
                   contrast=c("Site","15","110"))[order(results(dds)$padj), ]
  
  # Store data in single list with dataframes
  General_deseq_results_site[[i]] <- rbind(comp1, comp2, comp3)
  deseq_comparisons_site[[i]] <- data.frame(comparison = 
                                           c(rep("Muskegon Lake\nVs\nM110",
                                                 nrow(comp1)),
                                           rep("Muskegon Lake\nVs\nM15", 
                                               nrow(comp2)),
                                           rep("M15\nVs\nM110", 
                                               nrow(comp3))),
                                         design = "~ Season + Site"
  )
}

# Depth effect at M110
## Select M110 station to run depth comparison of gene expression
sel_env <- meta_metaT[, "Site"] == "110" & meta_metaT[, "Depth"] != "Mid" & meta_metaT[, "Season"] != "Spring" 
sum(sel_env) # 6 samples
meta_metaT_subs <- meta_metaT[sel_env, ] # subset metadata
expr_cov_long_sb <- expr_cov_long %>% 
  dplyr::filter(Sample %in% meta_metaT_subs[,"Sample_ID"])

# Format data for DESeq2
## Put count matrices in list including a count matrix for each bin 
expr_cov_bins <- list()
for(i in 1:nlevels(expr_cov_long_sb$Genome_ID)){
  expr_cov_bins[[i]] <-  expr_cov_long_sb %>% 
  dplyr::filter(Genome_ID == levels(expr_cov_long_sb$Genome_ID)[i]) %>% 
  dplyr::select(gene_oid, Sample, mapped_reads) %>% 
  distinct() %>% 
  tidyr::spread(Sample, mapped_reads)
  r.bin <- expr_cov_bins[[i]]$gene_oid
  expr_cov_bins[[i]] <- as.matrix(expr_cov_bins[[i]][, -1])
  rownames(expr_cov_bins[[i]]) <- r.bin
}

## Depth effect
General_deseq_results_depth <- list()
deseq_comparisons_depth <- list()
for(i in 1:nlevels(expr_cov_long_sb$Genome_ID)){
  cat(" --- Running DESeq2 on Genome_ID:",levels(expr_cov_long_sb$Genome_ID)[i], "\n",sep = " ")

  # Test for Depth but controlling for site
  dds <- DESeq2::DESeqDataSetFromMatrix(countData = expr_cov_bins[[i]],
                              colData = meta_metaT_subs,
                              design= ~ Time + Depth) 
  # Run DESeq
  dds <- DESeq2::DESeq(dds, quiet = TRUE)
  
  # Calculate contrasts for all comparisons
  comp1 <- DESeq2::results(dds, contrast=c("Depth","Surface","Deep"))[order(results(dds)$padj), ]
  
  # Store data in single list with dataframes
  General_deseq_results_depth[[i]] <- comp1
  deseq_comparisons_depth[[i]] <- data.frame(comparison = 
                                           rep("Surface-Deep", nrow(comp1)),
                                         design = "~ Site + Depth"
                                         )
}

# Pool results for each genome into a single result dataframe
res_deseq <- data.frame()

for(i in 1:length(General_deseq_results_season)){
  tmp <- data.frame(gene_oid = General_deseq_results_season[[i]]@rownames,
                    baseMean = General_deseq_results_season[[i]]@listData$baseMean,
                    log2FoldChange = General_deseq_results_season[[i]]@listData$log2FoldChange,
                    pvalue = General_deseq_results_season[[i]]@listData$pvalue,
                    padj = General_deseq_results_season[[i]]@listData$padj,
                    Genome_ID = levels(expr_cov_long$Genome_ID)[i],
                    Comparison = deseq_comparisons_season[[i]]$comparison,
                    Design = deseq_comparisons_season[[i]]$design)
  if(i == 1) res_deseq <- tmp else{
    res_deseq <- rbind(res_deseq, tmp)
  }
}

for(i in 1:length(General_deseq_results_site)){
  tmp <- data.frame(gene_oid = General_deseq_results_site[[i]]@rownames,
                    baseMean = General_deseq_results_site[[i]]@listData$baseMean,
                    log2FoldChange = General_deseq_results_site[[i]]@listData$log2FoldChange,
                    pvalue = General_deseq_results_site[[i]]@listData$pvalue,
                    padj = General_deseq_results_site[[i]]@listData$padj,
                    Genome_ID = levels(expr_cov_long$Genome_ID)[i],
                    Comparison = deseq_comparisons_site[[i]]$comparison,
                    Design = deseq_comparisons_site[[i]]$design)
  res_deseq <- rbind(res_deseq, tmp)
}

for(i in 1:length(General_deseq_results_depth)){
  tmp <- data.frame(gene_oid = General_deseq_results_depth[[i]]@rownames,
                    baseMean = General_deseq_results_depth[[i]]@listData$baseMean,
                    log2FoldChange = General_deseq_results_depth[[i]]@listData$log2FoldChange,
                    pvalue = General_deseq_results_depth[[i]]@listData$pvalue,
                    padj = General_deseq_results_depth[[i]]@listData$padj,
                    Genome_ID = levels(expr_cov_long$Genome_ID)[i],
                    Comparison = deseq_comparisons_depth[[i]]$comparison,
                    Design = deseq_comparisons_depth[[i]]$design)
  res_deseq <- rbind(res_deseq, tmp)
}

# Only retain significantly differentially expressed genes
res_deseq <- res_deseq %>% dplyr::filter(padj < 0.01)

new_bin_names2 <- read.table("./anvio_output/rebin/general_bins_summary_selected_final.tsv", header = TRUE)[, c(3,8:10)]; new_bin_names2$IMG_taxID <- as.character(new_bin_names2$IMG_taxID)
res_deseq <- left_join(res_deseq, new_bin_names2, by = c("Genome_ID" = "IMG_taxID"))
res_deseq$new_bin_name <- as.character(res_deseq$new_bin_name)
res_deseq$new_bin_name <- factor(res_deseq$new_bin_name, levels =
                                      c("MAG1.FA-MLB-DN","MAG2.FA-MLB-SN",
                                        "MAG3.FA-MLB-SN", "MAG4.FA-M110-DN",
                                        "MAG5.SP-M110-DD", "MAG6.SP-M15-SD",
                                        "MAG7.SU-MLB-SD", "MAG8.SU-M110-DCMD",
                                        "MAG9.SU-M15-SN", "MAG10.SU-M15-SN"))
res_deseq$regulation <- res_deseq$log2FoldChange > 0
res_deseq$regulation[res_deseq$regulation == "TRUE"] <- "upregulation"
res_deseq$regulation[res_deseq$regulation == "FALSE"] <- "downregulation"

# Add KO/COG/pfam/tigrfam gene annotations to DeSEQ2 results
res_deseq_anott <- dplyr::left_join(res_deseq, merged_file_annot, by = c("gene_oid"))

metaT_reads_mapped <- expr_cov_long %>% 
  group_by(Sample, Genome_ID) %>% 
  dplyr::summarise(sum_mapped_reads = sum(mapped_reads))

metaT_reads_mapped <- left_join(metaT_reads_mapped, new_bin_names2, by = c("Genome_ID" = "IMG_taxID"))
metaT_reads_mapped$new_bin_name <- as.character(metaT_reads_mapped$new_bin_name)
metaT_reads_mapped$new_bin_name <- factor(metaT_reads_mapped$new_bin_name, 
                                          levels =
                                      c("MAG1.FA-MLB-DN","MAG2.FA-MLB-SN",
                                        "MAG3.FA-MLB-SN", "MAG4.FA-M110-DN",
                                        "MAG5.SP-M110-DD", "MAG6.SP-M15-SD",
                                        "MAG7.SU-MLB-SD", "MAG8.SU-M110-DCMD",
                                        "MAG9.SU-M15-SN", "MAG10.SU-M15-SN"))

# p_deseq_reads <- ggplot2::ggplot(metaT_reads_mapped, aes(x = new_bin_name, y = sum_mapped_reads, fill = new_bin_name))+
#   geom_bar(color = "black", stat = "identity")+
#   theme_bw()+
#   scale_fill_brewer(palette = "Paired")+
#   theme(axis.text.x = element_blank(),
#         axis.text.y = element_text(size = 14),
#         legend.title = element_blank(),
#         axis.title = element_text(size = 14),
#         strip.text = element_text(size = 12),
#         axis.title.x = element_blank())+
#   ylab("Number of mapped reads")+
#   xlab("")+
#   facet_wrap(~Sample, nrow = 4)
# 
# print(p_deseq_reads)
```

#DESeq2::rlog

```{r plot-deseq, dpi = 500, warning = FALSE, fig.width = 15, fig.height = 10}
selected_ko <- c("Carbohydrate metabolism", "Energy metabolism",
                 "Amino acid metabolism", "Membrane transport",
                 "Replication and repair", "Translation",
                 "Energy metabolism")

# for(genome in levels(unique(res_deseq_anott$new_bin_name))){
p_deseq_1 <- res_deseq_anott %>% 
  dplyr::filter(!is.na(ko_id) & ko_level_B %in% selected_ko & Design == "~ Season + Site") %>% 
  ggplot2::ggplot(aes(x = new_bin_name, fill = regulation, y = log2FoldChange))+
  # geom_point(size = 4, shape = 21)+
  geom_boxplot(outlier.shape = NA)+
  # geom_bar(color = "black")+
  theme_bw()+
  scale_fill_brewer("", palette = "Paired")+
  theme(axis.text.x =  element_text(size = 12),
        axis.text.y = element_text(size = 14),
        legend.title = element_blank(),
        axis.title = element_text(size = 14),
        strip.text = element_text(size = 14)) +
  ylab("log2fold change")+
  xlab("")+
  facet_grid(.~ko_level_B)+
  guides(fill = FALSE)+
  # ggtitle("Muskegon lake vs. M110")+
  coord_flip()

print(p_deseq_1)
```

### Overview figure

```{r DEseq-overview, dpi = 500, warning = FALSE,fig.width = 7, fig.height = 5, dev=c("png","pdf")}
res_deseq_anott_changed <- res_deseq_anott
res_deseq_anott_changed$new_bin_name <- factor(res_deseq_anott_changed$new_bin_name,
                                               levels =
                                      c("MAG5.SP-M110-DD","MAG2.FA-MLB-SN",
                                        "MAG3.FA-MLB-SN", "MAG4.FA-M110-DN",
                                        "MAG1.FA-MLB-DN", "MAG10.SU-M15-SN",
                                        "MAG6.SP-M15-SD", "MAG8.SU-M110-DCMD",
                                        "MAG7.SU-MLB-SD","MAG9.SU-M15-SN"))

  res_deseq_anott_changed %>% 
  dplyr::select(gene_oid, log2FoldChange, new_bin_name, Comparison, Design) %>% 
  dplyr::filter(Design == "~ Season + Site" 
                & Comparison == "Muskegon Lake\nVs\nM110") %>% 
  # dplyr::filter(new_bin_name == "MAG8.SU-M110-DCMD") %>% 
    distinct() %>% 
    group_by(new_bin_name) %>% 
  summarise(nrow = length(log2FoldChange))
  
p_deseq_overview <- res_deseq_anott_changed %>% 
  dplyr::select(gene_oid, log2FoldChange, new_bin_name, Comparison, Design) %>% 
  dplyr::filter(Design == "~ Season + Site" 
                & Comparison == "Muskegon Lake\nVs\nM110") %>% 
  ggplot2::ggplot(aes(x = new_bin_name, y= abs(log2FoldChange)))+
  geom_boxplot(fill = "#333333", alpha = 0.5, outlier.shape = NA, size = 1.25)+
  theme_bw()+
  scale_fill_brewer("", palette = "Paired")+
  theme(axis.text.x =  element_text(size = 14),
        axis.text.y = element_text(size = 14),
        legend.title = element_blank(),
        axis.title = element_text(size = 14),
        strip.text = element_text(size = 14))+
  ylab("log2FoldChange")+
  ylim(0,5)+
  xlab("")+
  guides(fill = FALSE)+
  coord_flip()

# mean and median absolute change in gene expression
res_deseq_anott_changed %>% 
  dplyr::select(gene_oid, log2FoldChange, new_bin_name, Comparison, Design) %>% 
  dplyr::filter(Design == "~ Season + Site" 
                & Comparison == "Muskegon Lake\nVs\nM110") %>% 
  group_by(new_bin_name) %>% 
  summarise(mean_fold = mean(abs(log2FoldChange)),
            median_fold = median(abs(log2FoldChange)))

tmp_nutrient <- res_deseq_anott_changed %>% 
   dplyr::select(gene_oid, log2FoldChange, new_bin_name, Comparison, Design) %>% 
  dplyr::filter(Design == "~ Season + Site" 
                & Comparison == "Muskegon Lake\nVs\nM110")
  
kruskal.test(abs(log2FoldChange) ~ new_bin_name,
             data = tmp_nutrient)

pairwise.wilcox.test(abs(tmp_nutrient$log2FoldChange),
                     tmp_nutrient$new_bin_name,
                     p.adjust.method="BH")

# Add up and down regulation
res_deseq_anott_changed$regulation <- res_deseq_anott_changed$log2FoldChange > 0
res_deseq_anott_changed$regulation[res_deseq_anott_changed$regulation == TRUE] <- "Upregulated"
res_deseq_anott_changed$regulation[res_deseq_anott_changed$regulation == FALSE] <- "Downregulated"

p_deseq_overview2 <- res_deseq_anott_changed %>% 
  dplyr::select(gene_oid, log2FoldChange, new_bin_name, Comparison, Design) %>% 
  dplyr::filter(Design == "~ Site + Season" 
                & Comparison == "Fall-Spring") %>% 
  ggplot2::ggplot(aes(x = new_bin_name, y= log2FoldChange, fill = regulation))+
  geom_boxplot(fill = "#333333", alpha = 0.5, outlier.shape = NA, size = 1.25)+
  # geom_violin(fill = "#333333", alpha = 0.5, scale = "count")+
  theme_bw()+
  # scale_fill_brewer("", palette = "Paired")+
  theme(axis.text.x =  element_text(size = 14),
        axis.text.y = element_text(size = 14),
        legend.title = element_blank(),
        axis.title = element_text(size = 20),
        strip.text = element_text(size = 14))+
  ylab("log2FoldChange")+
  ylim(0,5)+
  xlab("")+
  guides(fill = FALSE)+
  coord_flip()
  # stat_summary(fun.data=mean_sdl, fun.args = list(mult = 1),
  #                geom="pointrange", color="#333333", size = 1.5)

# mean and median absolute change in gene expression
res_deseq_anott_changed %>% 
  dplyr::select(gene_oid, log2FoldChange, new_bin_name, Comparison, Design) %>% 
  dplyr::filter(Design == "~ Site + Season" 
                & Comparison == "Fall-Spring") %>% 
  group_by(new_bin_name) %>% 
  summarise(mean_fold = mean(abs(log2FoldChange)),
            median_fold = median(abs(log2FoldChange)))

tmp_season <- res_deseq_anott_changed %>% 
  dplyr::select(gene_oid, log2FoldChange, new_bin_name, Comparison, Design) %>% 
  dplyr::filter(Design == "~ Site + Season" 
                & Comparison == "Fall-Spring")
  
kruskal.test(abs(log2FoldChange) ~ new_bin_name,
             data = tmp_season)
pairwise.wilcox.test(abs(tmp_season$log2FoldChange),
                     tmp_season$new_bin_name,
                     p.adjust.method="BH")

print(p_deseq_overview2)

p_deseq_overview3 <- res_deseq_anott_changed %>% 
  dplyr::select(gene_oid, log2FoldChange, new_bin_name, Comparison, Design,
                regulation) %>% 
  dplyr::filter(Design == "~ Site + Season" 
                & Comparison == "Fall-Spring") %>% 
  ggplot2::ggplot(aes(x = new_bin_name, y= log2FoldChange, fill = regulation))+
  geom_boxplot( alpha = 0.5, outlier.shape = NA, size = 1.25)+
  # geom_violin(fill = "#333333", alpha = 0.5, scale = "count")+
  theme_bw()+
  # scale_fill_brewer("", palette = "Paired")+
  theme(axis.text.x =  element_text(size = 14),
        axis.text.y = element_text(size = 14),
        legend.title = element_blank(),
        axis.title = element_text(size = 20),
        strip.text = element_text(size = 14))+
  ylab("log2FoldChange")+
  ylim(-5,5)+
  xlab("")+
  guides(fill = FALSE)+
  coord_flip()
  # stat_summary(fun.data=mean_sdl, fun.args = list(mult = 1),
  #                geom="pointrange", color="#333333", size = 1.5)

print(p_deseq_overview3)

res_deseq_anott_changed %>% 
    dplyr::select(gene_oid, new_bin_name, Comparison, Design) %>% 
    dplyr::filter(Design == "~ Site + Season" 
                & Comparison == "Fall-Spring") %>% 
  distinct() %>% 
  group_by(new_bin_name) %>% summarise(sum_genes = n())


# Run statistics
test <- res_deseq_anott_changed %>% 
  dplyr::select(gene_oid, log2FoldChange, new_bin_name, Comparison, Design,
                regulation) %>% 
  dplyr::filter(Design == "~ Site + Season" 
                & Comparison == "Fall-Spring") 

pairwise.wilcox.test(test$log2FoldChange, g = test$new_bin_name,
                   p.adjust = "hochberg")


# Run statistics
test <- res_deseq_anott_changed %>%
  dplyr::select(gene_oid, log2FoldChange, new_bin_name, Comparison, Design,
                regulation) %>%
  dplyr::filter(Design == "~ Season + Site"
                & Comparison == "Muskegon Lake\nVs\nM110" &
                  new_bin_name == "MAG8.SU-M110-DCMD") %>% 
  distinct()

res_deseq_anott_changed %>% 
  dplyr::filter(Genome_ID == "2757320398")

```

## Run MAG8-DESeq

```{r MAG8-DESeq-1, dpi = 500, warning = FALSE, fig.width = 10, fig.height = 7}
# Select MAG8 genome and temperature gradient
expr_cov_MAG8 <- expr_cov_bins[levels(expr_cov_long$Genome_ID) == "2757320398"]
sel_MAG8 <- meta_metaT[, "Site"] == "110" & meta_metaT[, "Depth"] != "Mid" & meta_metaT[, "Season"] != "Spring" & meta_metaT[, "Time"] == "Night"
expr_cov_MAG8[[1]] <- expr_cov_MAG8[[1]][, sel_MAG8]
meta_metaT_MAG8 <- meta_metaT[sel_MAG8, ]


## Depth (temperature effect) effect and control for Seasonal variation
dds <- DESeq2::DESeqDataSetFromMatrix(countData = expr_cov_MAG8[[1]],
                              colData = meta_metaT_MAG8,
                              design= ~ Season + Depth)
# Run DESeq
dds <- DESeq2::DESeq(dds, quiet = TRUE)
  
# Calculate contrasts for all comparisons (deep vs surface logFC)
comp1 <- DESeq2::results(dds, 
                         contrast=c("Depth", "Deep","Surface"))[order(DESeq2::results(dds)$padj), ]
# comp2 <- DESeq2::results(dds, contrast=c("Time.Depth", "Night.Deep", "Night.Surface"))[order(results(dds)$padj), ]

# Store data in single dataframes
MAG8_deseq_results_depth_comp1 <- data.frame(gene_oid = comp1@rownames,
                    baseMean = comp1@listData$baseMean,
                    log2FoldChange = comp1@listData$log2FoldChange,
                    pvalue = comp1@listData$pvalue,
                    padj = comp1@listData$padj,
                    Genome_ID = "2757320398",
                    Comparison = "Day.Deep - Day.Surface",
                    Time_of_day = "Day"
)

# MAG8_deseq_results_depth_comp2 <- data.frame(gene_oid = comp2@rownames,
#                     baseMean = comp2@listData$baseMean,
#                     log2FoldChange = comp2@listData$log2FoldChange,
#                     pvalue = comp2@listData$pvalue,
#                     padj = comp2@listData$padj,
#                     Genome_ID = "2757320398",
#                     Comparison = "Night.Deep - Night.Surface",
#                     Time_of_day = "Night"
# )
# MAG8_deseq_results_depth <- rbind(MAG8_deseq_results_depth_comp1,
                                  # MAG8_deseq_results_depth_comp2)

MAG8_deseq_results_depth <- MAG8_deseq_results_depth_comp1

# Filter at p < 0.01
MAG8_deseq_results_depth <- MAG8_deseq_results_depth %>% dplyr::filter(padj < 0.01)

```

```{r MAG8-DESeq-1b, dpi = 500, warning = FALSE, fig.width = 10, fig.height = 7}
# Select MAG8 genome and MLB vs M110 at the surface
expr_cov_MAG8 <- expr_cov_bins[levels(expr_cov_long$Genome_ID) == "2757320398"]
sel_MAG8 <- meta_metaT[, "Depth"] == "Surface"
expr_cov_MAG8[[1]] <- expr_cov_MAG8[[1]][, sel_MAG8]
meta_metaT_MAG8 <- meta_metaT[sel_MAG8, ]


## Depth (temperature effect) effect and control for Seasonal variation
dds <- DESeq2::DESeqDataSetFromMatrix(countData = expr_cov_MAG8[[1]],
                              colData = meta_metaT_MAG8,
                              design= ~ Season + Site)
# Run DESeq
dds <- DESeq2::DESeq(dds, quiet = TRUE)
  
# Calculate contrasts for all comparisons (deep vs surface logFC)
comp1 <- DESeq2::results(dds, contrast=c("Site", "110", "Buoy"))[order(DESeq2::results(dds)$padj), ]
# comp2 <- DESeq2::results(dds, contrast=c("Time.Depth", "Night.Deep", "Night.Surface"))[order(results(dds)$padj), ]

# Store data in single dataframes
MAG8_deseq_results_nutrient_comp1 <- data.frame(gene_oid = comp1@rownames,
                    baseMean = comp1@listData$baseMean,
                    log2FoldChange = comp1@listData$log2FoldChange,
                    pvalue = comp1@listData$pvalue,
                    padj = comp1@listData$padj,
                    Genome_ID = "2757320398",
                    Comparison = "M110 - MLB"
)

MAG8_deseq_results_nutrient <- MAG8_deseq_results_nutrient_comp1

# Filter at p < 0.01
MAG8_deseq_results_nutrient <- MAG8_deseq_results_nutrient %>% dplyr::filter(padj < 0.01)

# Check share DE genes between MLB/M110 and bottom/surface M110
sum(MAG8_deseq_results_nutrient$gene_oid %in% MAG8_deseq_results_depth$gene_oid)
sum(!MAG8_deseq_results_nutrient$gene_oid %in% MAG8_deseq_results_depth$gene_oid)

sum(MAG8_deseq_results_depth$gene_oid %in% MAG8_deseq_results_nutrient$gene_oid)
sum(!MAG8_deseq_results_depth$gene_oid %in% MAG8_deseq_results_nutrient$gene_oid)

```


```{r MAG8-DESeq-2, dpi = 500, warning = FALSE, fig.width = 10, fig.height = 7, dev = c("png","pdf")}
# Merge with Kegg and COG annotations
# Import COG mapping file
cogid_2_cogcat <- read.csv("./Mapping_files/cogid_2_cogcat.csv", sep = ",", header = FALSE, fill = TRUE,col.names = c("COG_ID", "COG_class", "function"))[, 1:2]
cogid_2_cogcat <- cogid_2_cogcat[(cogid_2_cogcat$COG_class)!="", ]
cogid_2_cogcat <- droplevels(cogid_2_cogcat)

# Read COG category file
cog_categories <- read.table("./Mapping_files/cog_categories.tsv", header = TRUE, sep = "\t")

# Merge COG metadata
cog_meta <- dplyr::left_join(cog_categories, cogid_2_cogcat, by = c("COG_class" = "COG_class"))
cog_meta <- droplevels(cog_meta)

# Import KO hierarchical data
ko_path_df <- format_ko(path = "./Mapping_files/ko00000.keg")

# Import COG and KEGG annotation data of MAG8
MAG8_KO <- read.table("./IMG_annotation/IMG_2757320398/IMG Data/2757320398/2757320398.ko.tab.txt",
                              header = TRUE, fill = TRUE, sep = "\t" )
MAG8_KO$ko_id <- gsub("KO:", "", MAG8_KO$ko_id)

MAG8_COG <- read.table("./IMG_annotation/IMG_2757320398/IMG Data/2757320398/2757320398.cog.tab.txt",
                              stringsAsFactors = FALSE,  
                       sep = "\t", header = TRUE,
                       quote=NULL, comment='')

# add hierarchy to cog and KO annotations
MAG8_KO <- left_join(MAG8_KO, ko_path_df, by = "ko_id")
MAG8_COG <- left_join(MAG8_COG, cog_meta, by = c("cog_id" = "COG_ID"))

# Merge annotations and remove duplicate columns
MAG8_annot <- full_join(MAG8_KO[, c(1,10:11, 14:18)], 
                        MAG8_COG[, c(1,10:11, 13:15)], by = "gene_oid")
MAG8_annot$gene_oid <- as.character(MAG8_annot$gene_oid)

# Annotate differentially expressed genes
MAG8_deseq_results_depth_fin <- left_join(MAG8_deseq_results_depth, MAG8_annot,
                                      by = c("gene_oid") ) %>% distinct()
MAG8_deseq_results_nutrient_fin <- left_join(MAG8_deseq_results_nutrient, MAG8_annot,
                                      by = c("gene_oid") ) %>% distinct()
# Change NA to unknown
MAG8_deseq_results_depth_fin[,9:20] <- apply(MAG8_deseq_results_depth_fin[,9:20], 
                                             2, function(x) as.character(x))
MAG8_deseq_results_nutrient_fin[,8:19] <- apply(MAG8_deseq_results_nutrient_fin[,8:19], 
                                             2, function(x) as.character(x))
MAG8_deseq_results_depth_fin[is.na(MAG8_deseq_results_depth_fin)] <- "Unknown"
MAG8_deseq_results_nutrient_fin[is.na(MAG8_deseq_results_nutrient_fin)] <- "Unknown"

# Add label for up or downregulation
MAG8_deseq_results_depth_fin$regulation <- MAG8_deseq_results_depth_fin$log2FoldChange > 0
MAG8_deseq_results_depth_fin$regulation[MAG8_deseq_results_depth_fin$regulation == TRUE] <- "Upregulated"
MAG8_deseq_results_depth_fin$regulation[MAG8_deseq_results_depth_fin$regulation == FALSE] <- "Downregulated"
MAG8_deseq_results_depth_fin$regulation <- factor(MAG8_deseq_results_depth_fin$regulation,
                                                  levels =
                                         c("Upregulated", "Downregulated"))

MAG8_deseq_results_nutrient_fin$regulation <-
  MAG8_deseq_results_nutrient_fin$log2FoldChange > 0
MAG8_deseq_results_nutrient_fin$regulation[MAG8_deseq_results_nutrient_fin$regulation == TRUE] <- "Upregulated"
MAG8_deseq_results_nutrient_fin$regulation[MAG8_deseq_results_nutrient_fin$regulation == FALSE] <- "Downregulated"
MAG8_deseq_results_nutrient_fin$regulation <-
  factor(MAG8_deseq_results_nutrient_fin$regulation,
         levels =c("Upregulated", "Downregulated"))

```

```{r MAG8-DESeq-3, dpi = 500, warning = FALSE, fig.width = 13, fig.height = 7, dev = c("png","pdf")}
# Overview
## Log fold changes deep vs surface
p_mag8_deseq_ov <- MAG8_deseq_results_depth_fin %>% 
  dplyr::select(gene_oid:Time_of_day) %>% 
  distinct() %>% 
  ggplot(aes(x = gene_oid, y = log2FoldChange, fill = log2FoldChange))+
  geom_bar(size = 0.1, stat = "identity", color = "black")+
  # geom_boxplot(alpha = 0.5)+
 scale_fill_distiller(palette = "RdBu", name = "Fold Change",
                       limits = c(-15,15), oob=squish,
                      direction = -1) +
  theme_bw()+
  facet_grid(Time_of_day~.)+
  theme(axis.text.x = element_blank())+
  labs(x = "", y = "Fold Change (log2)")

print(p_mag8_deseq_ov)

# Focus on COG annotation and pool per level
p_mag8_deseq_cog <- MAG8_deseq_results_depth_fin %>% dplyr::select(gene_oid:Time_of_day, contains("cog")) %>% 
  distinct() %>% 
  ggplot(aes(x = COG_functional_category, y = log2FoldChange, fill = log2FoldChange))+
  geom_point(size = 3, color = "black", shape = 21)+
  # geom_boxplot(alpha = 0.5)+
 scale_fill_distiller(palette = "RdBu", name = "Fold Change",
                       limits = c(-15,15), oob=squish,
                      direction = -1) +
  theme_bw()+
  facet_grid(Time_of_day~.)+
  theme(axis.text.x = element_text(size = 12, angle = 90, hjust = 1))+
  labs(x = "", y = "Fold Change (log2)")

print(p_mag8_deseq_cog)

p_mag8_deseq_nutrient_cog <- MAG8_deseq_results_nutrient_fin %>% dplyr::select(gene_oid:Comparison, contains("cog")) %>% 
  distinct() %>% 
  ggplot(aes(x = COG_functional_category, y = log2FoldChange, fill = log2FoldChange))+
  geom_point(size = 3, color = "black", shape = 21)+
  # geom_boxplot(alpha = 0.5)+
 scale_fill_distiller(palette = "RdBu", name = "Fold Change",
                       limits = c(-15,15), oob=squish,
                      direction = -1) +
  theme_bw()+
  theme(axis.text.x = element_text(size = 12, angle = 90, hjust = 1))+
  labs(x = "", y = "Fold Change (log2)")

print(p_mag8_deseq_temp_cog)

# Focus on KO annotation and pool per level
p_mag8_deseq_KO <- MAG8_deseq_results_depth_fin %>% 
  dplyr::select(gene_oid:Time_of_day, contains("ko")) %>% 
  distinct() %>% 
  ggplot(aes(x = ko_level_B, y = log2FoldChange, fill = log2FoldChange))+
  geom_point(size = 3, color = "black", shape = 21)+
  # geom_boxplot(alpha = 0.5)+
 scale_fill_distiller(palette = "RdBu", name = "Fold Change",
                       limits = c(-15,15), oob=squish,
                      direction = -1) +
  theme_bw()+
  facet_grid(Time_of_day~.)+
  theme(axis.text.x = element_text(size = 12, angle = 90, hjust = 1))+
  labs(x = "", y = "Fold Change (log2)")

print(p_mag8_deseq_KO)
```

### Plot

```{r MAG8-DESeq-4, dpi = 500, warning = FALSE, fig.width = 4, fig.height = 5.5, dev = c("png","pdf")}
# Test for enrichment of functional categories in differentially expressed gene pool
## Test for enrichment of KO level C terms in transcriptome
bg_gsea <- MAG8_annot %>% 
  dplyr::select(ko_level_C, gene_oid) %>% 
  distinct()
bg_gsea[is.na(bg_gsea)] <- "Unknown"

metaT_gsea <- enricher(gene = unique(MAG8_deseq_results_depth_fin$gene_oid),
         universe = bg_gsea$gene_oid, 
         TERM2GENE = bg_gsea,
         pvalueCutoff = 0.05,
         qvalueCutoff = 0.2)
metaT_gsea_KO <- data.frame(metaT_gsea@result)
metaT_gsea_KO %>% 
  dplyr::select(Description, GeneRatio, BgRatio, p.adjust, qvalue, Count)%>%
  print()

## Test for enrichment of COG functional categories
bg_gsea <- MAG8_annot %>% 
  dplyr::select(COG_functional_category, gene_oid) %>% 
  distinct()
bg_gsea <- apply(bg_gsea, 2, function(x) as.character(x))
bg_gsea[is.na(bg_gsea)] <- "Unknown"
bg_gsea <- data.frame(bg_gsea)

metaT_gsea <- enricher(gene = unique(MAG8_deseq_results_depth_fin$gene_oid),
         universe = bg_gsea$gene_oid, 
         TERM2GENE = bg_gsea,
         pvalueCutoff = 0.05,
         qvalueCutoff = 0.2)

metaT_gsea_COG <- data.frame(metaT_gsea@result)
metaT_gsea_COG %>% 
  dplyr::select(Description, GeneRatio, BgRatio, p.adjust, qvalue, Count)%>%
  print()

# Focus on enriched KO level
p_mag8_deseq_depth_KO_gsea <- MAG8_deseq_results_depth_fin %>% 
  dplyr::select(regulation, gene_oid:Time_of_day, contains("ko")) %>% 
  distinct() %>% 
  dplyr::filter(ko_level_C %in% metaT_gsea_KO$Description) %>% 
  ggplot(aes(x = ko_level_C, y = log2FoldChange, fill = regulation))+
  geom_point(shape = 21, size = 3, position=position_jitterdodge(dodge.width=0.9,
                                                                 jitter.width=0.25)) +
  geom_boxplot(outlier.colour = NA, width = 0.5,
                        position = position_dodge(width=0.9), alpha = 0.3)+
  # geom_boxplot(alpha = 0.5)+
  scale_fill_manual(values = brewer.pal(11, "RdBu")[c(2,10)], name = "Fold Change\n") +
  theme_bw()+
  theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
        axis.text.y = element_text(size = 12),
        strip.text = element_text(size = 14),
        legend.text = element_text(size = 12),
        legend.title = element_blank(),
        legend.position = "top")+
  labs(x = "", y = "Fold Change (log2)")+
  ylim(-7, 7)

print(p_mag8_deseq_depth_KO_gsea)

# Focus on enriched KO level in sprin-summer samples too
p_mag8_deseq_temp_KO_gsea <- MAG8_deseq_results_nutrient_fin %>% 
  dplyr::select(regulation, gene_oid:Comparison, contains("ko")) %>% 
  distinct() %>% 
  dplyr::filter(ko_level_C %in% metaT_gsea_KO$Description) %>% 
  ggplot(aes(x = ko_level_C, y = log2FoldChange, fill = regulation))+
  geom_point(shape = 21, size = 3, position=position_jitterdodge(dodge.width=0.9,
                                                                 jitter.width=0.25)) +
  geom_boxplot(outlier.colour = NA, width = 0.5,
                        position = position_dodge(width=0.9), alpha = 0.3)+
  # geom_boxplot(alpha = 0.5)+
  scale_fill_manual(values = brewer.pal(11, "RdBu")[c(2,10)], name = "Fold Change\n") +
  theme_bw()+
  theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
        axis.text.y = element_text(size = 12),
        strip.text = element_text(size = 14),
        legend.text = element_text(size = 12),
        legend.title = element_blank(),
        legend.position = "top")+
  labs(x = "", y = "Fold Change (log2)")+
  ylim(-7, 7)

print(p_mag8_deseq_temp_KO_gsea)

```


```{r MAG8-DESeq-5, dpi = 500, warning = FALSE, fig.width = 5, fig.height = 7, dev = c("png","pdf")}
# M110 vs MLB station

# Test for enrichment of functional categories in differentially expressed gene pool
## Test for enrichment of KO level C terms in transcriptome
bg_gsea <- MAG8_annot %>% 
  dplyr::select(ko_level_C, gene_oid) %>% 
  distinct()
bg_gsea[is.na(bg_gsea)] <- "Unknown"

metaT_gsea <- enricher(gene = unique(MAG8_deseq_results_nutrient_fin$gene_oid),
         universe = bg_gsea$gene_oid, 
         TERM2GENE = bg_gsea,
         pvalueCutoff = 0.05,
         qvalueCutoff = 0.2)
metaT_gsea_KO <- data.frame(metaT_gsea@result)
metaT_gsea_KO %>% 
  dplyr::select(Description, GeneRatio, BgRatio, p.adjust, qvalue, Count)%>%
  print()

## Test for enrichment of COG functional categories
bg_gsea <- MAG8_annot %>% 
  dplyr::select(COG_functional_category, gene_oid) %>% 
  distinct()
bg_gsea <- apply(bg_gsea, 2, function(x) as.character(x))
bg_gsea[is.na(bg_gsea)] <- "Unknown"
bg_gsea <- data.frame(bg_gsea)

metaT_gsea <- enricher(gene = unique(MAG8_deseq_results_temp_fin$gene_oid),
         universe = bg_gsea$gene_oid, 
         TERM2GENE = bg_gsea,
         pvalueCutoff = 0.05,
         qvalueCutoff = 0.2)

metaT_gsea_COG <- data.frame(metaT_gsea@result)
metaT_gsea_COG %>% 
  dplyr::select(Description, GeneRatio, BgRatio, p.adjust, qvalue, Count)%>%
  print()

# # Focus on enriched KO level
# p_mag8_deseq_KO_gsea <- MAG8_deseq_results_temp_fin %>% 
#   dplyr::select(regulation, gene_oid:Comparison, contains("ko")) %>% 
#   distinct() %>% 
#   dplyr::filter(ko_level_C %in% metaT_gsea_KO$Description) %>% 
#   ggplot(aes(x = ko_level_C, y = log2FoldChange, fill = regulation))+
#   geom_point(shape = 21, size = 3, position=position_jitterdodge(dodge.width=0.9,
#                                                                  jitter.width=0.25)) +
#   geom_boxplot(outlier.colour = NA, width = 0.5,
#                         position = position_dodge(width=0.9), alpha = 0.3)+
#   # geom_boxplot(alpha = 0.5)+
#   scale_fill_manual(values = brewer.pal(11, "RdBu")[c(2,10)], name = "Fold Change\n") +
#   theme_bw()+
#   theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
#         axis.text.y = element_text(size = 12),
#         strip.text = element_text(size = 14),
#         legend.text = element_text(size = 12),
#         legend.title = element_blank(),
#         legend.position = "top")+
#   labs(x = "", y = "Fold Change (log2)")+
#   ylim(-7, 7)
# 
# print(p_mag8_deseq_KO_gsea)
```

# 7. Sequence discrete populations

**In order to check unspecific mapping sample reads were mapped consecutively to
every bin using `BBmap.sh`.** This approach allows us to check the mapping specificity by evaluating the distribution of read identity to the putative genome bin.  

Competitive mapping was performed through `blastn` searches against a 1M and 10M read subsample from the interleaved fasta generated after QC (`seqtk`). Shell script used to achieve this:
```
#!/bin/bash
set -e

for file in `cat map.list`
        do
        echo $file
		db=/scratch/vdenef_fluxm/rprops/DESMAN/metaG/vizbin_rebin_anvio_v230/SEQ_discrete/contigs/merged_bins-fixed.db
		seqtk sample -s 777 /scratch/vdenef_fluxm/rprops/DESMAN/metaG/data/${file}/dt_int.fasta 1000000 > /scratch/vdenef_fluxm/rprops/DESMAN/metaG/data/${file}/dt_int_subs_1000000.fasta
        qseqs=/scratch/vdenef_fluxm/rprops/DESMAN/metaG/data/${file}/dt_int_subs_1000000.fasta
		blastn -query ${qseqs} -task megablast -db ${db} -out ./blast_output/${file}_blast.tsv -outfmt 6 -max_target_seqs 1 -num_threads 40 -perc_identity 60
		rm ${qseqs}
done
```

<!-- # ```{r sequence discrete populations, dpi = 500, warning = FALSE, fig.width = 10, fig.height = 7} -->
<!-- # map_disc <- read.table("./SEQs_discrete/final.idhist", header = FALSE, -->
<!-- #                        row.names = NULL) -->
<!-- # colnames(map_disc) <- c("bin","sample", "identity", "reads_mapped", "bases_mapped") -->
<!-- #  -->
<!-- # # Add season metadata -->
<!-- # map_disc$season <- "Summer" -->
<!-- # map_disc$season[grep("Fa", map_disc$sample)] <- "Fall" -->
<!-- # map_disc$season[grep("Su", map_disc$sample)] <- "Summer" -->
<!-- # map_disc$season[grep("Sp", map_disc$sample)] <- "Spring" -->
<!-- # map_disc$sample <- gsub(".C","", map_disc$sample, fixed = TRUE) -->
<!-- # total_reads2 <- total_reads -->
<!-- # total_reads2$sample <- gsub("_", ".", fixed = TRUE,total_reads$sample) -->
<!-- # map_disc <- dplyr::left_join(map_disc, total_reads2, by = "sample") -->
<!-- #  -->
<!-- # # Throw away all %identity below 60% -->
<!-- # # map_disc <- map_disc %>% filter(identity > 60) -->
<!-- #  -->
<!-- # # Normalize reads_mapped to library sizes -->
<!-- # map_disc <- map_disc %>% group_by(sample) %>%  -->
<!-- #   mutate(rel_reads_mapped = 100*reads_mapped/Total_reads) -->
<!-- #  -->
<!-- # # Add observed genome size -->
<!-- # map_disc$bin <- gsub(".fa","",map_disc$bin) -->
<!-- # map_disc <- left_join(map_disc, bin_size, by = c("bin" = "bins")) -->
<!-- #  -->
<!-- # # Plot distributions -->
<!-- # for(bin2plot in unique(map_disc$bin)){ -->
<!-- #     p_sdisc <- map_disc %>% filter(bin == bin2plot) %>%  -->
<!-- #       ggplot(aes(x = identity, y = rel_reads_mapped, color = season))+ -->
<!-- #       theme_bw()+ -->
<!-- #       scale_color_brewer(palette = "Accent")+ -->
<!-- #       facet_wrap(~sample, nrow = 4)+ -->
<!-- #       geom_line(size = 1.5)+ -->
<!-- #       guides(color = FALSE)+ -->
<!-- #       ggtitle(bin2plot)+ -->
<!-- #       theme(axis.text=element_text(size=14), axis.title=element_text(size=20), -->
<!-- #         title=element_text(size=20), legend.text=element_text(size=14), -->
<!-- #         legend.background = element_rect(fill="transparent"), -->
<!-- #         axis.text.x = element_text(angle = 45, hjust = 1), -->
<!-- #         strip.text.y=element_text(size=14))+ -->
<!-- #       ylab("Proportion of reads mapped (%)")+ -->
<!-- #       xlab("Nucleotide identity (%)") -->
<!-- #       # ylim(0,.5) -->
<!-- #    -->
<!-- #   print(p_sdisc) -->
<!-- # } -->
<!-- # ``` -->
<!-- #  -->
<!-- # ```{r plot-cum-discrete, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 10} -->
<!-- # # Plot % reads corrected for genome size over threshold of 0.95 -->
<!-- # id_thresh <- 95 -->
<!-- # map_disc_cum <- map_disc  %>% filter(identity > id_thresh) %>% group_by(sample) %>%  -->
<!-- #   mutate(cum_rel_reads_mapped = cumsum(rel_reads_mapped))%>%  -->
<!-- #   filter(identity == 100) -->
<!-- # sum_cum <- map_disc_cum %>% group_by(sample, bin) %>% mutate(cum_bins_rel_reads_mapped = sum(cum_rel_reads_mapped)) -->
<!-- # colnames(sum_cum)[c(2)] <- "Sample2" -->
<!-- #  -->
<!-- # p_sdisc_cum1 <- ggplot(map_disc_cum, aes(x = sample, y = 1e6*cum_rel_reads_mapped/bin_size,  -->
<!-- #                                         fill = bin))+ -->
<!-- #   theme_bw()+ -->
<!-- #   scale_fill_brewer(palette = "Paired")+ -->
<!-- #   geom_point(size = 4, shape = 21, color = "black")+ -->
<!-- #   theme(axis.text=element_text(size=14), axis.title=element_text(size=20), -->
<!-- #       title=element_text(size=20), legend.text=element_text(size=14), -->
<!-- #       legend.background = element_rect(fill="transparent"), -->
<!-- #       axis.text.x = element_text(angle = 45, hjust = 1), -->
<!-- #       strip.text.y=element_text(size=14), legend.position = "bottom")+ -->
<!-- #   ylab(paste0("Proportion of reads mapped > ", id_thresh, "% NI"))+ -->
<!-- #   xlab("Sample")+ -->
<!-- #   guides(fill=guide_legend(nrow = 11))+ -->
<!-- #   # geom_point(data = sum_cum, aes(x = Sample2, y = cum_bins_rel_reads_mapped), -->
<!-- #              # shape = 22, fill = "black", size = 4)+ -->
<!-- #   ylim(0,2.5) -->
<!-- #  -->
<!-- # print(p_sdisc_cum1) -->
<!-- #  -->
<!-- # # Plot % reads over threshold of 0.99 -->
<!-- # id_thresh <- 99 -->
<!-- # map_disc_cum2 <- map_disc  %>% filter(identity > id_thresh) %>% group_by(sample) %>%  -->
<!-- #   mutate(cum_rel_reads_mapped = cumsum(rel_reads_mapped))%>%  -->
<!-- #   filter(identity == 100) -->
<!-- # sum_cum <- map_disc_cum2 %>% group_by(sample, bin) %>% mutate(cum_bins_rel_reads_mapped = sum(cum_rel_reads_mapped)) -->
<!-- # colnames(sum_cum)[c(2)] <- "Sample2" -->
<!-- #  -->
<!-- # p_sdisc_cum2 <- ggplot(map_disc_cum2, aes(x = sample, y = 1e6*cum_rel_reads_mapped/bin_size,  -->
<!-- #                                         fill = bin))+ -->
<!-- #   theme_bw()+ -->
<!-- #   scale_fill_brewer(palette = "Paired")+ -->
<!-- #   geom_point(size = 4, shape = 21, color = "black")+ -->
<!-- #   theme(axis.text=element_text(size=14), axis.title=element_text(size=20), -->
<!-- #       title=element_text(size=20), legend.text=element_text(size=14), -->
<!-- #       legend.background = element_rect(fill="transparent"), -->
<!-- #       axis.text.x = element_text(angle = 45, hjust = 1), -->
<!-- #       strip.text.y=element_text(size=14), legend.position = "bottom")+ -->
<!-- #   ylab(paste0("Proportion of reads mapped > ", id_thresh, "% NI"))+ -->
<!-- #   xlab("Sample")+ -->
<!-- #   guides(fill=guide_legend(nrow = 11))+ -->
<!-- #   # geom_point(data = sum_cum, aes(x = Sample2, y = cum_bins_rel_reads_mapped), -->
<!-- #              # shape = 22, fill = "black", size = 4)+ -->
<!-- #   ylim(0,1.0) -->
<!-- #  -->
<!-- # print(p_sdisc_cum2) -->
<!-- # ``` -->

### Competitive mapping using `blast` with 1M reads of interleaved fasta.  

```{r blast-approach, dpi = 500, warning = FALSE, fig.width = 10, fig.height = 7}
blast_df <- read.table("./SEQs_discrete/merged_blast_1M.tsv", header = FALSE)
colnames(blast_df) <- c("Sample", "Contig", "Identity")
blast_df_map <- read.table("./SEQs_discrete/merged_contig_list.tsv", header = FALSE)
colnames(blast_df_map) <- c("m_contig", "o_contig", "bin")

# Round identity to integer
# blast_df$Identity <- round(blast_df$Identity, 1)
# blast_df$Identity <- factor(blast_df$Identity)
blast_df <- left_join(blast_df, blast_df_map, by = c("Contig" = "m_contig"))

# Bin the %Identity in intervals of 0.5%
blast_df_sum <- transform(blast_df, bin_group=cut(Identity,  breaks=seq(0, 100, 0.5)))

# Format sample names and convert to data.frame
blast_df_sum$Sample <- gsub("_blast.tsv", "", blast_df_sum$Sample)
blast_df_sum <- data.frame(blast_df_sum)
# blast_df_sum$Identity <- as.numeric(as.character(blast_df_sum$Identity))
blast_df_sum$bin <- gsub(".fa","",blast_df_sum$bin)
blast_df_sum <- dplyr::left_join(blast_df_sum, bin_size, by = c("bin" = "bins"))

# Add extra column that converts the binning range to a numeric x-coordinate that
# is positioned in the middle of the binning interval
blast_df_sum$bin_group <- gsub("\\(|]", "", 
                               as.character(blast_df_sum$bin_group))
blast_df_sum$bin_xcoord <- as.numeric(do.call(rbind,
                                              strsplit(blast_df_sum$bin_group,
                                                       ","))[,1])+0.25

# Normalize mapped reads per sample based on sample reads
# blast_subs <- 1e6
# blast_df_sum <- blast_df_sum %>% ungroup %>% group_by(Sample) %>% 
#   mutate(n_prop = 100*n/blast_subs)

# Add season variable
blast_df_sum$season <- "Summer"
blast_df_sum$season[grep("Fa", blast_df_sum$Sample)] <- "Fall"
blast_df_sum$season[grep("Su", blast_df_sum$Sample)] <- "Summer"
blast_df_sum$season[grep("Sp", blast_df_sum$Sample)] <- "Spring"

# Reformat sample names
blast_df_sum$Sample <- gsub(".C", "", blast_df_sum$Sample, fixed = TRUE)
blast_df_sum$Sample <- gsub(".A", "", blast_df_sum$Sample, fixed = TRUE)
blast_df_sum$Sample <- gsub(".", "_", blast_df_sum$Sample, fixed = TRUE)

# Add metadata to dataframe
meta_blast <- meta[, -1] %>% distinct()
blast_df_sum <- dplyr::left_join(blast_df_sum, meta_blast, by = c("Sample" = "Sample_ID"))

# Reorder site factor
blast_df_sum$Site <- as.character(blast_df_sum$Site)
blast_df_sum$Site <- gsub("Buoy","Muskegon Lake", blast_df_sum$Site)
blast_df_sum$Site <- gsub("110","Lake Michigan\nsite M110", blast_df_sum$Site)
blast_df_sum$Site <- gsub("15","Lake Michigan\nsite M15", blast_df_sum$Site)
blast_df_sum$Site <- factor(blast_df_sum$Site, levels = c("Muskegon Lake",
                                                          "Lake Michigan\nsite M15",
                                                          "Lake Michigan\nsite M110"))
blast_df_sum$Depth <- as.character(blast_df_sum$Depth)
blast_df_sum$Depth <- factor(blast_df_sum$Depth, levels = c("Surface", "Mid", "Deep"))
blast_df_sum$season <- as.character(blast_df_sum$season)
blast_df_sum$season <- factor(blast_df_sum$season, levels = c("Spring", "Summer", "Fall"))

# remove non-Limnohabbitans bin
blast_df_sum <- blast_df_sum %>% dplyr::filter(bin != "B2_Fa13.BD.MLB.DN_rebin10")

# Add extra column with new bin names
new_bin_names <- read.table("./anvio_output/rebin/general_bins_summary_selected_final.tsv", header = TRUE)[, c(2,3)]
blast_df_sum <- left_join(blast_df_sum, new_bin_names, by = c("bin" = "bins"))
blast_df_sum$new_bin_name <- as.character(blast_df_sum$new_bin_name)
blast_df_sum$new_bin_name <- factor(blast_df_sum$new_bin_name, levels =
                                      c("MAG1.FA-MLB-DN","MAG2.FA-MLB-SN",
                                        "MAG3.FA-MLB-SN", "MAG4.FA-M110-DN",
                                        "MAG5.SP-M110-DD","MAG6.SP-M15-SD",
                                        "MAG7.SU-MLB-SD","MAG8.SU-M110-DCMD",
                                        "MAG9.SU-M15-SN","MAG10.SU-M15-SN"))
# plot for individual bins
for(bin2plot in unique(blast_df_sum$new_bin_name)){
  p_blast_sdisc <- blast_df_sum %>% dplyr::filter(new_bin_name == bin2plot) %>% 
     ggplot(aes(x = bin_xcoord, ..scaled.., fill = season))+
      theme_bw()+
      scale_fill_brewer(palette = "Accent")+
      facet_wrap(~Sample, nrow = 4)+
      geom_density(color = "black")+
      guides(color = FALSE)+
      ggtitle(bin2plot)+
      theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
        title=element_text(size=20), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        axis.text.x = element_text(angle = 45, hjust = 1),
        strip.text.y=element_text(size=14))+
      ylab("Density")+
      xlab("Nucleotide identity (%)")+
    xlim(75,100)
     
print(p_blast_sdisc)
}
```

```{r merged-blast-approach-1, dpi = 500, warning = FALSE, fig.width = 10, fig.height = 10, dev = c("png","pdf")}
# Plot one combined figure with proportions normalized for genome size
p_blast_sdisc_merged <- blast_df_sum %>% 
     ggplot(aes(x = bin_xcoord, ..density.., color = bin))+
      theme_gray()+
      scale_color_brewer("", palette = "Paired")+
      facet_wrap(~Sample, nrow = 4)+
      geom_density()+
      ggtitle("Merged bins per sample")+
      theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
        title=element_text(size=20), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        axis.text.x = element_text(angle = 45, hjust = 1),
        strip.text.y=element_text(size=14),
        legend.position = "bottom")+
      ylab("Reads per Mbp")+
      xlab("Nucleotide identity (%)")+
    xlim(75,100)+
  guides(color=guide_legend(ncol=3))
     
print(p_blast_sdisc_merged)
```

```{r merged-blast-approach-1b, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 5, dev = c("png","pdf")}
# Plot for most abundant bin (B63)
p_blast_sdisc_B63 <- blast_df_sum %>% 
  dplyr::filter(bin == "B63_Su13.BD.MM110.DCMD_rebin1") %>% 
  ggplot(aes(x = bin_xcoord, color = Sample))+
  theme_bw()+
  facet_grid(season~Site)+
  scale_color_manual(values = rep("#333333", 24))+
  geom_density(alpha = 0.4, size = 0.75, aes(fill = season), bw = "nrd0")+
  scale_shape_manual("", values = c(21,22,24))+
  scale_fill_brewer("", palette = "Accent")+
  guides(fill = FALSE)+
  theme(axis.text.y=element_text(size=14), axis.title=element_text(size=20),
        title=element_text(size=20), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        axis.text.x = element_text(size = 14),
        strip.text=element_text(size=16),
        panel.grid.minor = element_blank(),
        legend.position = "bottom")+
  ylab("Density")+
  xlab("Nucleotide identity (%)")+
  xlim(75,100)+
  guides(color = FALSE)

p_blast_sdisc_B63

blast_df_sum$new_bin_name <- factor(blast_df_sum$new_bin_name, levels =
                                      c("MAG5.SP-M110-DD","MAG2.FA-MLB-SN",
                                        "MAG3.FA-MLB-SN", "MAG4.FA-M110-DN",
                                        "MAG1.FA-MLB-DN", "MAG10.SU-M15-SN",
                                        "MAG6.SP-M15-SD", "MAG8.SU-M110-DCMD",
                                        "MAG7.SU-MLB-SD","MAG9.SU-M15-SN"))
# Plot for most abundant bin (B63)
p_blast_all_violin <- blast_df_sum %>% 
  ggplot(aes(x = new_bin_name, y= Identity, fill = new_bin_name))+
  theme_bw()+
  # facet_grid(season~Site)+
  # geom_density(alpha = 0.4, size = 0.75, color = "#333333")+
  geom_violin(alpha = 0.4, size = 0.75, color = "#333333",
              scale = "area", fill = "#333333")+
  scale_color_brewer("", palette = "Paired")+
  guides(fill = FALSE)+
  # facet_grid(new_bin_name~., scales = "free")+
  theme(axis.text.y=element_text(size=14), axis.title=element_text(size=20),
        title=element_text(size=20), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        axis.text.x = element_text(size = 14),
        strip.text=element_text(size=16),
        panel.grid.minor = element_blank(),
        legend.position = "bottom")+
  # scale_x_log10()+
  ylab("% Identity")+
  xlab("")+
  scale_y_continuous(limits = c(90,100))+
  guides(shape = FALSE)+
  # stat_summary(fun.data=mean_sdl, fun.args = list(mult = 1), 
                 # geom="pointrange", color="#333333", size = 1)+
  coord_flip()
  # xlim(90,100)

p_blast_all_violin

```

```{r merged-blast-approach-1bb, dpi = 500, warning = FALSE, fig.width = 11, fig.height = 12, dev = c("png","pdf")}
# Plot for all bins density plots
p_blast_all_dens <- blast_df_sum %>% 
  ggplot(aes(x = bin_xcoord, shape = Sample))+
  theme_bw()+
  # facet_grid(season~Site)+
  geom_density(alpha = 0.4, size = 0.4, color = "#333333",
               bw = "nrd0")+
  # geom_violin(alpha = 0.4, size = 0.75, color = "#333333",
  #             scale = "area", fill = "#333333")+
  scale_color_brewer("", palette = "Paired")+
  guides(fill = FALSE)+
  facet_wrap(~new_bin_name, ncol = 2, scales = "free")+
  theme(axis.text.y=element_text(size=14), axis.title=element_text(size=20),
        title=element_text(size=20), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        axis.text.x = element_text(size = 14),
        strip.text=element_text(size=14),
        panel.grid.minor = element_blank(),
        legend.position = "bottom")+
  # scale_x_log10()+
  ylab("% Identity")+
  xlab("")+
  scale_x_continuous(limits = c(90,100))+
  guides(shape = FALSE)
  # stat_summary(fun.data=mean_sdl, fun.args = list(mult = 1), 
                 # geom="pointrange", color="#333333", size = 1)+
  # coord_flip()
  # xlim(90,100)

p_blast_all_dens
```

```{r merged-blast-approach-2, dpi = 300, warning = FALSE, fig.width = 10, fig.height = 20}
# Plot for most Muskegon Lake for all bins
p_blast_sdisc_ML <- blast_df_sum %>% dplyr::filter(Site == "Muskegon Lake") %>% 
  ggplot(aes(x = bin_xcoord, ..scaled.., fill = new_bin_name, group = Sample,
             shape = Depth))+
  theme_bw()+
  facet_grid(new_bin_name~season, scales = "free_y")+
  geom_density(alpha = 0.3, color = "black", size = 1)+
  # geom_point(size = 3, alpha = 0.6)+
  scale_shape_manual("", values = c(21,22,24))+
  scale_fill_manual(values = fill_palette)+
  guides(color = FALSE, fill = FALSE)+
  # ggtitle(bin2plot)+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
        title=element_text(size=20), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        axis.text.x = element_text(angle = 45, hjust = 1),
        strip.text=element_text(size=14),
        panel.grid.minor = element_blank(),
        legend.position = "bottom")+
  ylab("Scaled density")+
  xlab("Nucleotide identity (%)")+
  xlim(75,100)

p_blast_sdisc_ML
```

```{r summary-blast-approach-1, dpi = 500, warning = FALSE, fig.width = 9, fig.height = 8}
blast_df_sum_comp <- blast_df_sum %>% group_by(Sample, bin) %>% dplyr::count(bin_xcoord)
blast_df_sum_comp <-  dplyr::left_join(blast_df_sum_comp, bin_size, by = c("bin" = "bins"))
blast_df_sum_comp <- left_join(blast_df_sum_comp, new_bin_names, by = c("bin" = "bins"))
blast_df_sum_comp$new_bin_name <- as.character(blast_df_sum_comp$new_bin_name)
blast_df_sum_comp$new_bin_name <- factor(blast_df_sum_comp$new_bin_name, levels =
                                      c("MAG1.FA-MLB-DN","MAG2.FA-MLB-SN",
                                        "MAG3.FA-MLB-SN", "MAG4.FA-M110-DN",
                                        "MAG5.SP-M110-DD","MAG6.SP-M15-SD",
                                        "MAG7.SU-MLB-SD","MAG8.SU-M110-DCMD",
                                        "MAG9.SU-M15-SN","MAG10.SU-M15-SN"))
blast_df_sum_comp <- dplyr::left_join(blast_df_sum_comp, meta_blast, by = c("Sample" = "Sample_ID"))

# Reorder site factor
blast_df_sum_comp$Site <- as.character(blast_df_sum_comp$Site)
blast_df_sum_comp$Site <- gsub("Buoy","Muskegon Lake", blast_df_sum_comp$Site)
blast_df_sum_comp$Site <- gsub("110","Lake Michigan\nsite M110", blast_df_sum_comp$Site)
blast_df_sum_comp$Site <- gsub("15","Lake Michigan\nsite M15", blast_df_sum_comp$Site)
blast_df_sum_comp$Site <- factor(blast_df_sum_comp$Site, 
                                 levels = c("Muskegon Lake",
                                            "Lake Michigan\nsite M15",
                                            "Lake Michigan\nsite M110"))
blast_df_sum_comp$Depth <- as.character(blast_df_sum_comp$Depth)
blast_df_sum_comp$Depth <- factor(blast_df_sum_comp$Depth, 
                                  levels = c("Surface", "Mid", "Deep"))
blast_df_sum_comp$Season <- as.character(blast_df_sum_comp$Season)
blast_df_sum_comp$Season <- factor(blast_df_sum_comp$Season, levels = c("Spring", "Summer", "Fall"))

# Normalize mapped reads per sample based on genome size
blast_df_sum_comp <- blast_df_sum_comp %>% group_by(bin) %>%
mutate(n_norm = 1e6*n/bin_size)

blast_df_sum_comp <- left_join(blast_df_sum_comp, total_reads, by = c("Sample" = "sample"))

# Divide normalized reads by 1M (fixed blast census)
blast_df_sum_comp <- blast_df_sum_comp %>% mutate(n_norm_perc = 100*n_norm/1e6)
```

```{r summary-blast-approach-2, dpi = 500, warning = FALSE, fig.width = 9, fig.height = 8}
# Plot % reads corrected for genome size over threshold of 0.95
id_thresh <- 95-0.25
map_disc_cum <- blast_df_sum_comp  %>% dplyr::filter(bin_xcoord > id_thresh) %>% group_by(Sample, bin) %>% 
  mutate(cum_rel_reads_mapped = cumsum(n_norm_perc))%>% 
  dplyr::filter(bin_xcoord == 100-0.25)
sum_cum <- map_disc_cum %>% group_by(Sample, bin) %>% mutate(cum_bins_rel_reads_mapped = sum(cum_rel_reads_mapped))

p_sdisc_cum3 <- ggplot(map_disc_cum, aes(x = new_bin_name, 
                                         y = cum_rel_reads_mapped, 
                                        fill = new_bin_name))+
  theme_bw()+
  scale_fill_manual("", values = fill_palette)+
  geom_jitter(size = 4, shape = 21, color = "black", alpha = 0.7, width = 0.15)+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
      title=element_text(size=20), legend.text=element_text(size=12),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_blank(),
      strip.text=element_text(size=14), legend.position = "bottom",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylab(paste0("Norm. relative abundance ( > ", id_thresh, "% NI)"))+
  xlab("")+
  guides(fill=guide_legend(nrow = 3))+
  facet_grid(Season~Site, scales ="free")+
  scale_y_continuous(labels=scaleFUN, limits = c(0,3))+
  coord_trans(y = "sqrt")

print(p_sdisc_cum3)

# Plot % reads over threshold of 0.99
id_thresh <- 99-0.25
map_disc_cum <- blast_df_sum_comp  %>% dplyr::filter(bin_xcoord > id_thresh) %>% group_by(Sample, bin) %>% 
  mutate(cum_rel_reads_mapped = cumsum(n_norm_perc))%>% 
  dplyr::filter(bin_xcoord == 100-0.25)
sum_cum <- map_disc_cum %>% group_by(Sample, bin) %>% mutate(cum_bins_rel_reads_mapped = sum(cum_rel_reads_mapped))

p_sdisc_cum4 <- ggplot(map_disc_cum, aes(x = new_bin_name, 
                                         y = cum_rel_reads_mapped, 
                                        fill = new_bin_name))+
  theme_bw()+
  scale_fill_manual("", values = fill_palette)+
  geom_jitter(size = 4, shape = 21, color = "black", alpha = 0.7, width = 0.15)+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
      title=element_text(size=20), legend.text=element_text(size=12),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_blank(),
      strip.text=element_text(size=14), legend.position = "bottom",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylab(paste0("Norm. relative abundance ( > ", id_thresh, "% NI)"))+
  xlab("")+
  guides(fill=guide_legend(nrow = 3))+
  facet_grid(Season~Site, scales ="free")+
  scale_y_continuous(labels=scaleFUN, limits = c(0,1.25))+
  coord_trans(y = "sqrt")

print(p_sdisc_cum4)
```

<!-- ### Filter out bins using MClust   -->

<!-- ** Here we will use model-based clustering to determine whether there is a MAG present or not in a certain sample. **  -->

<!-- ```{r filter-bins-1, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 6} -->
<!-- # Perform gaussian mixture clustering on whole dataset. -->
<!-- mod1 <- densityMclust(blast_df_sum$bin_xcoord) -->
<!-- summary(mod1) -->
<!-- par(mfrow=c(2,2)) -->
<!-- plot(mod1, what = "density", data = blast_df_sum$bin_xcoord, breaks = 15) -->
<!-- plot(mod1, what = "BIC") -->
<!-- plot(mod1, what = "diagnostic", type = "cdf") -->
<!-- plot(mod1, what = "diagnostic", type = "qq") -->
<!-- par(mfrow=c(1,1)) -->

<!-- # Return cluster labels and plot the Sequence discrete populations again. -->
<!-- blast_df_sum$cluster_label <- mod1$classification -->
<!-- blast_df_sum$model_density <- mod1$density -->

<!-- # Add cluster labels to blast_df_sum_comp -->
<!-- tmp <- blast_df_sum[, c("bin_xcoord", "cluster_label", "model_density")] %>% distinct() -->
<!-- tmp$bin_xcoord <- as.factor(tmp$bin_xcoord) -->
<!-- blast_df_sum_comp$bin_xcoord <- as.factor(blast_df_sum_comp$bin_xcoord) -->
<!-- blast_df_sum_comp <- left_join(blast_df_sum_comp, tmp, by = "bin_xcoord") -->
<!-- blast_df_sum_comp$bin_xcoord <-  -->
<!--   as.numeric(as.character(blast_df_sum_comp$bin_xcoord)) -->
<!-- blast_df_sum_comp$cluster_label <- as.factor(blast_df_sum_comp$cluster_label) -->

<!-- # Plot for global analysis -->
<!-- p_blast_sdisc_glob_clust <- blast_df_sum_comp %>%  -->
<!--   ggplot(aes(x = bin_xcoord, y = model_density,  -->
<!--              color = cluster_label, group = Sample))+ -->
<!--   theme_bw()+ -->
<!--   geom_line(alpha = 1, size = 1)+ -->
<!--   scale_color_brewer(palette = "Dark2")+ -->
<!--   guides(color = FALSE, fill = FALSE)+ -->
<!--   theme(axis.text=element_text(size=14), axis.title=element_text(size=20), -->
<!--         title=element_text(size=20), legend.text=element_text(size=14), -->
<!--         legend.background = element_rect(fill="transparent"), -->
<!--         axis.text.x = element_text(angle = 45, hjust = 1), -->
<!--         strip.text=element_text(size=16), -->
<!--         panel.grid.minor = element_blank(), -->
<!--         legend.position = "bottom")+ -->
<!--   ylab("Density")+ -->
<!--   xlab("Nucleotide identity (%)")+ -->
<!--   xlim(75,100) -->

<!-- print(p_blast_sdisc_glob_clust) -->

<!-- ``` -->

<!-- ```{r filter-bins-2, dpi = 500, warning = FALSE, fig.width = 10, fig.height = 20} -->
<!-- # Plot for most abundant bin (B63) -->
<!-- p_blast_sdisc_B63_clust <- blast_df_sum_comp %>% dplyr::filter(bin == "B63_Su13.BD.MM110.DCMD_rebin1") %>%  -->
<!--   ggplot(aes(x = bin_xcoord, y = n, color = cluster_label, group = Sample))+ -->
<!--   theme_bw()+ -->
<!--   facet_grid(Season~Site)+ -->
<!--   geom_line(alpha = 1, size = 1)+ -->
<!--   scale_color_brewer(palette = "Dark2")+ -->
<!--   guides(color = FALSE, fill = FALSE)+ -->
<!--   theme(axis.text=element_text(size=14), axis.title=element_text(size=20), -->
<!--         title=element_text(size=20), legend.text=element_text(size=14), -->
<!--         legend.background = element_rect(fill="transparent"), -->
<!--         axis.text.x = element_text(angle = 45, hjust = 1), -->
<!--         strip.text=element_text(size=16), -->
<!--         panel.grid.minor = element_blank(), -->
<!--         legend.position = "bottom")+ -->
<!--   ylab("Density")+ -->
<!--   xlab("Nucleotide identity (%)")+ -->
<!--   xlim(75,100) -->

<!-- print(p_blast_sdisc_B63_clust) -->
<!-- ``` -->

### Compare blast to bwa results  

```{r comparison-bwa-1, dpi = 500, warning = FALSE, fig.width = 18, fig.height = 8}
# Compare blast inferred abundances with bwa inferred abundances
grid_arrange_shared_legend(p_abs2, p_sdisc_cum4, ncol = 2)
```


```{r comparison-bwa-2, dpi = 500, warning = FALSE, fig.width = 9, fig.height = 8}
# Merge those two dataframes and make scatter plot highlighting the differences 
df_map_merged_sm <- df_map_merged[, c("Bin","Sample","rel_norm_abundance", "Season", "Site")]
df_map_merged_sm$Sample <- gsub(".", "_", df_map_merged_sm$Sample, fixed = TRUE)
df_map_merged_sm <- data.frame(df_map_merged_sm, 
                               interaction = interaction(df_map_merged_sm$Bin,
                                                                         df_map_merged_sm$Sample),
                              method = "bwa")

map_disc_cum_sm <- map_disc_cum[, c("bin","Sample","cum_rel_reads_mapped")] 
map_disc_cum_sm <- data.frame(map_disc_cum_sm, 
                              interaction = interaction(map_disc_cum_sm$bin,
                                                                         map_disc_cum_sm$Sample),
                              method = "blastn")

abs_merged <- left_join(df_map_merged_sm, map_disc_cum_sm, by = "interaction")
abs_merged <- abs_merged[,c("Bin","Sample.x","rel_norm_abundance","cum_rel_reads_mapped",
                            "Season", "Site")]
colnames(abs_merged) <- c("Bin","Sample",
                          "bwa_norm_abundance","blastn_norm_abundance",
                          "Season", "Site")



# Plot
p_scat_abund <- ggplot(abs_merged, aes(x = bwa_norm_abundance, y = blastn_norm_abundance))+
  theme_bw()+
  geom_point(size = 4, shape = 21, color = "black", alpha = 0.7,
             aes(fill = Bin))+
  scale_fill_manual("", values = fill_palette)+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
      title=element_text(size=20), legend.text=element_text(size=12),
      legend.background = element_rect(fill="transparent"),
      strip.text=element_text(size=14), legend.position = "bottom",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylab(paste0("Norm. relative abundance (blastn, > ", id_thresh, "% NI)"))+
  xlab("Norm. relative abundance (BWA)")+
  guides(fill=guide_legend(nrow = 3))+
  facet_grid(Season~Site, scales ="free")+
  scale_y_continuous(labels=scaleFUN)+
  scale_x_continuous(labels=scaleFUN)+
  geom_smooth(method = "lm", color = "black")
  # coord_trans(x = "atanh", y = "atanh")

print(p_scat_abund)
```

<!-- ### Calculate temporospatial diversity in Lhab -->

<!-- ```{r diversity-Lhab, dpi = 500, warning = FALSE, fig.width = 9, fig.height = 5} -->
<!-- abs_merged_div <- map_disc_cum[, c("Sample", "new_bin_name", "n_norm")] -->
<!-- abs_merged_div$n_norm <- as.integer(abs_merged_div$n_norm ) -->

<!-- # Format long to wide format -->
<!-- abs_table <- spread(abs_merged_div, Sample, n_norm) -->
<!-- abs_table <- data.frame(abs_table) -->
<!-- rownames(abs_table) <- abs_table[, 1]; abs_table <- abs_table[, -1] -->
<!-- tax.table <- data.frame(MAGs = rownames(abs_table)) -->
<!-- rownames(tax.table) <- tax.table$MAGs -->

<!-- # Make phyloseq object -->
<!-- MAG_phy <- phyloseq(otu_table(abs_table, taxa_are_rows = TRUE),  -->
<!--          tax_table(as.matrix(tax.table))) -->

<!-- # Run Diversity_16S() -->
<!-- MAG_div <- Diversity_16S(MAG_phy, ncore = 3, parallel = TRUE, -->
<!--                          R = 100, brea = FALSE) -->

<!-- MAG_div <- data.frame(Sample = rownames(MAG_div), MAG_div[,-c(3:6)]) -->

<!-- # Merge with metadata -->
<!-- MAG_div <- left_join(MAG_div, meta_blast, by = c("Sample" = "Sample_ID")) -->

<!-- # Annotate and order metavariables -->
<!-- MAG_div$Site <- as.character(MAG_div$Site) -->
<!-- MAG_div$Site <- gsub("Buoy","Muskegon Lake", MAG_div$Site) -->
<!-- MAG_div$Site <- gsub("110","Lake Michigan\nsite M110", MAG_div$Site) -->
<!-- MAG_div$Site <- gsub("15","Lake Michigan\nsite M15", MAG_div$Site) -->
<!-- MAG_div$Site <- factor(MAG_div$Site, levels = c("Muskegon Lake", -->
<!--                                                           "Lake Michigan\nsite M15", -->
<!--                                                           "Lake Michigan\nsite M110")) -->
<!-- MAG_div$Depth <- as.character(MAG_div$Depth) -->
<!-- MAG_div$Depth <- factor(MAG_div$Depth, levels = c("Surface", "Mid", "Deep")) -->
<!-- MAG_div$Season <- as.character(MAG_div$Season) -->
<!-- MAG_div$Season <- factor(MAG_div$Season, levels = c("Spring", "Summer", "Fall")) -->

<!-- # Plot results -->
<!-- p_MAG_div <- ggplot(MAG_div, aes(x = Season, y = D2,  -->
<!--                                         fill = Season, shape = Depth))+ -->
<!--   theme_bw()+ -->
<!--   scale_fill_brewer(palette = "Accent")+ -->
<!--   geom_point(size = 4, color = "black", alpha = 0.7)+ -->
<!--   scale_shape_manual(values = c(21,24,23))+ -->
<!--   # geom_boxplot(alpha = 0.4)+ -->
<!--   theme(axis.text=element_text(size=14), axis.title=element_text(size=20), -->
<!--       title=element_text(size=20), legend.text=element_text(size=12), -->
<!--       legend.background = element_rect(fill="transparent"), -->
<!--       axis.text.x = element_text(size = 14, angle = 45, hjust = 1), -->
<!--       strip.text=element_text(size=14), legend.position = "bottom", -->
<!--       strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+ -->
<!--   ylab(paste0("Limnohabitans population\n diversity (D2)"))+ -->
<!--   guides(fill=FALSE)+ -->
<!--   facet_grid(~Site, scales ="free")+ -->
<!--   xlab("")+ -->
<!--   geom_errorbar(aes(ymin = D2 - sd.D2, ymax = D2 + sd.D2), width = 0.05)+ -->
<!--   scale_y_continuous(labels=scaleFUN, limits = c(0,6.5)) -->
<!--   # coord_trans(y = "sqrt") -->

<!-- print(p_MAG_div) -->
<!-- ``` -->

# 8. DESMAN

## QC

### PMD

```{r desman-qc, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 10, dev = c("png","pdf")}
# For adding manuscript MAG information etc
MGT_df$Genome_ID <- as.character(MGT_df$Genome_ID)
MGT_df$MAG_id <- do.call(rbind, strsplit(MGT_df$Genome_ID,".", fixed = TRUE))[, 1]

# import metadata
metadata_desman <- readxl::read_excel("./DESMAN/desman_results.xlsx", sheet = 1)
metadata_desman <- left_join(metadata_desman, MGT_df, by = c("MAG_id"))
  
# import desman diagnostics
for(i in 1:10){
  tmp_desm <- read.csv(paste0("./DESMAN/", metadata_desman$MAG_id[i], "/",
    metadata_desman$MAG_id[i], "_Dev.csv"))
  # wide to long
  tmp_desm$Genome_ID <- metadata_desman$Genome_ID[i]
  if(i == 1) {
    results_desm_qc <- tmp_desm
    } else results_desm_qc <- rbind(results_desm_qc, tmp_desm)
}

ord_list_bin_ext <- c(
  "MAG5.SP-M110-DD", "MAG2.FA-MLB-SN",
  "MAG3.FA-MLB-SN", "MAG4.FA-M110-DN",
  "MAG1.FA-MLB-DN", "MAG10.SU-M15-SN",
  "MAG6.SP-M15-SD","MAG8.SU-M110-DCMD",
  "MAG7.SU-MLB-SD","MAG9.SU-M15-SN"
)

p_desman_qc <- results_desm_qc %>% 
  mutate(Genome_ID = factor(Genome_ID, levels = ord_list_bin_ext)) %>% 
  dplyr::filter(!Genome_ID %in% c("MAG5.SP-M110-DD", "MAG2.FA-MLB-SN")) %>% 
  droplevels() %>% 
  ggplot(., aes(x = G, y = Dev, fill = Genome_ID))+
  geom_point(col = "black", alpha = 0.1, size = 4, shape = 21, fill = "black")+
  # scale_fill_brewer("", palette = "Paired")+
  # geom_boxplot()+
 # geom_boxplot(notch=FALSE, outlier.shape=NA, alpha=0.8)+
  geom_line()+
  theme_bw()+
  facet_wrap(.~Genome_ID, ncol = 4)+
  scale_x_continuous(limits = c(1,12), breaks = 1:12)+
    theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12),
      strip.text=element_text(size=12), legend.position = "bottom")+
  scale_y_log10()+
  labs(x = "Number of Strains", y = "Posterior Mean Deviance")+
  geom_smooth(col = "black")+
  guides(fill = FALSE)+
  geom_vline(xintercept = 2, linetype = 2, col = "black")+
  geom_vline(xintercept = 3, linetype = 2, col = "black")+
  geom_vline(xintercept = 6, linetype = 2, col = "black")

png("./Figures/DESMAN/DESMAN_QC.png", width = 12.5, height = 6, res = 300, units = "in")
print(p_desman_qc)
dev.off()

svg("./Figures/DESMAN/DESMAN_QC.svg", width = 12.5, height = 6)
print(p_desman_qc)
dev.off()
```

### PMD

```{r desman-qc-2, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 10, dev = c("png","pdf")}
# For adding manuscript MAG information etc
MGT_df$Genome_ID <- as.character(MGT_df$Genome_ID)
MGT_df$MAG_id <- do.call(rbind, strsplit(MGT_df$Genome_ID,".", fixed = TRUE))[, 1]

# import metadata
metadata_desman <- readxl::read_excel("./DESMAN/desman_results.xlsx", sheet = 1)
metadata_desman <- left_join(metadata_desman, MGT_df, by = c("MAG_id"))
  
# import desman diagnostics
for(i in 1:10){
  tmp_desm <- read.csv(paste0("./DESMAN/", metadata_desman$MAG_id[i], "/",
    metadata_desman$MAG_id[i], "_Dev.csv"))
  # wide to long
  tmp_desm$Genome_ID <- metadata_desman$Genome_ID[i]
  if(i == 1) {
    results_desm_qc <- tmp_desm
    } else results_desm_qc <- rbind(results_desm_qc, tmp_desm)
}

ord_list_bin_ext <- c(
  "MAG5.SP-M110-DD", "MAG2.FA-MLB-SN",
  "MAG3.FA-MLB-SN", "MAG4.FA-M110-DN",
  "MAG1.FA-MLB-DN", "MAG10.SU-M15-SN",
  "MAG6.SP-M15-SD","MAG8.SU-M110-DCMD",
  "MAG7.SU-MLB-SD","MAG9.SU-M15-SN"
)

p_desman_qc <- results_desm_qc %>% 
  mutate(Genome_ID = factor(Genome_ID, levels = ord_list_bin_ext)) %>% 
  dplyr::filter(!Genome_ID %in% c("MAG5.SP-M110-DD", "MAG2.FA-MLB-SN")) %>% 
  droplevels() %>% 
  ggplot(., aes(x = G, y = Dev, fill = Genome_ID))+
  geom_point(col = "black", alpha = 0.1, size = 4, shape = 21, fill = "black")+
  # scale_fill_brewer("", palette = "Paired")+
  # geom_boxplot()+
 # geom_boxplot(notch=FALSE, outlier.shape=NA, alpha=0.8)+
  geom_line()+
  theme_bw()+
  facet_wrap(.~Genome_ID, ncol = 4)+
  scale_x_continuous(limits = c(1,12), breaks = 1:12)+
    theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12),
      strip.text=element_text(size=12), legend.position = "bottom")+
  scale_y_log10()+
  labs(x = "Number of Strains", y = "Posterior Mean Deviance")+
  geom_smooth(col = "black")+
  guides(fill = FALSE)+
  geom_vline(xintercept = 2, linetype = 2, col = "black")+
  geom_vline(xintercept = 3, linetype = 2, col = "black")+
  geom_vline(xintercept = 6, linetype = 2, col = "black")

png("./Figures/DESMAN/DESMAN_QC.png", width = 12.5, height = 6, res = 300, units = "in")
print(p_desman_qc)
dev.off()

svg("./Figures/DESMAN/DESMAN_QC.svg", width = 12.5, height = 6)
print(p_desman_qc)
dev.off()
```

## Strain abundance of MAGs

```{r desman-1, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 10, dev = c("png","pdf")}

# plot overview of DESMAN results
p_desman_overview <- ggplot(metadata_desman, aes(x = Genome_ID, y = number_of_robust_haplotypes,
  fill = Lineage, group = Genome_ID))+
  theme_bw()+
  geom_bar(alpha = 1, stat = "identity", color = "black",
           position = position_dodge(width = 1), width = 0.7)+
  scale_fill_manual("",
                    values = c("#deebf7ff", "#c6dbefff","#9ecae1ff",
                               "#6baed6ff"))+
  theme(axis.text=element_text(size=12, face = "bold"), axis.title=element_text(size=16),
        title=element_text(size=14), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        # axis.text.x = element_text(angle = 65, hjust = 1),
        strip.text.x=element_text(size = 16),
        legend.position="bottom",
        axis.text.x=element_text(size = 12, angle =45, hjust= 1),
        axis.title.x=element_blank(),
        plot.title = element_text(hjust = 0, size=18),
            plot.margin = unit(c(1.1,1.1,1.1,1.1), "cm"))+
  guides(fill=FALSE)+
  ylab("")+
  xlab("")+
  ggtitle("Number of resolved haplotypes")

print(p_desman_overview)

png("./Figures/DESMAN/desman_overview.png", height = 4.5, 
  width = 5, units = "in", res = 500)
print(p_desman_overview)
dev.off()

svg("./Figures/DESMAN/desman_overview.svg", height = 4.5, 
  width = 5)
print(p_desman_overview)
dev.off()

# DESMAN error on best runs
p_desman_overview_error <- ggplot(metadata_desman, aes(x = Genome_ID, y = error, fill = Lineage, group = Genome_ID))+
  theme_bw()+
  geom_bar(alpha = 1, stat = "identity", color = "black",
           position = position_dodge(width = 1), width = 0.7)+
  scale_fill_manual("",
                    values = c("#deebf7ff", "#c6dbefff","#9ecae1ff",
                               "#6baed6ff"))+
  theme(axis.text=element_text(size=12, face = "bold"), axis.title=element_text(size=20),
        title=element_text(size=20), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        # axis.text.x = element_text(angle = 65, hjust = 1),
        strip.text.x=element_text(size = 18),
        legend.position="bottom",
        axis.text.x=element_text(size = 13, angle =45, hjust= 1),
        axis.title.x=element_blank(),
        plot.title = element_text(hjust = 0, size=18),
            plot.margin = unit(c(1.1,1.1,1.1,1.1), "cm"))+
  guides(fill=FALSE)+
  ylab("Error on best run")+
  xlab("")+
  ylim(0,0.1)

png("./Figures/DESMAN/desman_overview_error.png", height = 5, 
  width = 6, units = "in", res = 500)
print(p_desman_overview_error)
dev.off()

# Match abundance data with number of detected haplotypes
p_desm_Topt <- metadata_desman %>% 
  ggplot(., aes(x = Topt, y = number_of_robust_haplotypes, fill = Lineage, group = Genome_ID))+
  theme_bw()+
  geom_point(alpha = 1, size = 4, shape = 21)+
  scale_fill_manual("",
                    values = c("#deebf7ff", "#c6dbefff","#9ecae1ff",
                               "#6baed6ff"))+
  theme(axis.text=element_text(size=12, face = "bold"), axis.title=element_text(size=16),
        title=element_text(size=14), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        # axis.text.x = element_text(angle = 65, hjust = 1),
        strip.text.x=element_text(size = 16),
        legend.position="bottom",
        axis.text.x=element_text(size = 12),
        plot.title = element_text(hjust = 0, size=18),
            plot.margin = unit(c(1.1,1.1,1.1,1.1), "cm"))+
  guides(fill=FALSE)+
  ylab("Number of strains")+
  xlab("Temperature (°C)")+
  ggtitle("Number of resolved haplotypes")

p_desm_abund <- data_total_abund %>% 
  ungroup() %>% 
  dplyr::select(norm_mean_rel_abundance, new_bin_name) %>% 
  left_join(., metadata_desman, by = c("new_bin_name" = "Genome_ID")) %>% 
  # dplyr::filter(!MAG_id %in% c("MAG5", "MAG2")) %>% 
  ggplot(., aes(x = number_of_robust_haplotypes, y = norm_mean_rel_abundance, 
    fill = Lineage, group = MAG_id))+
  theme_bw()+
  geom_boxplot(alpha = 1)+
  scale_fill_manual("",
                    values = c("#deebf7ff", "#c6dbefff","#9ecae1ff",
                               "#6baed6ff"))+
  theme(axis.text=element_text(size=12, face = "bold"), axis.title=element_text(size=16),
        title=element_text(size=14), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        # axis.text.x = element_text(angle = 65, hjust = 1),
        strip.text.x=element_text(size = 16),
        legend.position="top",
        axis.text.x=element_text(size = 12),
        plot.title = element_text(hjust = 0, size=18),
            plot.margin = unit(c(1.1,1.1,1.1,1.1), "cm"))+
  scale_y_log10()+
  xlab("Number of strains")+
  ylab("Normalized abundance")+
  ggtitle("")
  
png("./Figures/DESMAN/desman_overview_abund.png", height = 5, 
  width = 6, units = "in", res = 500)
print(p_desm_abund)
dev.off()
```

### Plots

```{r desman-2, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 10, dev = c("png","pdf")}
# import desman haplotype frequency data
for(i in 1:10){
  tmp_desm <- read.csv(paste0("./DESMAN/", metadata_desman$MAG_id[i], "/",
    metadata_desman$best_desman_run[i],"/Gamma_starR.csv"))
  # wide to long
  strain_ids <- data.frame(desman_id = as.numeric(gsub("X","",colnames(tmp_desm)[-1])), 
  my_id = paste0("Strain ", 1:(ncol(tmp_desm)-1)),
  MAG = metadata_desman$MAG_id[i])
  colnames(tmp_desm) <- c("Samples", paste0("Strain ", 1:(ncol(tmp_desm)-1)))
  temp_desm <- gather(tmp_desm, Strain, Frequency, 2:ncol(tmp_desm), factor_key = TRUE)
  temp_desm$MAG <- metadata_desman$MAG_id[i]

  if(i == 1) {
    results_desm_all <- temp_desm
    strain_ids_desman <- strain_ids
    } else {
    results_desm_all <- rbind(results_desm_all, temp_desm)
    strain_ids_desman <- rbind(strain_ids_desman, strain_ids)}
   
}
results_desm_all$Samples <- gsub(".MAG*..","", results_desm_all$Samples)
results_desm_all$Samples <- gsub(".A", "", results_desm_all$Samples, fixed = TRUE)
results_desm_all$Samples <- gsub(".C", "", results_desm_all$Samples, fixed = TRUE)

# Merge DESMAN data with environmental metadata
results_desm_all <- dplyr::left_join(results_desm_all, meta_em, by = c("Samples" = "Sample_ID"))
results_desm_all$Site <- as.character(results_desm_all$Site)
results_desm_all$Site <- gsub("110", "Lake Michigan\nsite M110", results_desm_all$Site)
results_desm_all$Site <- gsub("15", "Lake Michigan\nsite M15", results_desm_all$Site)
results_desm_all$Site <- gsub("Buoy", "Muskegon Lake", results_desm_all$Site)
results_desm_all$Site <- factor(results_desm_all$Site, levels = c("Muskegon Lake",
                                                            "Lake Michigan\nsite M15",
                                                            "Lake Michigan\nsite M110"))
results_desm_all$Season <- as.character(results_desm_all$Season)
results_desm_all$Season <- factor(results_desm_all$Season, levels = c("Spring", "Summer","Fall"))
# Factor to numeric
results_desm_all$Temperature..C. <- as.numeric(as.character(results_desm_all$Temperature..C.))
results_desm_all$PAR <- as.numeric(as.character(results_desm_all$PAR))
results_desm_all$Chl.Lab..ug.L. <- as.numeric(as.character(results_desm_all$Chl.Lab..ug.L.))
results_desm_all$DO.Probe..mg.L. <- as.numeric(as.character(results_desm_all$DO.Probe..mg.L.))
results_desm_all$TP.ug.L <- as.numeric(as.character(results_desm_all$TP.ug.L))
results_desm_all$DOC.mg.L <- as.numeric(as.character(results_desm_all$DOC.mg.L))

# Add phylogenetic information of MAG/strains
results_desm_all <- left_join(results_desm_all, MGT_df[, c("MAG_id", "Lineage", "Genome_ID")], by = c("MAG" = "MAG_id"))

# Plot MAG abundances across environment/seasons
for(i in 1:10){
p_desm <- results_desm_all %>% 
    dplyr::filter(MAG %in% paste0("MAG",i)) %>% 
    ggplot(., aes(x = Season, y = Frequency, fill = Strain, col = Strain))+
  geom_point(color = "black", alpha = 0.7, size = 3, shape = 21)+
  # geom_boxplot(alpha = 0.3, col = "black", outlier.shape = NA, size = 0.3)+
  scale_fill_brewer("", palette = "Accent")+
  scale_color_brewer("",palette = "Accent")+
  theme_bw()+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
      strip.text=element_text(size=12), legend.position = "bottom",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylab(paste0("Strain frequency"))+
  guides(fill=FALSE)+
  facet_grid(Site~Strain, scales ="free")+
  xlab("")+
  scale_y_continuous(labels=scaleFUN, limits = c(0,1))+
  guides(fill=FALSE)+
  stat_summary(fun.y = mean, fun.ymin = mean, fun.ymax = mean,
                 geom = "crossbar", width = 0.5, col = "black")

ncol <- results_desm_all %>% 
    dplyr::filter(MAG %in% paste0("MAG",i)) %>% 
  summarize(number_of_strain= length(unique(Strain)))

# export png
png(paste0("./Figures/DESMAN/desman-MAG", i, ".png"), height = 6, 
  width = 3 + 4*(ncol$number_of_strain/6), units = "in", res = 500)
print(p_desm)
dev.off()

# export svg
svg(paste0("./Figures/DESMAN/desman-MAG", i, ".svg"), height = 6, 
  width = 3 + 4*(ncol$number_of_strain/6))
print(p_desm)
dev.off()
}
```

### Correlation network

```{r, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 10, dev = c("png","pdf")}
# format data for sparcc
data_total_abund_sparcc <- data_total_abund
data_total_abund_sparcc$Sample_ID <- gsub("_", ".", data_total_abund$Sample_ID)

strain_cor <- results_desm_all %>% 
  dplyr::filter(!MAG %in% c("MAG2", "MAG5")) %>% 
  left_join(., data_total_abund_sparcc, by = c("Genome_ID" = "new_bin_name", "Samples" = "Sample_ID")) %>% 
  mutate(Strain_MAG = paste0(Strain,"-", MAG)) %>% 
  dplyr::select(Samples, Strain_MAG, norm_mean_rel_abundance, Frequency) %>% 
  mutate(norm_strain_abundance = norm_mean_rel_abundance*Frequency) %>% 
  distinct() %>% 
  dplyr::select(Samples, Strain_MAG, norm_strain_abundance) %>% 
  spread(., Strain_MAG, norm_strain_abundance) %>%
  dplyr::select(-Samples) %>% 
  cor(use = "complete.obs")

# run sparcc
res_sparcc <- sparcc(strain_cor, iter = 100, inner_iter = 100, th = 0.2)
colnames(res_sparcc$Cor) <- colnames(strain_cor)
rownames(res_sparcc$Cor) <- rownames(strain_cor)
sparcc.graph <- abs(res_sparcc$Cor) >= 0.3
diag(sparcc.graph) <- 0
sparcc.graph <- Matrix::Matrix(sparcc.graph, sparse=TRUE)
ig.sparcc <- adj2igraph(sparcc.graph, rmEmptyNodes = TRUE, vertex.attr = list(name = colnames(res_sparcc$Cor)))
MAG_ids <- data.frame(MAG = factor(do.call(rbind, strsplit(V(ig.sparcc)$name, split = "-"))[, 2]))
V(ig.sparcc)$color <- brewer.pal(n = 9, "Set1")[MAG_ids$MAG]

igraph::degree(ig.sparcc)

# Visualize
am.coord <- layout.fruchterman.reingold(ig.sparcc)

# Cluster into modules
modulos <- cluster_spinglass(delete.vertices(igraph::simplify(ig.sparcc), degree(ig.sparcc)==0))

svg(paste0("./Figures/DESMAN/desman-strain-network.svg"), height = 6,
  width = 6)
l<-layout_with_fr(ig.sparcc)
plot.igraph(delete.vertices(igraph::simplify(ig.sparcc), degree(ig.sparcc)==0),  
  main="",
  edge.color = "black",
  edge.width=2,
  edge.curved=0.3,
     layout=l,
  vertex.label.color = "black",
  vertex.label.font=1,
   mark.groups= modulos, mark.border=NA,
  mark.col = brewer.pal(n = 3, "Accent")[2:1])
dev.off()

png(paste0("./Figures/DESMAN/desman-strain-network.png"), height = 6,
  width = 6, res = 500, units = "in")
plot.igraph(delete.vertices(igraph::simplify(ig.sparcc), degree(ig.sparcc)==0),  
  main="",
  edge.color = "black",
  edge.width=2,
  edge.curved=0.3,
     layout=l,
  vertex.label.color = "black",
  vertex.label.font=1,
     mark.groups= modulos, mark.border=NA,
  mark.col = brewer.pal(n = 3, "Accent")[2:1])
dev.off()


png(paste0("./Figures/DESMAN/desman-strain-network-degree.png"), height = 4,
  width = 6, res = 500, units = "in")
cbind(MAG_ids, degree = (degree(ig.sparcc, mode="all")), membership = factor(modulos$membership)) %>% 
  ggplot(., aes(x = MAG, y = degree, fill = membership))+
  geom_point(size = 4, shape = 21, position = position_dodge(0.75))+
  geom_boxplot(alpha = 0.5)+
  theme_bw()+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
      strip.text=element_text(size=12), legend.position = "bottom",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
    scale_fill_manual("module", values = brewer.pal(n = 3, "Accent")[1:2])
dev.off()

```

## Correlations

### Check correlation with environmental variables 

```{r desman-2abc, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 4, dev = c("png","pdf")}
# Just depth profile for M110 station
desm_p2b <- results_desm_all %>% dplyr::filter(Site == "Lake Michigan\nsite M110"& 
                                                  Depth != "Mid" & MAG == "MAG8") %>%
  mutate(Depth = factor(Depth, levels = c("Surface", "Deep"))) %>% 
  ggplot(., aes(x = Season, y = Frequency, 
                                         fill = Depth, 
                                         col = Depth))+
  geom_point(color = "black", alpha = 0.7, size = 4, shape = 21)+
  scale_fill_manual("", values = brewer.pal(n = 3, "Set1"))+
  theme_bw()+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=14),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
      strip.text=element_text(size=12), legend.position = "top",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylab(paste0("Strain frequency"))+
  facet_grid(.~Strain, scales ="free")+
  xlab("")+
  scale_y_continuous(labels=scaleFUN, limits = c(0,1))

png(paste0("./Figures/DESMAN/desman-MAG8-M110.png"), height = 5,  
  width = 10, units = "in", res = 500)
print(desm_p2b)
dev.off()

```

```{r desman-2ab, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 6, dev = c("png","pdf")}
for(i in 1:10){
  results_desm_all_sb <- results_desm_all %>% 
     dplyr::filter(MAG %in% paste0("MAG",i)) %>% 
    dplyr::select(Frequency,`Temperature..C.`, Strain, `SpCond..µS.cm.`, DOC.mg.L, PAR, TP.ug.L, Season) %>% 
    tidyr::spread(., Strain, Frequency) %>% 
    dplyr::select(-Season)
  results_desm_all_sb <- apply(results_desm_all_sb, 2, function(x) as.numeric(as.character(x)))
  p.mat <- data.frame(cor(results_desm_all_sb, method = "pearson", use="pairwise"))
  p.mat <- data.frame(cor_variable = rownames(p.mat), p.mat)
  p.mat <- p.mat %>% dplyr::select(-`Temperature..C.`, -`SpCond..µS.cm.`, -`DOC.mg.L`, -PAR, -`TP.ug.L`) 
  p.mat <- p.mat %>% 
    gather(., Strain, Correlation, 2:ncol(p.mat))%>% 
    dplyr::filter(abs(Correlation) < 1)
  p.mat <- data.frame(p.mat, MAG =  paste0("MAG",i))
  if(i == 1) p.res <- p.mat else p.res <- rbind(p.res, p.mat)
}

p_cor_desman <- p.res %>% 
  dplyr::filter(cor_variable %in% c("Temperature..C.", "SpCond..µS.cm.", "DOC.mg.L", "PAR", "TP.ug.L")) %>% 
  dplyr::mutate(cor_variable = factor(cor_variable, levels = c("PAR", "SpCond..µS.cm.", "TP.ug.L", "DOC.mg.L", "Temperature..C."))) %>% 
  dplyr::filter(!MAG %in% c("MAG5", "MAG2")) %>% 
  # dplyr::filter(abs(Correlation) > 0.5) %>% 
  ggplot(., aes(y = MAG, x = Strain, z = Correlation, fill = Correlation, size = abs(Correlation))) +
  facet_grid(.~cor_variable)+
  ggplot2::geom_count(color = "black", shape = 21, stat = "identity")+
  scale_size("Pearson's\ncorrelation", range = c(3.5,8), breaks=seq(1,10,by=3)) +
  # geom_tile(aes(fill=Correlation), col = "black")+
  ggplot2::scale_fill_distiller("Pearson's\ncorrelation",palette="RdBu", na.value="white", 
    limits = c(-1,1)) + 
  theme_bw()+
  ylab("")+
  xlab("")+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
      strip.text=element_text(size=12), legend.position = "top",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))

png(paste0("./Figures/DESMAN/desman-strain-correlations-1.png"), height = 6, 
  width = 12, units = "in", res = 500)
print(p_cor_desman)
dev.off()

p_cor_desman_2 <- p.res %>% 
  dplyr::filter(cor_variable %in% c("PAR","Temperature..C.", "TP.ug.L")) %>% 
  dplyr::mutate(cor_variable = factor(cor_variable, levels = c("PAR", "Temperature..C.", "TP.ug.L"))) %>% 
  dplyr::filter(!MAG %in% c("MAG5", "MAG2")) %>% 
  dplyr::mutate(MAG = factor(MAG, levels = c("MAG3", "MAG4", "MAG1", "MAG10", "MAG6", "MAG8", "MAG7", "MAG9"))) %>% 
  # dplyr::filter(abs(Correlation) > 0.5) %>% 
  ggplot(., aes(y = cor_variable, x = Strain, z = Correlation, fill = Correlation, size = abs(Correlation))) +
  facet_wrap(.~MAG, ncol = 4)+
  ggplot2::geom_count(color = "black", shape = 21, stat = "identity")+
  scale_size("Pearson's\ncorrelation", range = c(3.5,8), breaks=seq(1,10,by=3)) +
  # geom_tile(aes(fill=Correlation), col = "black")+
  ggplot2::scale_fill_distiller("Pearson's\ncorrelation",palette="RdBu", na.value="white", 
    limits = c(-1,1)) + 
  theme_bw()+
  ylab("")+
  xlab("")+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
      strip.text=element_text(size=12), legend.position = "top",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))

png(paste0("./Figures/DESMAN/desman-strain-correlations-2.png"), height = 5, 
  width = 10, units = "in", res = 500)
print(p_cor_desman_2)
dev.off()

svg(paste0("./Figures/DESMAN/desman-strain-correlations-2.svg"), height = 5, 
  width = 10)
print(p_cor_desman_2)
dev.off()
```

### Strain diversity analysis

```{r desman-alpha-div, dpi = 500, warning = FALSE, fig.width = 8, fig.height = 4, dev = c("png","pdf")}
df_div_desman <- results_desm_all %>% 
  dplyr::filter(MAG == "MAG8") %>% 
  dplyr::select(Samples:Frequency) %>% 
  spread(., Strain, Frequency)
alpha_div_desman <- data.frame(Sample = df_div_desman[,1], 
      D2 = vegan::diversity(df_div_desman[,2:ncol(df_div_desman)], index = "invsimpson")
)

alpha_div_desman <- left_join(alpha_div_desman, meta_em,
                              by = c("Sample" = "Sample_ID"))
alpha_div_desman$Site <- as.character(alpha_div_desman$Site)
alpha_div_desman$Site <- gsub("110", "Lake Michigan - M110", alpha_div_desman$Site)
alpha_div_desman$Site <- gsub("15", "Lake Michigan - M15", alpha_div_desman$Site)
alpha_div_desman$Site <- gsub("Buoy", "Muskegon Lake", alpha_div_desman$Site)
alpha_div_desman$Site <- factor(alpha_div_desman$Site, levels = c("Muskegon Lake",
                                                            "Lake Michigan - M15",
                                                            "Lake Michigan - M110"))
alpha_div_desman$Season <- as.character(alpha_div_desman$Season)
alpha_div_desman$Season <- factor(alpha_div_desman$Season, levels = c("Spring", "Summer","Fall"))

p_alpha_desman <- alpha_div_desman %>% 
  ggplot(., aes(x = Season, y = D2, fill = Season, shape = Depth))+
  geom_point(size = 4) +
  scale_shape_manual("", values = c(21,22,24))+
  facet_grid(.~Site)+
  theme_bw()+
  scale_fill_brewer(palette = "Accent")+
  labs(y = expression("MAG8 strain diversity (D"[2]*")"), x = "")+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
        title=element_text(size=16), legend.text=element_text(size=16),
        strip.text = element_text(size = 16))+
  ylim(0,5)+
  guides(fill  = guide_legend(title = "", override.aes = list(size = 4, shape = 21),
                         nrow = 4)
   )

png(paste0("./Figures/DESMAN/desman-alpha-div-MAG8.png"), height = 4,
  width = 10, res = 500, units = "in")
print(p_alpha_desman)
dev.off()
```



```{r desman-3, dpi = 500, warning = FALSE, fig.width = 8, fig.height = 5, dev = c("png","pdf")}
# Make PCoA
variants <- df_div_desman[,2:ncol(df_div_desman)]
rownames(variants) <- df_div_desman[, 1]
taxa <- as.matrix(data.frame(variant_name = colnames(df_div_desman)[2:ncol(df_div_desman)]))
rownames(taxa) <- colnames(df_div_desman)[2:ncol(df_div_desman)]

# Store as phyloseq object
physeq_desm <- phyloseq(otu_table(variants, taxa_are_rows = FALSE),
         tax_table(taxa)
)

# Run beta diversity analysis on 16s data
pcoa <- ordinate(
  physeq = physeq_desm, 
  method = "PCoA", 
  distance = "bray",
  correction = "lingoes",
  k=2
)
pcoa.df <- data.frame(Samples = sample_names(physeq_desm), pcoa$vectors)
var <- round(pcoa$values$Eigenvalues/sum(pcoa$values$Eigenvalues)*100,1)

# Add metadata
pcoa.df$Samples <- gsub(".A", "", pcoa.df$Samples, fixed = TRUE)
pcoa.df$Samples <- gsub(".C", "", pcoa.df$Samples, fixed = TRUE)
pcoa.df <- dplyr::left_join(pcoa.df, meta_em, by = c("Samples" = "Sample_ID"))

pcoa.df$Site <- as.character(pcoa.df$Site)
pcoa.df$Site <- gsub("110", "Lake Michigan - M110", pcoa.df$Site)
pcoa.df$Site <- gsub("15", "Lake Michigan - M15", pcoa.df$Site)
pcoa.df$Site <- gsub("Buoy", "Muskegon Lake", pcoa.df$Site)
pcoa.df$Site <- factor(pcoa.df$Site, levels = c("Muskegon Lake",
                                                            "Lake Michigan - M15",
                                                            "Lake Michigan - M110"))
pcoa.df$Season <- as.character(pcoa.df$Season)
pcoa.df$Season <- factor(pcoa.df$Season, levels = c("Spring", "Summer","Fall"))

# Run permanova
x <- df_div_desman[,2:ncol(df_div_desman)]
rownames(x) <- df_div_desman[, 1]
dist_desm <- vegan::vegdist(x)
perm <- permute::how(nperm = 999)
permanova_desm <- vegan::adonis(dist_desm ~ Season * Site, 
                        data = meta_em, 
                        permutations = perm)

# Plot PCoA
my_grob = grobTree(textGrob(bquote(paste(r[Season]^2 == 
    .(round(100 * permanova_desm$aov.tab[1, 5], 1)), 
    "%")), x = 0.7, y = 0.95, hjust = 0, gp = gpar(col = "black", 
    fontsize = 14, fontface = "italic")))
my_grob2 = grobTree(textGrob(bquote(paste(r[Site]^2 == 
    .(format(round(100 * permanova_desm$aov.tab[2, 5], 
        1), nsmall = 1)), "%")), x = 0.7, y = 0.87, 
    hjust = 0, gp = gpar(col = "black", fontsize = 14, 
        fontface = "italic")))
my_grob3 = grobTree(textGrob(bquote(paste(r[Season:Site]^2 == 
    .(round(100 * permanova_desm$aov.tab[3, 5], 1)), 
    "%")), x = 0.7, y = 0.79, hjust = 0, gp = gpar(col = "black", 
    fontsize = 14, fontface = "italic")))

pcoa.ord <- ggplot(data=pcoa.df, aes(x=Axis.1, y=Axis.2, shape = Site))+
  scale_shape_manual(values=c(21,23, 24))+
  geom_point(size=7, alpha=0.7, aes(fill = Season))+
  theme_bw()+
  scale_fill_brewer(palette = "Accent")+
  labs(x = paste0("PCoA axis 1 (",var[1], "%)"),
       y = paste0("PCoA axis 2 (",var[2], "%)"), fill="", shape = "", colour="")+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=16),
        title=element_text(size=16), legend.text=element_text(size=16))+ 
  guides(fill = guide_legend(override.aes = list(shape = 22)))+
  annotation_custom(my_grob)+
  annotation_custom(my_grob2)+
  annotation_custom(my_grob3)+
  ggtitle("PCoA of resolved MAG8 variant composition")

png(paste0("./Figures/DESMAN/desman-beta-div-MAG8.png"), height = 6,
  width = 10, res = 500, units = "in")
print(pcoa.ord)
dev.off()
```

Same analysis but now for all strains

```{r desman-alpha-div-all, dpi = 500, warning = FALSE, fig.width = 8, fig.height = 4, dev = c("png","pdf")}
df_div_desman_all <- results_desm_all %>% 
  dplyr::filter(!MAG %in% c("MAG2", "MAG5")) %>% 
  left_join(., data_total_abund_sparcc, by = c("Genome_ID" = "new_bin_name", "Samples" = "Sample_ID")) %>% 
  mutate(Strain_MAG = paste0(Strain,"-", MAG)) %>% 
  dplyr::select(Samples, Strain_MAG, norm_mean_rel_abundance, Frequency) %>% 
  mutate(norm_strain_abundance = norm_mean_rel_abundance*Frequency) %>% 
  distinct() %>% 
  dplyr::select(Samples, Strain_MAG, norm_strain_abundance) %>% 
  spread(., key = Strain_MAG, value = norm_strain_abundance) %>% 
  dplyr::filter(!Samples %in% c("Su13.BD.MM110.DN", "Sp13.BD.MM15.DD"))
df_div_desman_all[is.na(df_div_desman_all)] <- 0
  
# Make PCoA
variants <- df_div_desman_all[,2:ncol(df_div_desman_all)]
rownames(variants) <- df_div_desman_all[, 1]
taxa <- as.matrix(data.frame(variant_name = colnames(df_div_desman_all)[2:ncol(df_div_desman_all)]))
rownames(taxa) <- colnames(df_div_desman_all)[2:ncol(df_div_desman_all)]

# Store as phyloseq object
physeq_desm <- phyloseq(otu_table(variants, taxa_are_rows = FALSE),
         tax_table(taxa)
)

# Run beta diversity analysis on 16s data
pcoa <- ordinate(
  physeq = physeq_desm, 
  method = "PCoA", 
  distance = "bray",
  correction = "lingoes",
  k=2
)
pcoa.df <- data.frame(Samples = sample_names(physeq_desm), pcoa$vectors)
var <- round(pcoa$values$Eigenvalues/sum(pcoa$values$Eigenvalues)*100,1)

# Add metadata
pcoa.df$Samples <- gsub(".A", "", pcoa.df$Samples, fixed = TRUE)
pcoa.df$Samples <- gsub(".C", "", pcoa.df$Samples, fixed = TRUE)
pcoa.df <- dplyr::left_join(pcoa.df, meta_em, by = c("Samples" = "Sample_ID"))

pcoa.df$Site <- as.character(pcoa.df$Site)
pcoa.df$Site <- gsub("110", "Lake Michigan - M110", pcoa.df$Site)
pcoa.df$Site <- gsub("15", "Lake Michigan - M15", pcoa.df$Site)
pcoa.df$Site <- gsub("Buoy", "Muskegon Lake", pcoa.df$Site)
pcoa.df$Site <- factor(pcoa.df$Site, levels = c("Muskegon Lake",
                                                            "Lake Michigan - M15",
                                                            "Lake Michigan - M110"))
pcoa.df$Season <- as.character(pcoa.df$Season)
pcoa.df$Season <- factor(pcoa.df$Season, levels = c("Spring", "Summer","Fall"))

# Run permanova
x <- df_div_desman[,2:ncol(df_div_desman)]
rownames(x) <- df_div_desman[, 1]
dist_desm <- vegan::vegdist(x)
perm <- permute::how(nperm = 999)
permanova_desm <- vegan::adonis(dist_desm ~ Season * Site, 
                        data = meta_em, 
                        permutations = perm)

# Plot PCoA
my_grob = grobTree(textGrob(bquote(paste(r[Season]^2 == 
    .(round(100 * permanova_desm$aov.tab[1, 5], 1)), 
    "%")), x = 0.7, y = 0.95, hjust = 0, gp = gpar(col = "black", 
    fontsize = 14, fontface = "italic")))
my_grob2 = grobTree(textGrob(bquote(paste(r[Site]^2 == 
    .(format(round(100 * permanova_desm$aov.tab[2, 5], 
        1), nsmall = 1)), "%")), x = 0.7, y = 0.87, 
    hjust = 0, gp = gpar(col = "black", fontsize = 14, 
        fontface = "italic")))
my_grob3 = grobTree(textGrob(bquote(paste(r[Season:Site]^2 == 
    .(round(100 * permanova_desm$aov.tab[3, 5], 1)), 
    "%")), x = 0.7, y = 0.79, hjust = 0, gp = gpar(col = "black", 
    fontsize = 14, fontface = "italic")))

pcoa.ord <- ggplot(data=pcoa.df, aes(x=Axis.1, y=Axis.2, shape = Site))+
  scale_shape_manual(values=c(21,23, 24))+
  geom_point(size=7, alpha=0.7, aes(fill = Season))+
  theme_bw()+
  scale_fill_brewer(palette = "Accent")+
  labs(x = paste0("PCoA axis 1 (",var[1], "%)"),
       y = paste0("PCoA axis 2 (",var[2], "%)"), fill="", shape = "", colour="")+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=16),
        title=element_text(size=16), legend.text=element_text(size=16))+ 
  guides(fill = guide_legend(override.aes = list(shape = 22)))+
  annotation_custom(my_grob)+
  annotation_custom(my_grob2)+
  annotation_custom(my_grob3)+
  ggtitle("PCoA of resolved MAG variant compositions")

```

### Gene assignment based on DESMAN inference

```{r desman-4, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 10, dev = c("png","pdf")}
# Import etaS.csv which contains gene assignment
for(i in 1:10){
    # Select only robust number of strains
  strain_id <- strain_ids_desman %>% 
    dplyr::filter(MAG == paste0("MAG",i))
  
  tmp_desm <- read.csv(paste0("./DESMAN/Gene_assignments/MAG",
    i,"_etaS_df.csv")) %>% 
    dplyr::select(c(1,strain_id$desman_id+2))
  
  # wide to long
  colnames(tmp_desm) <- c("Samples", paste0("Strain ", 1:(ncol(tmp_desm)-1)))
  temp_desm <- gather(tmp_desm, Strain, Presence, 2:ncol(tmp_desm), factor_key = TRUE)
  
  # Total genome size of strains
  temp_desm_tg <- temp_desm %>% 
  group_by(Strain) %>% 
  summarize(genome_size = sum(Presence))
  
  # Core genome size across strains
  core_genome_size = sum(apply(tmp_desm[,-1], 1, function(x) sum(x) == length(x)))
  
  # Store in data frame
  temp_desm_tg$core_genome_size <- core_genome_size
  temp_desm_tg$MAG <- metadata_desman$MAG_id[i]
  temp_desm$MAG <- metadata_desman$MAG_id[i]
  
  # Return list with both the summary statistics and the detailed contig_assignments
  if(i == 1) {
    results_desm_ga_all <- list()
    results_desm_ga_all[[1]] <- temp_desm_tg
    results_desm_ga_all[[2]] <- temp_desm
  } else {
    results_desm_ga_all[[1]] <- rbind(results_desm_ga_all[[1]], temp_desm_tg)
    results_desm_ga_all[[2]] <- rbind(results_desm_ga_all[[2]], temp_desm)
  }
}

# # Check for presence of core gene presence in strains
# core_cogs_lim <- unique(read.csv("./DESMAN/MAG8_scg_10_9/Filtered_Tau_star.csv")[,1])
# 
# # Check predicted presence of core genes
# geneAssign_df %>% dplyr::filter(Gene %in% core_cogs_lim)
# 
# # Wide to long format
# geneAssign_df <- tidyr::gather(geneAssign_df, Variant, Presence,
#                              `Strain 1`:`Strain 5`, factor_key=TRUE)

# Strain list based on network analysis that is either in lake MI/muskegon lake
strain_pref <- readxl::read_excel("./DESMAN/desman_strain_preference.xlsx") %>% 
  mutate(MAG_strain = paste0(MAG,"-",Strain)) %>% 
  dplyr::select(MAG_strain, Preference)

p_genome_size_strains <- results_desm_ga_all[[1]] %>% 
  dplyr::filter(!MAG %in% c("MAG5", "MAG2")) %>% 
  mutate(MAG = factor(MAG, levels = c(paste0("MAG",c(3,4,1,10,6,8,7,9))))) %>% 
  mutate(MAG_strain = paste0(MAG,"-",Strain)) %>% 
  left_join(., strain_pref, by = "MAG_strain") %>% 
  ggplot(., aes(y = MAG, x = Strain, fill = Preference, size = abs(genome_size))) +
  ggplot2::geom_count(color = "black", shape = 21, stat = "identity")+
  scale_radius("Genome size", range = c(3.5,8.5), breaks = seq(500,2500,by=500)) +
  # ggplot2::scale_fill_distiller("Genome size",palette="RdBu", na.value="white", 
    # limits = c(1000,3000), trans = 'sqrt') + 
    scale_fill_manual("Preference", values = brewer.pal(n = 3, "Accent")[1:2])+
  theme_bw()+
  ylab("")+
  xlab("")+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
      strip.text=element_text(size=12), legend.position = "top",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
    guides(fill  = guide_legend(title = "Preference", override.aes = list(size = 5),
                         nrow = 2),
    size = guide_legend(title = "Genome Size", nrow = 4, override.aes = list(fill = "grey", col = "black")))

print(p_genome_size_strains)

svg("./Figures/DESMAN/gene_assignments1.svg", width =9, height = 6)
print(p_genome_size_strains)
dev.off()

png("./Figures/DESMAN/gene_assignments1.png", width =9, height = 6, res = 500, units = "in")
print(p_genome_size_strains)
dev.off()

# Plot results 2
p_genome_size_strains2 <- results_desm_ga_all[[1]] %>% 
  dplyr::filter(!MAG %in% c("MAG5", "MAG2")) %>% 
  mutate(MAG = factor(MAG, levels = c(paste0("MAG",c(3,4,1,10,6,8,7,9))))) %>% 
  mutate(MAG_strain = paste0(MAG,"-",Strain)) %>% 
  left_join(., strain_pref, by = "MAG_strain") %>% 
  ggplot(., aes(y = genome_size, x = Preference)) +
  geom_jitter(size = 4, shape= 21, aes(fill = MAG),  width = 0.1)+
  # geom_violin(alpha = 0)+
  geom_boxplot(outlier.shape = NA, alpha = 0, width = 0.3)+
  scale_fill_brewer("", palette = "Set1")+
  theme_bw()+
  ylab("Genome size (number of genes)")+
  xlab("")+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12),
      strip.text=element_text(size=12), legend.position = "top",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylim(0, 3000)

print(p_genome_size_strains2)

png("./Figures/DESMAN/gene_assignments2.png", width =6, height = 5, res = 500, units = "in")
print(p_genome_size_strains2)
dev.off()

# Compare completeness with strain genome sizes
p_genome_size_strains3 <- results_desm_ga_all[[1]] %>% 
  dplyr::filter(!MAG %in% c("MAG5", "MAG2")) %>%
  left_join(., MGT_df, by = c("MAG" = "MAG_id")) %>% 
  mutate(MAG = factor(MAG, levels = c(paste0("MAG",c(3,4,1,10,6,8,7,9))))) %>% 
  mutate(MAG_strain = paste0(MAG,"-",Strain)) %>% 
  left_join(., strain_pref, by = "MAG_strain") %>% 
  ggplot(., aes(y = genome_size, x = Completeness)) +
  geom_point(size = 4, shape= 21, aes(fill = MAG))+
  # geom_violin(alpha = 0)+
  # geom_boxplot(outlier.shape = NA, alpha = 0, width = 0.3)+
  scale_fill_brewer("", palette = "Set1")+
  theme_bw()+
  ylab("Genome size (number of genes)")+
  xlab("% completeness")+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12),
      strip.text=element_text(size=12), legend.position = "top",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylim(0, 3000)+
  xlim(50,100)+
  geom_smooth(method = "lm", col = "black")

print(p_genome_size_strains3)

png("./Figures/DESMAN/gene_assignments3.png", width =6, height = 5, res = 500, units = "in")
print(p_genome_size_strains3)
dev.off()

#
p_genome_size_strains4a <- results_desm_ga_all[[1]] %>% 
  dplyr::filter(!MAG %in% c("MAG5", "MAG2")) %>%
  left_join(., MGT_df, by = c("MAG" = "MAG_id")) %>% 
  mutate(MAG = factor(MAG, levels = c(paste0("MAG",c(3,4,1,10,6,8,7,9))))) %>% 
  mutate(MAG_strain = paste0(MAG,"-",Strain)) %>% 
  left_join(., strain_pref, by = "MAG_strain") %>% 
  ggplot(., aes(y = genome_size, x = Topt)) +
  geom_point(size = 4, shape= 21, aes(fill = MAG))+
  # geom_violin(alpha = 0)+
  # geom_boxplot(outlier.shape = NA, alpha = 0, width = 0.3)+
  scale_fill_brewer("", palette = "Set1")+
  theme_bw()+
  ylab("Genome size (number of genes)")+
  xlab("OGT - °C")+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12),
      strip.text=element_text(size=12), legend.position = "top",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylim(0, 3250)+
  # xlim(50,100)+
  geom_smooth(method = "lm", col = "black")

p_genome_size_strains4b <- results_desm_ga_all[[1]] %>% 
  dplyr::filter(!MAG %in% c("MAG5", "MAG2")) %>%
  left_join(., MGT_df, by = c("MAG" = "MAG_id")) %>% 
  mutate(MAG = factor(MAG, levels = c(paste0("MAG",c(3,4,1,10,6,8,7,9))))) %>% 
  mutate(MAG_strain = paste0(MAG,"-",Strain)) %>% 
  left_join(., strain_pref, by = "MAG_strain") %>% 
  ggplot(., aes(y = genome_size, x = log(2)/MGT)) +
  geom_point(size = 4, shape= 21, aes(fill = MAG))+
  # geom_violin(alpha = 0)+
  # geom_boxplot(outlier.shape = NA, alpha = 0, width = 0.3)+
  scale_fill_brewer("", palette = "Set1")+
  theme_bw()+
  ylab("Genome size (number of genes)")+
  xlab(expression("Growth rate - "~"h"^-1))+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12),
      strip.text=element_text(size=12), legend.position = "top",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylim(0, 3250)+
  # xlim(50,100)+
  geom_smooth(method = "lm", col = "black")

png("./Figures/DESMAN/gene_assignments4.png", width = 10, height = 5, res = 500, units = "in")
cowplot::plot_grid(p_genome_size_strains4a, p_genome_size_strains4b, ncol = 2, align ="hv")
dev.off()
```

#### **upset diagram**

```{r desman-5, dpi = 500, warning = FALSE, fig.width = 7.5, fig.height = 5, dev = c("png","pdf")}
# Long to wide
geneAssign_df_wide <- tidyr::spread(results_desm_ga_all[[2]], Strain, Presence) %>% 
  dplyr::filter(MAG == "MAG8")

# Plot
upset(geneAssign_df_wide, sets = c("Strain 1", "Strain 2", "Strain 3", "Strain 4",
                                   "Strain 5", "Strain 6"), mb.ratio = c(0.6, 0.45),
      order.by = "freq", point.size = 3.5,
      mainbar.y.label = "Gene intersections", sets.x.label = "Number of genes",
      text.scale = c(1.5, 1.5, 1.5, 1.4, 2, 1.5),
      # scale.intersections = "log2",
      keep.order = FALSE,
      line.size = 1.5)

# Indicate site and season-specific differential abundant genes
geneAssign_df_wide <- left_join(geneAssign_df_wide, res_deseq,
                                by = c("sseqid" = "gene_oid"))
```


#### Transcript assignment

```{r MAG8-DESeq-plot-upset1, dpi = 500, warning = FALSE, fig.width = 7.5, fig.height = 5, dev = c("png","pdf")}
# Extract upregulated genes of depth comparison
geneAssign_df_wide_transcripts_depth <- geneAssign_df_wide %>%
  dplyr::filter(sseqid %in% unique(MAG8_deseq_results_depth_fin$gene_oid)) %>%
  dplyr::select(sseqid:`Strain 5`)

geneAssign_df_wide_transcripts_depth <- MAG8_deseq_results_depth_fin %>%
  dplyr::select(log2FoldChange, gene_oid, regulation) %>%
  left_join(geneAssign_df_wide_transcripts_depth, ., by = c("sseqid" = "gene_oid"))

# Extract upregulated genes of surface comparison
geneAssign_df_wide_transcripts_temp <- geneAssign_df_wide %>%
  dplyr::filter(sseqid %in% unique(MAG8_deseq_results_temp_fin$gene_oid)) %>%
  dplyr::select(sseqid:`Strain 5`)

geneAssign_df_wide_transcripts_temp <- MAG8_deseq_results_temp_fin %>%
  dplyr::select(log2FoldChange, gene_oid, regulation) %>%
  left_join(geneAssign_df_wide_transcripts_temp, ., by = c("sseqid" = "gene_oid"))

# Evaluate distribution of transcripts expression over strains
# Plot upregulated genes
geneAssign_df_wide_transcripts_depth %>%
  unique() %>%
  upset(.,
      sets = c("Strain 1", "Strain 2","Strain 3", "Strain 4", "Strain 5"),
      mb.ratio = c(0.55, 0.45),
      order.by = "freq", point.size = 3.5,
      mainbar.y.label = "Gene intersections", sets.x.label = "Number of genes",
      text.scale = c(1.5, 1.5, 1.5, 1.4, 2, 3),
      scale.intersections = "log2",
      keep.order = FALSE,
      line.size = 1.2)

geneAssign_df_wide_transcripts_depth %>%
  dplyr::filter(regulation == "Downregulated") %>%
  unique() %>%
  summary()

#
# geneAssign_df_wide_transcripts_temp %>% dplyr::filter(regulation == "Upregulated") %>%
#   upset(.,
#       sets = c("Strain 1", "Strain 2", "Strain 3", "Strain 4",
#                                    "Strain 5"), mb.ratio = c(0.55, 0.45),
#       order.by = "freq", number.angles = 30, point.size = 3.5,
#       mainbar.y.label = "Gene intersections", sets.x.label = "Number of genes",
#       text.scale = c(1.5, 1.5, 1.5, 1.4, 2, 0.75),
#       show.numbers = FALSE,
#       scale.intersections = "log2",
#       keep.order = FALSE,
#       line.size = 1.2,
#       boxplot.summary = "log2FoldChange")


# geneAssign_df_wide_transcripts_temp %>% dplyr::filter(regulation == "Downregulated") %>%
#   upset(.,
#       sets = c("Strain 1", "Strain 2", "Strain 3", "Strain 4",
#                                    "Strain 5"), mb.ratio = c(0.55, 0.45),
#       order.by = "freq", number.angles = 30, point.size = 3.5,
#       mainbar.y.label = "Gene intersections", sets.x.label = "Number of genes",
#       text.scale = c(1.5, 1.5, 1.5, 1.4, 2, 0.75),
#       show.numbers = FALSE,
#       scale.intersections = "log2",
#       keep.order = FALSE,
#       line.size = 1.2,
#       boxplot.summary = "log2FoldChange")

```

#### Assign functions
```{r MAG8-DESeq-plot-upset2, dpi = 500, warning = FALSE, fig.width = 8.5, fig.height = 5, dev = c("png","pdf")}
# Assign interactins for each strain
geneAssign_df_wide_transcripts_depth$Strain_interaction <-
  interaction(geneAssign_df_wide_transcripts_depth$`Strain 1`,
              geneAssign_df_wide_transcripts_depth$`Strain 2`,
              geneAssign_df_wide_transcripts_depth$`Strain 3`,
              geneAssign_df_wide_transcripts_depth$`Strain 4`,
              geneAssign_df_wide_transcripts_depth$`Strain 5`,
              sep = "-")

# Assign functions
geneAssign_df_wide_transcripts_depth <- left_join(geneAssign_df_wide_transcripts_depth,
                                                  merged_file_annot,
                                                  by = c("sseqid" = "gene_oid"))

# Check for gene enrichment in interactions of strain 1 - 4 / strain 2 - 5 / strain 3
# these interactions correspond to interaction term 1-0-0-1-0 / 0-1-0-0-1 / 0-0-1-0-0

# geneAssign_df_wide_transcripts_depth %>%
#   dplyr::filter(grepl("1-.-.-1-.", Strain_interaction))

geneAssign_df_wide_transcripts_depth %>%
  # dplyr::filter(!grepl("0-0-0-0-0", Strain_interaction)) %>%
  # dplyr::filter(grepl("1-0-1-0-0", Strain_interaction)) %>%
  dplyr::select(sseqid:ko_name) %>%
  # dplyr::filter(!is.na(ko_name)) %>%
  unique() %>%
  ggplot(., aes(x = Strain_interaction, y = log2FoldChange, fill = regulation))+
  # geom_point(shape = 21, size = 4)+
  geom_boxplot(alpha = 0.4, outlier.shape = NA)+
  # geom_bar(stat = "count")+
  scale_fill_brewer("", palette = "Set1")+
  theme_bw()+
  theme(axis.text.x=element_text(size=12, angle = 30, hjust = 1),
        axis.text.y = element_text(size = 12),
        axis.title=element_text(size=16),
        title=element_text(size=16), legend.text=element_text(size=16),
        strip.text = element_text(size = 16))+
  guides(fill = FALSE)
  # ylim(0,40)

```


### Gene enrichment

```{r MAG8-strains-gsea, dpi = 500, warning = FALSE, fig.width = 10, fig.height = 4, dev = c("png","pdf")}
# Test for enrichment of functional categories in strains with small inferred genomes
## Test for enrichment of KO level C terms
bg_gsea <- geneAssign_df_annot_func %>%
  dplyr::select(ko_level_C, sseqid) %>%
  distinct()
bg_gsea[is.na(bg_gsea)] <- "Unknown"

s1g <- geneAssign_df_annot_func %>%
  dplyr::filter(`Strain 5` == 1) %>%
  dplyr::select(sseqid) %>%
  unique()

strain1_gsea <- enricher(gene = s1g$sseqid,
         universe = bg_gsea$sseqid,
         TERM2GENE = bg_gsea,
         pvalueCutoff = 0.05,
         qvalueCutoff = 0.2)

```


### Strain DOM usage

```{r MAG8-strains-DOM, dpi = 500, warning = FALSE, fig.width = 10, fig.height = 4, dev = c("png","pdf")}
# Import DOM usage table
DOM_usage_df <- read.csv("./IMG_annotation/DOM_usage.csv", stringsAsFactors = FALSE,
                         check.names =FALSE)
DOM_usage_df$DOM_type <- gsub("\\n","\n", DOM_usage_df$DOM_type, fixed = TRUE)

# Retain COG_ids that are found in DOM_usage_df list
COG_profiles_sub <- geneAssign_df_annot_func %>%
  dplyr::filter(cog_id %in% DOM_usage_df$COG_ID)

COG_profiles_sub <- dplyr::left_join(COG_profiles_sub, DOM_usage_df,
                                     by = c("cog_id" = "COG_ID")) %>%
  dplyr::select(sseqid, Strain_interaction, cog_id, DOM_type) %>%
  unique()

# Add COG counts
COG_profiles_sub_cnt <- COG_profiles_sub %>%
  group_by(Strain_interaction, DOM_type) %>%
  mutate(sum_counts = n())

# Order COG ids according to classification
COG_order <- DOM_usage_df %>%
  dplyr::filter(COG_ID %in% unique(COG_profiles_sub_cnt$cog_id))
COG_profiles_sub_cnt$cog_id <- factor(COG_profiles_sub_cnt$cog_id,
                                        levels = COG_order$COG_ID)

# # make heatmap
# hm_DOC1 <- complete(COG_profiles_sub_cnt, cog_id,
#          fill = list(sum_counts = 0)) %>%
#   # dplyr::filter(Genome %in% selected_genomes) %>%
#   ggplot(aes(y = Strain_interaction, x= cog_id)) + # x and y axes => Var1 and Var2
#   geom_tile(aes(fill = sum_counts), col = "lightgrey") + # background colours are mapped according to the value column
#   geom_text(aes(label = round(sum_counts, 0)), size = 3) + # write the values
#   # scale_fill_gradientn(colours = terrain.colors(10), trans = "log1p")+
#   # scale_fill_gradient(low = "lightblue", high = "darkslategray", na.value="white",
#                       # trans = "log1p", limits=c(1, 40)) +
#   scale_fill_distiller(palette="YlOrRd", na.value="lightgrey", trans = "sqrt",
#                        direction = 1) +
#   theme(panel.grid.major.x=element_blank(), #no gridlines
#         panel.grid.minor.x=element_blank(),
#         panel.grid.major.y=element_blank(),
#         panel.grid.minor.y=element_blank(),
#         panel.background=element_rect(fill="white"), # background=white
#         axis.text.x = element_text(angle=45, hjust = 1, vjust=1, size = 12,face = "bold"),
#         plot.title = element_text(size=20,face="bold"),
#         axis.text.y = element_text(size = 12,face = "bold"))+
#   theme(legend.title=element_text(face="bold", size=14)) +
#   scale_x_discrete(name="") +
#   scale_y_discrete(name="") +
#   labs(fill="Gene\ncount")

# print(hm_DOC1)

# Barplot
p_DOM_dist <- COG_profiles_sub_cnt %>%
  dplyr::filter(Strain_interaction != "0-0-0-0-0") %>%
  dplyr::select(Strain_interaction, DOM_type, sum_counts) %>%
  unique() %>%
  ggplot2::ggplot(aes(x = Strain_interaction, y = sum_counts, fill = DOM_type))+
  geom_bar(alpha = 0.8, stat = "identity", color = "black", width = 0.5)+
  scale_fill_brewer("", palette = "Paired")+
  theme_bw()+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
        title=element_text(size=20), legend.text=element_text(size=10),
        legend.background = element_rect(fill="transparent"),
        # axis.text.x = element_text(angle = 65, hjust = 1),
        strip.text.x=element_text(size = 18),
        legend.position="right",
        axis.text.x=element_text(size = 14, angle =45, hjust= 1),
        plot.title = element_text(hjust = 0, size=18))+
  guides(fill=guide_legend(ncol=1))+
  ylab("Number of genes")+
  xlab('Intersection')

print(p_DOM_dist)
```

# 9. iRep analysis
```{r irep-1, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 10}
# Import data
irep_files <- list.files("./iRep/results", pattern = "irep_")
irep_sample_order <- read.table("./iRep/results/sample_order.tsv")
for(file in irep_files){
  # print(file)
  file_tmp <- read.table(paste("./iRep/results/", file, sep = ""),
                         stringsAsFactors = FALSE)
  bin_name <- gsub("irep_results_|.tsv", "", file)
  # print(bin_name)
  file_tmp <- data.frame(Genome_bin = bin_name, iRep_value = file_tmp[,2],
                         iRep_type = rep(c("iRep_filtered","iRep_unfiltered"), 24),
                         Sample = rep(irep_sample_order$V1, each = 2))
  if(file == irep_files[1]) results_irep <- file_tmp else{
    results_irep <- rbind(results_irep, file_tmp)
  }
}

# Rename sample names
results_irep$Sample <- gsub(".A", "", results_irep$Sample, fixed = TRUE)
results_irep$Sample <- gsub(".C", "", results_irep$Sample, fixed = TRUE)

# Rename genome bins
results_irep <- dplyr::left_join(results_irep, new_bin_names, by = c("Genome_bin" = "bins"))
results_irep$new_bin_name <- as.character(results_irep$new_bin_name)
results_irep$new_bin_name <- factor(results_irep$new_bin_name, levels =
                                      c("MAG1.FA-MLB-DN","MAG2.FA-MLB-SN",
                                        "MAG3.FA-MLB-SN", "MAG4.FA-M110-DN",
                                        "MAG5.SP-M110-DD","MAG6.SP-M15-SD",
                                        "MAG7.SU-MLB-SD","MAG8.SU-M110-DCMD",
                                        "MAG9.SU-M15-SN","MAG10.SU-M15-SN"))

# Merge with metadata
results_irep <- dplyr::left_join(results_irep, meta, by = "Sample")
results_irep$iRep_value <- as.numeric(as.character(results_irep$iRep_value))

# Format metadata
results_irep$Site <- as.character(results_irep$Site)
results_irep$Site <- gsub("110", "Lake Michigan\nsite M110", results_irep$Site)
results_irep$Site <- gsub("15", "Lake Michigan\nsite M15", results_irep$Site)
results_irep$Site <- gsub("Buoy", "Muskegon Lake", results_irep$Site)
results_irep$Site <- factor(results_irep$Site, levels = c("Muskegon Lake",
                                                            "Lake Michigan\nsite M15",
                                                            "Lake Michigan\nsite M110"))
results_irep$Season <- as.character(results_irep$Season)
results_irep$Season <- factor(results_irep$Season, levels = c("Spring", "Summer","Fall"))

# Check correlation with the number of haplotypes
results_irep <- left_join(results_irep, metadata_desman, by = c("new_bin_name" = "Genome_ID"))
```

## Plot results  

```{r irep-2, dpi = 500, warning = FALSE, fig.width = 10.5, fig.height = 9.5}
# Plot unfiltered iRep
irep_p1 <- results_irep %>% 
  dplyr::filter(iRep_type == "iRep_unfiltered") %>% 
  ggplot(aes(x = new_bin_name, y = iRep_value, fill = new_bin_name, col = new_bin_name))+
  geom_point(color = "black", alpha = 0.7, size = 4, shape = 21)+
  # scale_shape_manual(values = c(21,22,23))+
  geom_boxplot(alpha = 0.3, col = "black", outlier.shape = NA, size = 0.3)+
  scale_fill_manual("", values = fill_palette)+
  scale_color_brewer("",palette = "Accent")+
  theme_bw()+
  # geom_point(size = 4, color = "black", alpha = 0.7)+
  # scale_shape_manual(values = c(21,24,23))+
  # geom_boxplot(alpha = 0.4)+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
      strip.text=element_text(size=12), legend.position = "bottom",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylab(paste0("iRep"))+
  guides(fill=FALSE)+
  facet_grid(Season~Site, scales ="free")+
  xlab("")+
  scale_y_continuous(labels=scaleFUN, limits = c(1,4.5))+
  # geom_smooth(se = FALSE)+
  guides(fill=FALSE)
  # coord_trans(y = "sqrt")

print(irep_p1)

# Plot filtered irep
irep_p2 <- results_irep %>% 
  dplyr::filter(iRep_type == "iRep_filtered") %>% 
  ggplot(aes(x = new_bin_name, y = iRep_value, fill = new_bin_name, col = new_bin_name))+
  geom_point(color = "black", alpha = 0.7, size = 4, shape = 21)+
  # scale_shape_manual(values = c(21,22,23))+
  geom_boxplot(alpha = 0.3, col = "black", outlier.shape = NA, size = 0.3)+
  scale_fill_manual("", values = fill_palette)+
  scale_color_brewer("",palette = "Accent")+
  theme_bw()+
  # geom_point(size = 4, color = "black", alpha = 0.7)+
  # scale_shape_manual(values = c(21,24,23))+
  # geom_boxplot(alpha = 0.4)+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
      strip.text=element_text(size=12), legend.position = "bottom",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylab(paste0("iRep"))+
  guides(fill=FALSE)+
  facet_grid(Season~Site, scales ="free")+
  xlab("")+
  scale_y_continuous(labels=scaleFUN, limits = c(1,4.5))+
  # geom_smooth(se = FALSE)+
  guides(fill=FALSE)
  # coord_trans(y = "sqrt")

print(irep_p2)
```

```{r irep-3, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 10}
# Only focus on MAGs for which we can determine accurate iRep values 
irep_p3 <- results_irep %>% 
  dplyr::filter(iRep_type == "iRep_filtered") %>% 
  dplyr::filter(!is.na(iRep_value)) %>% 
  ggplot(aes(x = Season, y = iRep_value, fill = new_bin_name, col = new_bin_name))+
  geom_point(color = "black", alpha = 0.7, size = 4, shape = 21)+
  # scale_shape_manual(values = c(21,22,23))+
  geom_boxplot(alpha = 0.3, col = "black", outlier.shape = NA, size = 0.3)+
  scale_fill_manual("", values = fill_palette)+
  scale_color_brewer("",palette = "Accent")+
  theme_bw()+
  # geom_point(size = 4, color = "black", alpha = 0.7)+
  # scale_shape_manual(values = c(21,24,23))+
  # geom_boxplot(alpha = 0.4)+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
      strip.text=element_text(size=12), legend.position = "bottom",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylab(paste0("iRep"))+
  guides(fill=FALSE)+
  facet_grid(new_bin_name~Site, scales ="free")+
  xlab("")+
  scale_y_continuous(labels=scaleFUN, limits = c(1,4.5))+
  # geom_smooth(se = FALSE)+
  guides(fill=FALSE)
  # coord_trans(y = "sqrt")

print(irep_p3)

```

```{r irep-4, dpi = 500, warning = FALSE, fig.width = 6, fig.height = 4}
# Only focus on MAGs for which we can determine accurate iRep values 
irep_p4 <- results_irep %>% 
  dplyr::filter(iRep_type == "iRep_filtered") %>% 
  dplyr::filter(!is.na(iRep_value)) %>% 
  ggplot(aes(x = new_bin_name, y = iRep_value, fill = Site, col = Site))+
  # geom_point(color = "black", alpha = 0.7, size = 4, shape = 21)+
  # scale_shape_manual(values = c(21,22,23))+
  geom_boxplot(alpha = 0.3, col = "black", outlier.shape = NA, size = 0.3)+
  scale_fill_brewer("", palette = "Accent")+
  scale_color_brewer("",palette = "Accent")+
  theme_bw()+
  # geom_point(size = 4, color = "black", alpha = 0.7)+
  # scale_shape_manual(values = c(21,24,23))+
  # geom_boxplot(alpha = 0.4)+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
      strip.text=element_text(size=12), legend.position = "bottom",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylab(paste0("iRep"))+
  # facet_grid(new_bin_name~Site, scales ="free")+
  xlab("")+
  scale_y_continuous(labels=scaleFUN, limits = c(1,4.5))
  # geom_smooth(se = FALSE)+
  # guides(fill=FALSE)
  # coord_trans(y = "sqrt")

print(irep_p4)
```

```{r irep-5, dpi = 500, warning = FALSE, fig.width = 8, fig.height = 4}
# Focus on all MAGs
irep_p5 <- results_irep %>% 
  dplyr::filter(iRep_type == "iRep_unfiltered") %>% 
  dplyr::filter(!is.na(iRep_value)) %>% 
  ggplot(aes(x = new_bin_name, y = iRep_value, fill = Site, col = Site))+
  # geom_point(color = "black", alpha = 0.7, size = 4, shape = 21)+
  # scale_shape_manual(values = c(21,22,23))+
  geom_boxplot(alpha = 0.3, col = "black", outlier.shape = NA, size = 0.3)+
  scale_fill_brewer("", palette = "Accent")+
  scale_color_brewer("",palette = "Accent")+
  theme_bw()+
  # geom_point(size = 4, color = "black", alpha = 0.7)+
  # scale_shape_manual(values = c(21,24,23))+
  # geom_boxplot(alpha = 0.4)+
  theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=10),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
      strip.text=element_text(size=12), legend.position = "bottom",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylab(paste0("iRep"))+
  # facet_grid(new_bin_name~Site, scales ="free")+
  xlab("")+
  scale_y_continuous(labels=scaleFUN, limits = c(1,4.5))
  # geom_smooth(se = FALSE)+
  # guides(fill=FALSE)
  # coord_trans(y = "sqrt")

print(irep_p5)


```

# Pangenome analysis  


```{r panG-1, dpi = 500, warning = FALSE, fig.width = 7, fig.height = 10}
# Import and format panG data
# Need to use fread because of the way additional COG categories are interspersed by tabs...
panG <- data.table::fread("./panG/SUMMARY_Final_bins_wMIXED/panG-LIM-ALL_protein_clusters_summary.txt")

# Remove COG annotation
panG <- panG %>%
  dplyr::select(c("bin_name", "genome_name",  "gene_callers_id",
     "COG_FUNCTION_ACC", "COG_FUNCTION", "aa_sequence"))

panG_MAG <- panG
panG_MAG$gene_callers_id <- as.character(panG_MAG$gene_callers_id)

# Add unique gene identifier 
panG_MAG$unique_gene_callers_id <- interaction(panG_MAG$genome_name, 
                                               panG_MAG$gene_callers_id,
                                               sep = ".")

sum(panG_MAG$genome_name=="MAG8_SU_M110_DCMD")
panG_MAG$aa_sequence <- gsub("-", "", panG_MAG$aa_sequence)

# Export all AA sequences for annotation with KAAS or for blast
# for(i in 1:length(unique(panG_MAG$genome_name))){
#   tmp <- panG_MAG %>% dplyr::filter(genome_name == unique(panG_MAG$genome_name)[i]) %>%
#     droplevels()
#     write.table(file = paste0("./panG/", unique(panG_MAG$genome_name)[i],
#                               "_aa_export.fa"),
#            paste0(">", tmp$unique_gene_callers_id,
#       "\n", tmp$aa_sequence, sep = ""),
#       quote = FALSE, row.names = FALSE, col.names = FALSE
#       )
# }

# Import KEGG annotation through KAAS (http://www.genome.jp/tools/kaas/) of amino
# acid sequences
ko_files <- list.files(".", pattern = "KO_annot",
                       recursive = TRUE)
panG_ko <- data.frame()
for(ko_file in ko_files){
  tmp <- data.table::fread(file = ko_file, sep = '\t', fill = TRUE, header = TRUE)
  tmp <- data.frame(tmp, Genome = do.call(rbind, strsplit(ko_file, "-"))[,2])
  if(ko_file == ko_files[1]) panG_ko <- tmp else{
    panG_ko <- rbind(panG_ko, tmp)
  }
}
panG_ko$Genome <- gsub(".txt","",panG_ko$Genome)
colnames(panG_ko)[1:2] <- c("unique_gene_callers_id", "ko_id")
panG_ko <- panG_ko[panG_ko$ko_id != "",]
panG_ko$ko_id <- gsub(" ","", panG_ko$ko_id)

# Annotate KO_IDs with KEGG hierarchy
panG_ko <- dplyr::left_join(panG_ko, ko_path_df, by = "ko_id")

# join with corresponding COG ids
panG_ko_cog <- dplyr::left_join(panG_MAG, panG_ko, 
                                by = "unique_gene_callers_id")

# Shorten/change ko_level_B annotation a bit
panG_ko_cog$ko_level_B[panG_ko_cog$ko_level_B == "Cellular community - prokaryotes"] <- "Biofilm formation & quorum sensing"
panG_ko_cog$ko_level_B[panG_ko_cog$ko_level_B == "Xenobiotics biodegradation and metabolism"] <- "Xenobiotics degradation"
panG_ko_cog$ko_level_C[panG_ko_cog$ko_level_C == "Biofilm formation - Escherichia coli "] <- "Biofilm formation"
panG_ko_cog$ko_level_C[panG_ko_cog$ko_level_C == "Biofilm formation - Pseudomonas aeruginosa "] <- "Biofilm formation"

panG_ko$ko_level_C[panG_ko$ko_level_C == "Biofilm formation - Escherichia coli "] <- "Biofilm formation"
panG_ko$ko_level_C[panG_ko$ko_level_C == "Biofilm formation - Pseudomonas aeruginosa "] <- "Biofilm formation"

# Change the gene clusters shared between MAG1_6_10 to mixed category
panG_ko_cog$bin_name[panG_ko_cog$bin_name == "MAG1_6_10_PC"] <- "MIXED_PCs"

# Add column denoting whether it is core/mixed or accessory
panG_ko_cog$bin_core <- factor(panG_ko_cog$bin_name == "CORE_PC" | panG_ko_cog$bin_name == "MIXED_PCs")
panG_ko_cog$bin_core <- plyr::revalue(panG_ko_cog$bin_core, replace = c("TRUE" = "CORE/Mixed", "FALSE" = "Accessory"))
```

```{r panG-2, dpi = 500, warning = FALSE, fig.width = 13, fig.height = 8, dev=c("png","pdf")}
# Perform hypergeometric test to see if there is functional enrichment in the pangenome
for(i_genome in 1:length(unique(panG_ko$Genome))){
  # Extract background gene annotations
  bg_gsea <- panG_ko %>% dplyr::filter(Genome %in% unique(panG_ko$Genome)[i_genome]) %>% 
  dplyr::select(ko_level_C,unique_gene_callers_id) %>% 
  distinct()
  
  panG_gseq <- panG_ko_cog %>% 
    dplyr::filter(Genome == unique(panG_ko$Genome)[i_genome] & 
                    bin_core == "Accessory") %>%
    dplyr::select(ko_level_C, unique_gene_callers_id) %>% 
    distinct()
  
  panG_gsea <- enricher(gene = panG_gseq$unique_gene_callers_id,
         universe = bg_gsea$unique_gene_callers_id, 
         TERM2GENE = bg_gsea,
         pvalueCutoff = 0.05,
         qvalueCutoff = 0.2)

  # Important to filter here
  tmp_gseq <- data.frame(panG_gsea@result, Genome = unique(panG_ko$Genome)[i_genome]) %>% 
    dplyr::filter(p.adjust < 0.05)
  
  if(i_genome == 1) results_gseq <- tmp_gseq else {
    results_gseq <- rbind(results_gseq, tmp_gseq)
  }
  
  genes_per_gseq <- panG_gseq %>% 
     dplyr::filter(ko_level_C %in% panG_gsea@result$Description) %>%
     group_by(ko_level_C) %>% 
     summarize(Counts = n())
  
  genes_per_gseq <- data.frame(genes_per_gseq, 
                               Genome = unique(panG_ko$Genome)[i_genome])
  
  if(i_genome == 1) results_pergene <- genes_per_gseq else {
    results_pergene <- rbind(results_pergene, genes_per_gseq)
  }
}

# Export enricher results
write.table(results_gseq, file = "./PanG/results_enrichment.csv", row.names = FALSE,
  col.names = TRUE, sep = ",")

# Calculate fraction of enriched genes vs. annotated panG genes
results_pergene_sum <- results_pergene %>% 
  group_by(Genome) %>% 
  summarize(enrich_counts = sum(Counts))

annotated_fraction_panG <- panG_ko_cog %>% 
  dplyr::filter(bin_core == "Accessory") %>% 
  group_by(genome_name) %>% 
  dplyr::select(unique_gene_callers_id) %>% 
  distinct() %>% 
  summarize(panG_counts = n())

print(annotated_fraction_panG)

results_pergene_sum <- left_join(results_pergene_sum, annotated_fraction_panG,
                                 by = c("Genome" = "genome_name"))

# Merge actual unique gene counts per gsea category with the accessory genome sizes
results_pergene_sum <- results_pergene_sum %>% 
  mutate(frac_enrich = enrich_counts/panG_counts)
print(results_pergene_sum)

# Make plot
results_gseq$Genome <- gsub("_","-",results_gseq$Genome)
results_gseq$Genome <- gsub("-FA",".FA",results_gseq$Genome)
results_gseq$Genome <- gsub("-SU",".SU",results_gseq$Genome)
results_gseq$Genome <- gsub("-SP",".SP",results_gseq$Genome)
results_gseq$Genome <- factor(results_gseq$Genome, levels =
                                      c("MAG5.SP-M110-DD","MAG2.FA-MLB-SN",
                                        "MAG3.FA-MLB-SN", "MAG4.FA-M110-DN",
                                        "MAG1.FA-MLB-DN", "MAG10.SU-M15-SN",
                                        "MAG6.SP-M15-SD", "MAG8.SU-M110-DCMD",
                                        "MAG7.SU-MLB-SD","MAG9.SU-M15-SN"))

# Make heatmap instead
results_gseq_wide <- complete(results_gseq, Genome, Description,
         fill = list(Count = 0)) %>% 
  dplyr::select(Description, Genome, Count) %>% 
  spread(., Description, Count)

dat <- results_gseq_wide[,2:ncol(results_gseq_wide)]  # numerical columns
rownames(dat) <- results_gseq_wide$Genome
# row.order <- hclust(dist(dat))$order # clustering
col.order <- hclust(dist(t(dat)))$order
dat_new <- dat[, col.order] # re-order matrix accoring to clustering
rownames(dat_new) <- results_gseq_wide$Genome
  
df_molten_dat <- melt(as.matrix(dat_new)) # reshape into dataframe
names(df_molten_dat)[c(1:2)] <- c("Genome", "Description")

p_panG6 <- complete(df_molten_dat, Genome, Description,
         fill = list(value = 0)) %>%
  ggplot(aes(x = Genome, y = Description)) + # x and y axes => Var1 and Var2
  geom_tile(aes(fill = value), col = "lightgrey") + # background colours are mapped according to the value column
  geom_text(aes(label = round(value, 0)), size = 4) + # write the values
  scale_fill_distiller(palette="YlOrRd", na.value="lightgrey", trans = "sqrt",
                       direction = 1) +
  theme(panel.grid.major.x=element_blank(), #no gridlines
        panel.grid.minor.x=element_blank(),
        panel.grid.major.y=element_blank(),
        panel.grid.minor.y=element_blank(),
        panel.background=element_rect(fill="white"), # background=white
        axis.text.x = element_text(angle=45, 
                                   hjust = 0, vjust=0, size = 12,face = "bold"),
        plot.title = element_text(size=20,face="bold"),
        axis.text.y = element_text(size = 12,face = "bold"))+
  theme(legend.title=element_text(face="bold", size=14)) +
  scale_x_discrete(name="", position = "top") +
  scale_y_discrete(name="") +
  labs(fill="Gene\ncount")

print(p_panG6)

p_panG7 <- 
  df_molten_dat %>% 
  left_join(., MGT_df, by = c("Genome" = "Genome_ID")) %>% 
  mutate(Genome = factor(Genome, levels = c(
                                          "MAG5.SP-M110-DD",
                                          "MAG2.FA-MLB-SN",
                                          "MAG3.FA-MLB-SN",
                                          "MAG4.FA-M110-DN",
                                          "MAG1.FA-MLB-DN",
                                          "MAG10.SU-M15-SN",
                                          "MAG6.SP-M15-SD",
                                          "MAG8.SU-M110-DCMD",
                                          "MAG7.SU-MLB-SD", 
                                          "MAG9.SU-M15-SN"
                                          ))) %>% 
  dplyr::filter(value > 0) %>% 
  ggplot(aes(x = Genome, y = Description, size = value, fill = Lineage)) + # x and y axes => Var1 and Var2
 ggplot2::geom_count(color = "black", shape = 21, alpha = 1)+
      scale_fill_manual(expression("Growth rate - "~"h"^-1),
                    values = c(rgb(red=t(col2rgb("#deebf7ff")), maxColorValue  = 255), 
                               rgb(red=t(col2rgb("#c6dbefff")), maxColorValue  = 255),
                               rgb(red=t(col2rgb("#9ecae1ff")), maxColorValue  = 255),
                               rgb(red=t(col2rgb("#6baed6ff")), maxColorValue  = 255)
                              ))+
  scale_size("Gene\ncount", range = c(3,14), breaks=seq(0,50,by=10)) +
  ggplot2::theme_bw()+
    theme(axis.text.y=element_text(size=14), axis.title=element_text(size=20),
        axis.text.x=element_text(size=14, angle = -315, hjust = 0),
        title=element_text(size=16), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        strip.text=element_text(size=18),
        legend.position = "right")+
  theme(legend.title=element_text(face="bold", size=14)) +
  scale_x_discrete(name="", position = "top") +
  scale_y_discrete(name="") +
  guides(fill=FALSE)

svg("./Figures/panG.svg", width =13, height = 7)
print(p_panG7)
dev.off()

png("./Figures/panG.png", width =13, height = 7, res = 500, units = "in")
print(p_panG7)
dev.off()
```

# DOC transporters

```{r DOC-1, dpi = 500, warning = FALSE, fig.width = 13, fig.height = 8, include = FALSE, dev = c("png","pdf")}
# Annotate the core and acessory genomes with DOC transporter labels
matches <- unique(grep(paste(DOM_usage_df$COG_ID,collapse = "|"), 
                        panG_ko_cog$COG_FUNCTION_ACC, value = TRUE))


DOC_panG <- panG_ko_cog %>% dplyr::select(bin_name, genome_name,
                              unique_gene_callers_id, COG_FUNCTION_ACC,
                              bin_core) %>% 
  distinct() %>% 
  dplyr::filter(COG_FUNCTION_ACC %in% matches) %>% 
  fuzzyjoin::fuzzy_inner_join(DOM_usage_df,
                            by = c("COG_FUNCTION_ACC" = "COG_ID"), 
                            match_fun = str_detect) %>% 
  dplyr::select(bin_name, bin_core, unique_gene_callers_id, genome_name, COG_ID) %>% 
  distinct()

# Order DOC categories
COG_order <- DOM_usage_df %>% 
  dplyr::filter(COG_ID %in% unique(DOC_panG$COG_ID))
DOC_panG$COG_ID <- factor(DOC_panG$COG_ID,
                                        levels = COG_order$COG_ID)

# Get COG counts in core and accessory genome separately
DOC_panG_cnt_split <- DOC_panG %>% 
  group_by(genome_name, bin_name, COG_ID) %>% 
  summarise(Counts = n())

# make heatmap of accessory genome
hm_DOC1_acc <- DOC_panG_cnt_split %>% 
  complete(., genome_name, bin_name,COG_ID,
         fill = list(Counts = 0)) %>% 
  ungroup() %>% 
  dplyr::filter(bin_name != "CORE_PC" & bin_name != "MIXED_PCs") %>%
  ggplot(aes(y= genome_name, x= COG_ID)) + 
  geom_tile(aes(fill = Counts), col = "lightgrey") + # according to the value column
  geom_text(aes(label = round(Counts, 0)), size = 3) + # write the values
  scale_fill_distiller(palette="YlOrRd", na.value="lightgrey", trans = "sqrt",
                       direction = 1) +
  theme(panel.grid.major.x=element_blank(), #no gridlines
        panel.grid.minor.x=element_blank(), 
        panel.grid.major.y=element_blank(), 
        panel.grid.minor.y=element_blank(),
        panel.background=element_rect(fill="white"), # background=white
        axis.text.x = element_text(angle=45, hjust = 1, vjust=1, size = 12,face = "bold"),
        plot.title = element_text(size=20,face="bold"),
        axis.text.y = element_text(size = 12,face = "bold"))+
  theme(legend.title=element_text(face="bold", size=14)) + 
  scale_x_discrete(name="") +
  scale_y_discrete(name="") +
  labs(fill="Gene\ncount")

print(hm_DOC1_acc)

# Get COG counts in full genome
DOC_panG_cnt_full <- DOC_panG %>% 
  group_by(genome_name, COG_ID) %>% 
  summarise(Counts = n()) %>% 
  dplyr::filter(!genome_name %in% c("MAG2_FA_MLB_SN", "MAG5_SP_M110_DD"))

# Cluster genomes according to DOC profiles
DOC_panG_cnt_full_wide <- spread(DOC_panG_cnt_full, COG_ID, Counts)
dat <- DOC_panG_cnt_full_wide[,2:ncol(DOC_panG_cnt_full_wide)]  # numerical columns
rownames(dat) <- DOC_panG_cnt_full_wide$genome_name
row.order <- hclust(dist(dat))$order # clustering
# col.order <- hclust(dist(t(dat)))$order
dat_new <- dat[row.order, ] # re-order matrix accoring to clustering
rownames(dat_new) <- DOC_panG_cnt_full_wide$genome_name[row.order]
df_molten_dat <- melt(as.matrix(dat_new)) # reshape into dataframe
names(df_molten_dat) <- c("genome_name", "COG_ID", "Counts")

# make heatmap of full genome DOC transporters
hm_DOC2_full <- df_molten_dat %>% 
  complete(., genome_name, COG_ID,
         fill = list(Counts = 0)) %>% 
  left_join(., DOM_usage_df, by = "COG_ID") %>% 
  ungroup() %>% 
  dplyr::filter(!is.na(Counts)) %>%
  dplyr::filter(Counts != 0 & !genome_name %in% c("MAG2_FA_MLB_SN", "MAG5_SP_M110_DD")) %>%
  distinct() %>% 
  mutate(COG_ID = factor(COG_ID, levels = COG_order$COG_ID)) %>% 
  mutate(DOM_type = factor(DOM_type, levels = levels(factor(COG_order$DOM_type))[c(3,1,2,10,8,9,4,5,6,7)])) %>% 
  ggplot(aes(y= genome_name, x= factor(COG_ID), fill = DOM_type, size = Counts)) + 
  geom_count(stat = "identity", shape = 21, alpha = 1)+
  scale_size("Number of Genes", range = c(3,8), breaks=seq(0,30,by=5)) +
  # geom_tile(aes(fill = Counts), col = "lightgrey") + # according to the value column
  # geom_text(aes(label = round(Counts, 0)), size = 3) + # write the values
  scale_fill_brewer(palette="Set3", na.value="lightgrey") +
  theme_bw()+
 theme(axis.text=element_text(size=12), axis.title=element_text(size=16),
      title=element_text(size=16), legend.text=element_text(size=12),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
      strip.text=element_text(size=12), legend.position = "top",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  scale_x_discrete(name="") +
  scale_y_discrete(name="") +
  labs(fill="DOC transporter\ncategory")+
  guides(fill  = guide_legend(override.aes = list(size = 5),
                         nrow = 3),
    color = FALSE)

# png("./Figures/heatmap_DOC.png", height =paper 8, width = 14, units = "in", res = 500)
svg("./Figures/heatmap_DOC.svg", height = 8, width = 14)
print(hm_DOC2_full)
dev.off()
```


# External metagenomes

Get data from here: https://doi.org/10.6084/m9.figshare.7608848

```{r SDP_external, warning = FALSE, fig.width = 12, fig.height = 10, dpi = 500, dev = c("png", "pdf")}
df_BlastExt <- data.table::fread("./ExternalData/merged_blastfiles.tsv", 
                         header = TRUE)
# Import metadata
meta_Ext <- xlsx::read.xlsx("./ExternalData/ismej2017156x1.xlsx", sheetIndex = 1)
meta_Ext <- meta_Ext[!is.na(meta_Ext$metagenomic.sample),]
meta_Ext <- meta_Ext[, c(1, 2, 4, 5)]
meta_Ext <- meta_Ext %>% dplyr::filter(grepl("SRR", accession.number)) # Only data used for which SRR accession IDs (NCBI) were available

# Import contig_ids of each bin
IMG_folders <- list.files("./", pattern = "assembled.names_map", include.dirs = TRUE, recursive = TRUE)
IMG_folders <- IMG_folders[1:10]

for(i in 1:length(IMG_folders)){
  contig_tmp <- cbind(read.table(IMG_folders[i], stringsAsFactors = FALSE)[,1], strsplit(IMG_folders[i], "/")[[1]][2])
  contig_tmp <- data.frame(contig_tmp)
  colnames(contig_tmp) <- c("contig_id", "IMG_taxID")
  if(i == 1){
    contig_ids <- contig_tmp
  } else {
    contig_ids <- rbind(contig_ids, contig_tmp)
  }
}
contig_ids$IMG_taxID <- gsub("IMG_","",contig_ids$IMG_taxID)

# Add manuscript bin IDs and genome sizes
new_bin_names3 <- dplyr::left_join(bin_size, new_bin_names, by = "bins")
new_bin_names3 <- dplyr::left_join(new_bin_names3, new_bin_names2, by = "new_bin_name")
contig_ids <- dplyr::left_join(contig_ids, new_bin_names3, by = "IMG_taxID")

# Order bins according to phylogeny
ord_list_bin_ext <- c(
  "MAG5.SP-M110-DD", "MAG2.FA-MLB-SN",
  "MAG3.FA-MLB-SN", "MAG4.FA-M110-DN",
  "MAG1.FA-MLB-DN", "MAG10.SU-M15-SN",
  "MAG6.SP-M15-SD","MAG8.SU-M110-DCMD",
  "MAG7.SU-MLB-SD","MAG9.SU-M15-SN"
)

contig_ids$new_bin_name <- factor(contig_ids$new_bin_name, levels = ord_list_bin_ext)

# Add Lineage label
contig_ids <- left_join(contig_ids, MGT_df, by = c("new_bin_name" = "Genome_ID"))

# Clean up SRA identifier label
df_BlastExt$SRA <- gsub("_blast.tsv", "", df_BlastExt$SRA)

# Remove contigs not in contig_ids
df_BlastExt <- df_BlastExt %>% dplyr::filter(contig_id %in% contig_ids$contig_id)

# Add contig labels of MAGs
df_BlastExt <- dplyr::left_join(df_BlastExt, contig_ids, by = c("contig_id"))

# Bin the %Identity in intervals of 0.5%
df_BlastExt_sum <- transform(df_BlastExt, bin_group = cut(identity,  breaks=seq(0, 100, 0.5)))

# Add extra column that converts the binning range to a numeric x-coordinate that
# is positioned in the middle of the binning interval
df_BlastExt_sum$bin_group <- gsub("\\(|]", "", 
                               as.character(df_BlastExt_sum$bin_group))
df_BlastExt_sum$bin_xcoord <- as.numeric(do.call(rbind,
                                              strsplit(df_BlastExt_sum$bin_group,
                                                       ","))[,1])+0.25

# Add metadata
df_BlastExt_sum <- left_join(df_BlastExt_sum, meta_Ext, by = c("SRA" = "accession.number"))
```

```{r SDP_external-2, warning = FALSE, fig.width = 12, fig.height = 7, dpi = 500, dev = c("png", "pdf")}
# Plot % reads corrected for genome size over threshold of 0.95
blast_df_sum_comp <- df_BlastExt_sum %>%
  group_by(SRA, new_bin_name) %>%
  dplyr::count(bin_xcoord)

# Add Lineage label
blast_df_sum_comp <- left_join(blast_df_sum_comp, MGT_df,
                               by = c("new_bin_name" = "Genome_ID"))
blast_df_sum_comp <- left_join(blast_df_sum_comp, new_bin_names3,
                               by = "new_bin_name")

id_thresh <- 95-0.25
map_disc_cum <- blast_df_sum_comp  %>% 
  dplyr::filter(bin_xcoord > id_thresh) %>% group_by(SRA, new_bin_name) %>% 
  mutate(cum_rel_reads_mapped = cumsum(n))%>% 
  dplyr::filter(bin_xcoord == 100 - 0.25)

map_disc_cum <- left_join(map_disc_cum, meta_Ext, by = c("SRA" = "accession.number"))

SRA_ranking <- map_disc_cum %>% 
  dplyr::filter(new_bin_name == "MAG8.SU-M110-DCMD" & bin_xcoord > 94.5) %>%
  dplyr::group_by(metagenomic.sample, new_bin_name) %>% 
  summarise(mean_n = median(cum_rel_reads_mapped))
SRA_ranking <- as.character(SRA_ranking$metagenomic.sample[rev(order(SRA_ranking$mean_n))])
map_disc_cum$metagenomic.sample <- factor(as.character(map_disc_cum$metagenomic.sample), levels = SRA_ranking)

MAG_ranking <- map_disc_cum %>% 
  dplyr::filter(bin_xcoord > 94.5) %>%
  dplyr::group_by(new_bin_name) %>% 
  summarise(sum_n = sum(cum_rel_reads_mapped))
MAG_ranking <- as.character(MAG_ranking$new_bin_name[rev(order(MAG_ranking$sum_n))])
map_disc_cum$new_bin_name <- factor(as.character(map_disc_cum$new_bin_name), levels = MAG_ranking)

# Evaluate normalized abundance (mapped reads/bin size at > 94.5% identity).
p_sdisc_cum3 <- map_disc_cum %>% 
  dplyr::filter(cum_rel_reads_mapped > 10) %>% 
  dplyr::mutate(norm_cum_rel_reads_mapped = cum_rel_reads_mapped/(bin_size/1e6)) %>% 
  ggplot(aes(x = metagenomic.sample, y =  100*norm_cum_rel_reads_mapped/1e6, 
                                        fill = Lineage))+
  theme_bw()+
  scale_fill_manual("",
                    values = c("#deebf7ff", "#c6dbefff","#9ecae1ff",
                               "#6baed6ff"))+
  geom_jitter(size = 2, shape = 21, color = "black", alpha = 0.75, width = 0.15)+
  geom_boxplot(alpha = 0.5, outlier.shape = NA)+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
      title=element_text(size=20), legend.text=element_text(size=12),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size=11, angle = 45, hjust = 1),
      strip.text=element_text(size=12), legend.position = "bottom",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylab(paste0("Normalized relative abundance\n (> ", id_thresh-0.25, "% NI)"))+
  xlab("")+
  guides(fill=FALSE)+
  facet_wrap(.~new_bin_name, nrow = 2)+
  scale_y_sqrt()

print(p_sdisc_cum3)
```

```{r SDP_external-3, warning = FALSE, fig.width = 9, fig.height = 8, dpi = 500, dev = c("png", "pdf")}
# Make one graph of abundances across different external metagenomic data sets for each MAG
p_sdisc_cum4 <- map_disc_cum %>% 
  dplyr::mutate(norm_cum_rel_reads_mapped = cum_rel_reads_mapped/(bin_size/1e6)) %>% 
  # dplyr::filter(cum_rel_reads_mapped > 10) %>% # In case you want to filter samples with < x reads out
  ggplot(aes(x = new_bin_name, y =  100*cum_rel_reads_mapped/1e6))+
  theme_bw()+
  scale_fill_brewer("", palette = "Paired")+
  geom_jitter(size = 4, shape = 21, color = "black", alpha = 0.5, width = 0.15,
              aes(fill = metagenomic.sample))+
  geom_violin(alpha = 0.5, adjust = 1, draw_quantiles = TRUE, scale = "width")+
  stat_summary(fun.data=mean_sdl, fun.args = list(mult = 1), 
                 geom="pointrange", color="#333333", size = 1.05)+
  theme(axis.title=element_text(size=20),
      title=element_text(size=20), legend.text=element_text(size=12),
      legend.background = element_rect(fill="transparent"),
      axis.text.x = element_text(size=14),
      axis.text.y = element_text(size = 14),
      strip.text=element_text(size=12), legend.position = "bottom",
      strip.background = element_rect(fill = adjustcolor("gray", 0.15)))+
  ylab(paste0("Normalized relative abundance\n (> ", id_thresh-0.25, "% NI)"))+
  xlab("")+
  scale_y_sqrt()+ 
  coord_flip()+
  guides(fill = guide_legend(nrow = 4))

print(p_sdisc_cum4)

map_disc_cum_phy <- map_disc_cum
map_disc_cum_phy$new_bin_name <- factor(map_disc_cum_phy$new_bin_name,
                                        levels = c(
                                          "MAG5.SP-M110-DD",
                                          "MAG2.FA-MLB-SN",
                                          "MAG3.FA-MLB-SN",
                                          "MAG4.FA-M110-DN",
                                          "MAG1.FA-MLB-DN",
                                          "MAG10.SU-M15-SN",
                                          "MAG6.SP-M15-SD",
                                          "MAG8.SU-M110-DCMD",
                                          "MAG7.SU-MLB-SD", 
                                          "MAG9.SU-M15-SN"
                                          )
)

# Make one graph of abundances across different external metagenomic datasets 
# for each MAG with a single boxplot to add to phylogenomic tree
p_sdisc_cum5 <- map_disc_cum_phy %>% 
  # dplyr::filter(cum_rel_reads_mapped > 10) %>% 
  dplyr::mutate(norm_cum_rel_reads_mapped = cum_rel_reads_mapped/(bin_size/1e6)) %>% 
  ggplot(aes(x = new_bin_name, y =  100*cum_rel_reads_mapped/1e6, fill = Lineage))+
  theme_bw()+
  scale_fill_manual("",
                    values = c("#deebf7ff", "#c6dbefff","#9ecae1ff",
                               "#6baed6ff"))+
  # geom_violin(scale = "width")+
  # stat_summary(fun.data=mean_sdl, fun.args = list(mult = 1), 
                 # geom="pointrange", color="#333333", size = 0.5)+
  # geom_point(size = 4, shape = 21, color = "black", alpha = 0.7)+
  geom_boxplot(alpha = 1, outlier.shape = 21, outlier.size = 4, size = 0.5)+
  theme(axis.text=element_text(size=14), axis.title=element_text(size=20),
        title=element_text(size=20), legend.text=element_text(size=14),
        legend.background = element_rect(fill="transparent"),
        axis.text.x = element_text(angle = 90, hjust = 1),
        strip.text=element_text(size=18),
        axis.line = element_line(size = 1, colour = "grey80"),
        panel.border = element_blank())+
  ylab(paste0("Normalized relative abundance\n (> ", id_thresh-0.25, "% NI)"))+
  xlab("")+
  scale_y_sqrt(breaks = c(0,0.5, 1, 2 ,3), labels = c(0, 0.5, 1, 2 ,3))+ 
  guides(fill = guide_legend(nrow = 4))

print(p_sdisc_cum5)

```

```{r SDP_external-4, warning = FALSE, fig.width = 10, fig.height = 12, dpi = 500, dev = c("png", "pdf"), dependson="SDP_external-3"}
# Plot together with abundances estimated from lake MI metaG data
cowplot::plot_grid(p_sdisc_cum5, p_season1, align = "hv", nrow = 2)
```

# Posigene

```{r posi-1, warning = FALSE, fig.width = 10, fig.height = 7, dpi = 500, dev = c("png", "pdf")}
# Read results
posi_MAG8 <- data.table::fread("./posigene/Limnohabitans_sp_Rim11_results_short.tsv") %>% 
  dplyr::select(1:6)
colnames(posi_MAG8) <- c(colnames(posi_MAG8)[2:6], "Number of Sites under positive Selection")
posi_MAG8 <- posi_MAG8 %>% 
    dplyr::filter(FDR < 0.05 & `P-Value` < 0.05)

# merge with annotation 
merged_file_MAG8 <- merged_file %>% 
  dplyr::filter(Genome_id %in% 2757320398) 
posi_MAG8 <- posi_MAG8 %>% 
  mutate(Gene = as.character(Gene)) %>% 
  dplyr::left_join(., merged_file_MAG8, by = c("Gene" = "gene_oid"))

# check genes under positive selection in MAG8


```

